{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5B_allele_specific_egenes (GAVEUP)\n",
    "\n",
    "\n",
    "get allele specific egenes\n",
    "\n",
    "\n",
    "so get the egenes, cgene and ngenes for the whole library\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os, glob,sys\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import pybedtools\n",
    "import matplotlib.pyplot as plt\n",
    "import requests\n",
    "import re\n",
    "from collections import defaultdict\n",
    "import subprocess\n",
    "import xml.etree.ElementTree as ET\n",
    "import json\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_dir = 'vcf_files/analyses'\n",
    "if not os.path.exists(save_dir):\n",
    "    os.makedirs(save_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 0. get chr locations for as a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# def snp_pos2rsid(chr_id, chr_pos, org='HUMAN'):\n",
    "#     # first search for chrom info (result is in xml format)\n",
    "#     cmd = ['esearch', '-db', 'snp', '-query', '\"{}[CHR]\" AND {}[ORGN] AND {}[CPOS]'.format(chr_id, org, chr_pos)]\n",
    "#     cmd_str = ' '.join(cmd)\n",
    "#     esearch = subprocess.Popen(cmd_str, shell=True, stdout=subprocess.PIPE)\n",
    "#     esearch_res = esearch.communicate()[0].decode('utf-8')\n",
    "#     web_id = ET.fromstring(esearch_res).findall('WebEnv')[0].text\n",
    "# #     print(esearch_res)\n",
    "# #     print(web_id)\n",
    "#     # fetch the esearch results (json)\n",
    "#     efetch_command = ['efetch', '-db', 'snp', '-id',web_id]\n",
    "#     efetch_command_str = ' '.join(efetch_command)\n",
    "#     efetch_snp = subprocess.Popen(efetch_command, shell=True, stdout=subprocess.PIPE)\n",
    "#     root = ET.fromstring(efetch_snp.stdout.read().decode('utf-8'))\n",
    "#     result = root.findall('./DocumentSummary/SNP_ID')\n",
    "#     result = ['rs' + r.text for r in result]\n",
    "#     return ','.join(result)\n",
    "\n",
    "\n",
    "def snp_pos2rsid(chr_id, chr_pos):\n",
    "    url = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=snp&term={}%5BPOSITION_GRCH37%5D+AND+{}%5BCHR%5D'.format(chr_pos,chr_id)\n",
    "    res = requests.get(url)\n",
    "    rsid_arr = ['rs'+x.text for x in ET.fromstring(res.text).findall('IdList')[0].findall('Id')]\n",
    "    return rsid_arr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "asatac_file = 'vcf_files/vcf_df_filt_atac_all.csv'\n",
    "ashichip_file = 'vcf_files/vcf_df_filt_hichip_all.csv'\n",
    "asatac_df = pd.read_csv('vcf_files/vcf_df_filt_atac_all.csv',index_col=0)\n",
    "asatac_df['chr_id'] = asatac_df.CHROM.apply(lambda x: x[3:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>sample</th>\n",
       "      <th>...</th>\n",
       "      <th>FS</th>\n",
       "      <th>GT_1</th>\n",
       "      <th>GT_2</th>\n",
       "      <th>AD_1_ref</th>\n",
       "      <th>AD_1_alt</th>\n",
       "      <th>AD_2_ref</th>\n",
       "      <th>AD_2_alt</th>\n",
       "      <th>AD_1_min</th>\n",
       "      <th>AD_2_min</th>\n",
       "      <th>chr_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>762632</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>528.60</td>\n",
       "      <td>VQSRTrancheSNP99.90to100.00</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;AS_QD=14.30;BaseQRankSum=1....</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:18,19:37:99:536,0,512</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chr1</td>\n",
       "      <td>848445</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>362.60</td>\n",
       "      <td>VQSRTrancheSNP99.00to99.90</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;AS_QD=12.52;BaseQRankSum=1....</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:13,16:29:99:370,0,251</td>\n",
       "      <td>...</td>\n",
       "      <td>1.455</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chr1</td>\n",
       "      <td>856476</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>47.60</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;AS_QD=6.86;BaseQRankSum=0.3...</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:5,2:7:55:55,0,160</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>chr1</td>\n",
       "      <td>894890</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>AAGAC</td>\n",
       "      <td>244.64</td>\n",
       "      <td>.</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;AS_QD=15.31;BaseQRankSum=0....</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:10,6:16:99:252,0,395</td>\n",
       "      <td>...</td>\n",
       "      <td>9.276</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>chr1</td>\n",
       "      <td>912049</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>753.60</td>\n",
       "      <td>VQSRTrancheSNP99.00to99.90</td>\n",
       "      <td>AC=1;AF=0.500;AN=2;AS_QD=14.78;BaseQRankSum=1....</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:24,27:51:99:761,0,589</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>24</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     POS ID REF    ALT    QUAL                       FILTER  \\\n",
       "4   chr1  762632  .   T      A  528.60  VQSRTrancheSNP99.90to100.00   \n",
       "6   chr1  848445  .   G      A  362.60   VQSRTrancheSNP99.00to99.90   \n",
       "7   chr1  856476  .   A      G   47.60                         PASS   \n",
       "20  chr1  894890  .   A  AAGAC  244.64                            .   \n",
       "25  chr1  912049  .   T      C  753.60   VQSRTrancheSNP99.00to99.90   \n",
       "\n",
       "                                                 INFO          FORMAT  \\\n",
       "4   AC=1;AF=0.500;AN=2;AS_QD=14.30;BaseQRankSum=1....  GT:AD:DP:GQ:PL   \n",
       "6   AC=1;AF=0.500;AN=2;AS_QD=12.52;BaseQRankSum=1....  GT:AD:DP:GQ:PL   \n",
       "7   AC=1;AF=0.500;AN=2;AS_QD=6.86;BaseQRankSum=0.3...  GT:AD:DP:GQ:PL   \n",
       "20  AC=1;AF=0.500;AN=2;AS_QD=15.31;BaseQRankSum=0....  GT:AD:DP:GQ:PL   \n",
       "25  AC=1;AF=0.500;AN=2;AS_QD=14.78;BaseQRankSum=1....  GT:AD:DP:GQ:PL   \n",
       "\n",
       "                       sample  ...     FS  GT_1  GT_2  AD_1_ref AD_1_alt  \\\n",
       "4   0/1:18,19:37:99:536,0,512  ...  0.000   0/1   0/1        18       19   \n",
       "6   0/1:13,16:29:99:370,0,251  ...  1.455   0/1   0/1        13       16   \n",
       "7       0/1:5,2:7:55:55,0,160  ...  0.000   0/1   0/1         5        2   \n",
       "20   0/1:10,6:16:99:252,0,395  ...  9.276   0/1   0/1        10        6   \n",
       "25  0/1:24,27:51:99:761,0,589  ...  0.000   0/1   0/1        24       27   \n",
       "\n",
       "   AD_2_ref  AD_2_alt  AD_1_min  AD_2_min  chr_id  \n",
       "4        18        19        18        18       1  \n",
       "6        13        16        13        13       1  \n",
       "7         5         2         2         2       1  \n",
       "20       10         6         6         6       1  \n",
       "25       24        27        24        24       1  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asatac_df[:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### testing get rsid from hg19 loc\n",
    "\n",
    "requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=snp&term={}%5BPOSITION_GRCH37%5D+AND+{}%5BCHR%5D'.format(762632,'1')\n",
    "res = requests.get(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\\n<!DOCTYPE eSearchResult PUBLIC \"-//NLM//DTD esearch 20060628//EN\" \"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20060628/esearch.dtd\">\\n<eSearchResult><Count>3</Count><RetMax>3</RetMax><RetStart>0</RetStart><IdList>\\n<Id>61768173</Id>\\n<Id>58119245</Id>\\n<Id>3131948</Id>\\n</IdList><TranslationSet/><TranslationStack>   <TermSet>    <Term>762632[POSITION_GRCH37]</Term>    <Field>POSITION_GRCH37</Field>    <Count>9</Count>    <Explode>N</Explode>   </TermSet>   <TermSet>    <Term>1[CHR]</Term>    <Field>CHR</Field>    <Count>56436628</Count>    <Explode>N</Explode>   </TermSet>   <OP>AND</OP>  </TranslationStack><QueryTranslation>762632[POSITION_GRCH37] AND 1[CHR]</QueryTranslation></eSearchResult>\\n'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs61768173', 'rs58119245', 'rs3131948']"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "['rs'+x.text for x in ET.fromstring(res.text).findall('IdList')[0].findall('Id')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs61768173', 'rs58119245', 'rs3131948']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_pos2rsid('1',762632)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rs60458502', 'rs4828647']"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_pos2rsid('X',150934518)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### deprecated testing using esearch "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# cmd = 'esearch', '-db', 'snp', '-query', '\"{}[CHR]\" AND {}[ORGN] AND {}[CPOS]'.format('1', 'HUMAN', '762632')\n",
    "# cmd_str = ' '.join(cmd)\n",
    "# print(cmd)\n",
    "# print(cmd_str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# ##testing \n",
    "# try:\n",
    "#     subprocess.check_output(' '.join(cmd), shell=True,stderr=subprocess.STDOUT)\n",
    "# except subprocess.CalledProcessError as e:\n",
    "#     print (e.returncode)\n",
    "#     print (e.output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# esearch = subprocess.Popen(cmd_str, shell=True, stdout=subprocess.PIPE)\n",
    "# esearch_res = esearch.communicate()[0].decode('utf-8')\n",
    "# web_id = ET.fromstring(esearch_res).findall('WebEnv')[0].text\n",
    "# print(esearch_res)\n",
    "# print(web_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# efetch_command = ['efetch', '-db', 'snp', '-id',web_id, '-format', 'json']\n",
    "# efetch_command_str = ' '.join(efetch_command)\n",
    "# efetch_snp = subprocess.Popen(efetch_command_str, shell=True, stdout=subprocess.PIPE)\n",
    "# efetch_snp_res = efetch_snp.communicate()[0].decode('utf-8')\n",
    "# print(efetch_snp_res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# ET.fromstring(efetch_snp_res)#.findall('SNP_ID')[0].text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# result = root.findall('./DocumentSummary/SNP_ID')\n",
    "# result = ['rs' + r.text for r in result]\n",
    "# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# x = subprocess.Popen('esearch -db snp -query \"X[CHR]\" AND HUMAN[ORGN] AND 150934518[CPOS]',\n",
    "#                shell=True,stdout=subprocess.PIPE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# esearch = subprocess.Popen(['esearch', '-db', 'snp', '-query', '\"{}[CHR]\" AND {}[ORGN] AND {}[CPOS]'.format('X', 'HUMAN', '150934518')],\n",
    "#                 shell=True,stdout=subprocess.PIPE,stderr=subprocess.PIPE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# esearch.communicate()[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "10\n",
      "CPU times: user 1.7 s, sys: 51.7 ms, total: 1.76 s\n",
      "Wall time: 4.31 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "asatac_rowname_to_rsid = {}\n",
    "for idx, row in asatac_df.reset_index(drop=True).iterrows():\n",
    "    rowname = row['CHROM']+'_'+str(row['POS'])\n",
    "    if idx > 10:\n",
    "        break ### DEBUG\n",
    "    rsids = snp_pos2rsid(chr_pos=row['POS'],chr_id=row['CHROM'][3:])\n",
    "    asatac_rowname_to_rsid[rowname] = rsids\n",
    "    if (idx%10)==0:\n",
    "        print(idx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['name' 'X' 'Linked_SNP' 'Chr' 'Position' 'Index_SNP' 'Disease'\n",
      " 'DHS_tissue' 'eGene' 'eQTL_tissue' 'Pval' 'source' 'RS' 'pos' 'info'\n",
      " 'allele_type' 'allele' 'seq150_new' 'seq21_new' 'schizo' 'depress'\n",
      " 'bipolar' 'anxiety' 'attent' 'personality' 'panic' 'traum' 'autism' 'ocd'\n",
      " 'Ref' 'Alt' 'Type' 'Length' 'AnnoType' 'Consequence' 'ConsScore'\n",
      " 'ConsDetail' 'GC' 'CpG' 'motifECount' 'motifEName' 'motifEHIPos'\n",
      " 'motifEScoreChng' 'oAA' 'nAA' 'GeneID' 'FeatureID' 'GeneName' 'CCDS'\n",
      " 'Intron' 'Exon' 'cDNApos' 'relcDNApos' 'CDSpos' 'relCDSpos' 'protPos'\n",
      " 'relProtPos' 'Domain' 'Dst2Splice' 'Dst2SplType' 'minDistTSS'\n",
      " 'minDistTSE' 'SIFTcat' 'SIFTval' 'PolyPhenCat' 'PolyPhenVal' 'priPhCons'\n",
      " 'mamPhCons' 'verPhCons' 'priPhyloP' 'mamPhyloP' 'verPhyloP' 'bStatistic'\n",
      " 'targetScan' 'mirSVR.Score' 'mirSVR.E' 'mirSVR.Aln' 'cHmmTssA'\n",
      " 'cHmmTssAFlnk' 'cHmmTxFlnk' 'cHmmTx' 'cHmmTxWk' 'cHmmEnhG' 'cHmmEnh'\n",
      " 'cHmmZnfRpts' 'cHmmHet' 'cHmmTssBiv' 'cHmmBivFlnk' 'cHmmEnhBiv'\n",
      " 'cHmmReprPC' 'cHmmReprPCWk' 'cHmmQuies' 'GerpRS' 'GerpRSpval' 'GerpN'\n",
      " 'GerpS' 'TFBS' 'TFBSPeaks' 'TFBSPeaksMax' 'tOverlapMotifs' 'motifDist'\n",
      " 'Segway' 'EncH3K27Ac' 'EncH3K4Me1' 'EncH3K4Me3' 'EncExp' 'EncNucleo'\n",
      " 'EncOCC' 'EncOCCombPVal' 'EncOCDNasePVal' 'EncOCFairePVal'\n",
      " 'EncOCpolIIPVal' 'EncOCctcfPVal' 'EncOCmycPVal' 'EncOCDNaseSig'\n",
      " 'EncOCFaireSig' 'EncOCpolIISig' 'EncOCctcfSig' 'EncOCmycSig' 'Grantham'\n",
      " 'Dist2Mutation' 'Freq100bp' 'Rare100bp' 'Sngl100bp' 'Freq1000bp'\n",
      " 'Rare1000bp' 'Sngl1000bp' 'Freq10000bp' 'Rare10000bp' 'Sngl10000bp'\n",
      " 'dbscSNV.ada_score' 'dbscSNV.rf_score' 'RawScore' 'PHRED' 'num_disease'\n",
      " 'disease_interest' 'pass_thres_low' 'pass_thres' 'CADD_score_bool'\n",
      " 'rowname']\n",
      "(4440, 140)\n",
      "(2221,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/mguo123/opt/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (63,65) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>X</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Index_SNP</th>\n",
       "      <th>Disease</th>\n",
       "      <th>DHS_tissue</th>\n",
       "      <th>eGene</th>\n",
       "      <th>eQTL_tissue</th>\n",
       "      <th>...</th>\n",
       "      <th>dbscSNV.ada_score</th>\n",
       "      <th>dbscSNV.rf_score</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>num_disease</th>\n",
       "      <th>disease_interest</th>\n",
       "      <th>pass_thres_low</th>\n",
       "      <th>pass_thres</th>\n",
       "      <th>CADD_score_bool</th>\n",
       "      <th>rowname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1.111243375.all_alt.G</td>\n",
       "      <td>75004</td>\n",
       "      <td>rs12085135</td>\n",
       "      <td>chr1</td>\n",
       "      <td>111243375</td>\n",
       "      <td>rs12061304</td>\n",
       "      <td>panic disorder</td>\n",
       "      <td>t-helper_17_cell</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.197208</td>\n",
       "      <td>0.728</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_111243375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1.111243375.Ref.C</td>\n",
       "      <td>75004</td>\n",
       "      <td>rs12085135</td>\n",
       "      <td>chr1</td>\n",
       "      <td>111243375</td>\n",
       "      <td>rs12061304</td>\n",
       "      <td>panic disorder</td>\n",
       "      <td>t-helper_17_cell</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.197208</td>\n",
       "      <td>0.728</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_111243375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1.113938499.all_alt.C</td>\n",
       "      <td>1977</td>\n",
       "      <td>rs74395031</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113938499</td>\n",
       "      <td>rs115870915</td>\n",
       "      <td>type 2 diabetes,Obsessive Compulsive Disorder</td>\n",
       "      <td>astrocyte_of_the_spinal_cord</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.309426</td>\n",
       "      <td>7.336</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_113938499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1.113938499.Ref.T</td>\n",
       "      <td>1977</td>\n",
       "      <td>rs74395031</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113938499</td>\n",
       "      <td>rs115870915</td>\n",
       "      <td>type 2 diabetes,Obsessive Compulsive Disorder</td>\n",
       "      <td>astrocyte_of_the_spinal_cord</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.309426</td>\n",
       "      <td>7.336</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_113938499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chr1.11712109.all_alt.T</td>\n",
       "      <td>569</td>\n",
       "      <td>rs2103665</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11712109</td>\n",
       "      <td>rs115049252</td>\n",
       "      <td>schizo</td>\n",
       "      <td>spleen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.193843</td>\n",
       "      <td>5.751</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_11712109</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 140 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       name      X  Linked_SNP   Chr   Position    Index_SNP  \\\n",
       "1  chr1.111243375.all_alt.G  75004  rs12085135  chr1  111243375   rs12061304   \n",
       "2      chr1.111243375.Ref.C  75004  rs12085135  chr1  111243375   rs12061304   \n",
       "3  chr1.113938499.all_alt.C   1977  rs74395031  chr1  113938499  rs115870915   \n",
       "4      chr1.113938499.Ref.T   1977  rs74395031  chr1  113938499  rs115870915   \n",
       "5   chr1.11712109.all_alt.T    569   rs2103665  chr1   11712109  rs115049252   \n",
       "\n",
       "                                         Disease  \\\n",
       "1                                 panic disorder   \n",
       "2                                 panic disorder   \n",
       "3  type 2 diabetes,Obsessive Compulsive Disorder   \n",
       "4  type 2 diabetes,Obsessive Compulsive Disorder   \n",
       "5                                         schizo   \n",
       "\n",
       "                     DHS_tissue eGene eQTL_tissue  ...  dbscSNV.ada_score  \\\n",
       "1              t-helper_17_cell   NaN         NaN  ...                NaN   \n",
       "2              t-helper_17_cell   NaN         NaN  ...                NaN   \n",
       "3  astrocyte_of_the_spinal_cord   NaN         NaN  ...                NaN   \n",
       "4  astrocyte_of_the_spinal_cord   NaN         NaN  ...                NaN   \n",
       "5                        spleen   NaN         NaN  ...                NaN   \n",
       "\n",
       "  dbscSNV.rf_score  RawScore  PHRED num_disease disease_interest  \\\n",
       "1              NaN -0.197208  0.728           1             True   \n",
       "2              NaN -0.197208  0.728           1             True   \n",
       "3              NaN  0.309426  7.336           1             True   \n",
       "4              NaN  0.309426  7.336           1             True   \n",
       "5              NaN  0.193843  5.751           1            False   \n",
       "\n",
       "  pass_thres_low pass_thres CADD_score_bool         rowname  \n",
       "1           True      False             NaN  chr1_111243375  \n",
       "2           True      False             NaN  chr1_111243375  \n",
       "3           True       True             NaN  chr1_113938499  \n",
       "4           True       True             NaN  chr1_113938499  \n",
       "5           True       True             NaN   chr1_11712109  \n",
       "\n",
       "[5 rows x 140 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_csv = '../../psych_lib_info.csv'\n",
    "lib_df = pd.read_csv(lib_csv,index_col=0)\n",
    "lib_df['rowname']= lib_df.Chr + '_' + lib_df.Position.map(str)\n",
    "\n",
    "print(lib_df.columns.values)\n",
    "print(lib_df.shape)\n",
    "print(lib_df.rowname.value_counts().shape)\n",
    "# lib_df = lib_df[lib_df.allele_type=='all_alt']\n",
    "# lib_df = lib_df[['name','rowname','Linked_SNP', 'Chr', 'Position', 'Index_SNP','eGene','eQTL_tissue','Pval','source',\n",
    "#                  'schizo', 'depress', 'bipolar', 'anxiety', 'attent', 'personality', 'panic', 'traum', 'autism', 'ocd','Consequence',\n",
    "#                 ]]\n",
    "\n",
    "lib_df[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "diseases =sorted( ['schizo', 'depress', 'bipolar', 'anxiety', 'attent',\n",
    " 'personality', 'panic', 'traum', 'autism', 'ocd'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_loc_to_linked_rsid = pd.Series(lib_df.Linked_SNP.values, index=lib_df.rowname.values).to_dict()\n",
    "snp_loc_to_index_rsid = pd.Series(lib_df.Index_SNP.values, index=lib_df.rowname.values).to_dict()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. ngenes (max dist == 100 kb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "MAX_DIST = 100000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>TSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1846266</td>\n",
       "      <td>1846267</td>\n",
       "      <td>CALML6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>231376933</td>\n",
       "      <td>231376934</td>\n",
       "      <td>C1orf131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>241683061</td>\n",
       "      <td>241683062</td>\n",
       "      <td>FH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>120202421</td>\n",
       "      <td>120202422</td>\n",
       "      <td>PHGDH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>204135465</td>\n",
       "      <td>204135466</td>\n",
       "      <td>REN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr      start       stop       TSS\n",
       "0  chr1    1846266    1846267    CALML6\n",
       "1  chr1  231376933  231376934  C1orf131\n",
       "2  chr1  241683061  241683062        FH\n",
       "3  chr1  120202421  120202422     PHGDH\n",
       "4  chr1  204135465  204135466       REN"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_region = pd.read_csv('/Users/mguo123/Google Drive/1_khavari/omics_project-LD/pan_omics/data/external/gencode.v19.gene.bed',sep='\\t',header=None)\n",
    "gene_region.columns = ['chr','start','stop','gene']\n",
    "gene_region_bed = pybedtools.BedTool.from_dataframe(gene_region).sort()\n",
    "\n",
    "exon_region = pd.read_csv('/Users/mguo123/Google Drive/1_khavari/omics_project-LD/pan_omics/data/external/gencode.v19.exon.bed',sep='\\t',header=None)\n",
    "exon_region.columns = ['chr','start','stop','gene']\n",
    "exon_region_bed = pybedtools.BedTool.from_dataframe(exon_region).sort()\n",
    "\n",
    "TSS_file = '/Users/mguo123/Google Drive/1_khavari/omics_project-LD/pan_omics/data/external/TSS_annon_hg19.csv'\n",
    "TSS_df = pd.read_csv(TSS_file,index_col=0)\n",
    "TSS_df['stop'] = TSS_df['loc']+1\n",
    "TSS_df = TSS_df[['chr', 'loc','stop','TSS']]\n",
    "TSS_df.columns = ['chr','start','stop','TSS']\n",
    "TSS_bed = pybedtools.BedTool.from_dataframe(TSS_df).sort()\n",
    "TSS_df[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/mguo123/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2069681</td>\n",
       "      <td>2069682</td>\n",
       "      <td>chr1_2069681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2069681</td>\n",
       "      <td>2069682</td>\n",
       "      <td>chr1_2069681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2337032</td>\n",
       "      <td>2337033</td>\n",
       "      <td>chr1_2337032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2337032</td>\n",
       "      <td>2337033</td>\n",
       "      <td>chr1_2337032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2337277</td>\n",
       "      <td>2337278</td>\n",
       "      <td>chr1_2337277</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom    start      end          name\n",
       "0  chr1  2069681  2069682  chr1_2069681\n",
       "1  chr1  2069681  2069682  chr1_2069681\n",
       "2  chr1  2337032  2337033  chr1_2337032\n",
       "3  chr1  2337032  2337033  chr1_2337032\n",
       "4  chr1  2337277  2337278  chr1_2337277"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_df = lib_df[['Chr','Position','rowname']]\n",
    "gwas_df.columns = ['chr','start','name']\n",
    "gwas_df['stop'] = gwas_df.start +1\n",
    "gwas_df = gwas_df[['chr','start','stop','name']]\n",
    "gwas_bed = pybedtools.BedTool.from_dataframe(gwas_df).sort()\n",
    "gwas_bed.to_dataframe()[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(610,)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp_id</th>\n",
       "      <th>TSS</th>\n",
       "      <th>distance</th>\n",
       "      <th>name</th>\n",
       "      <th>X</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Index_SNP</th>\n",
       "      <th>Disease</th>\n",
       "      <th>...</th>\n",
       "      <th>dbscSNV.ada_score</th>\n",
       "      <th>dbscSNV.rf_score</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>num_disease</th>\n",
       "      <th>disease_interest</th>\n",
       "      <th>pass_thres_low</th>\n",
       "      <th>pass_thres</th>\n",
       "      <th>CADD_score_bool</th>\n",
       "      <th>rowname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr1_2069681</td>\n",
       "      <td>AL590822.2</td>\n",
       "      <td>51556</td>\n",
       "      <td>chr1.2069681.all_alt.T</td>\n",
       "      <td>1545</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2069681</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>Schizophrenia</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.131842</td>\n",
       "      <td>1.130</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1_2069681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1_2069681</td>\n",
       "      <td>AL590822.2</td>\n",
       "      <td>51556</td>\n",
       "      <td>chr1.2069681.Ref.C</td>\n",
       "      <td>1545</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2069681</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>Schizophrenia</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.131842</td>\n",
       "      <td>1.130</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1_2069681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1_2069681</td>\n",
       "      <td>AL590822.2</td>\n",
       "      <td>51556</td>\n",
       "      <td>chr1.2069681.all_alt.T</td>\n",
       "      <td>1545</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2069681</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>Schizophrenia</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.131842</td>\n",
       "      <td>1.130</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1_2069681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1_2069681</td>\n",
       "      <td>AL590822.2</td>\n",
       "      <td>51556</td>\n",
       "      <td>chr1.2069681.Ref.C</td>\n",
       "      <td>1545</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2069681</td>\n",
       "      <td>rs3753242</td>\n",
       "      <td>Schizophrenia</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.131842</td>\n",
       "      <td>1.130</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1_2069681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1_2337032</td>\n",
       "      <td>PEX10</td>\n",
       "      <td>8204</td>\n",
       "      <td>chr1.2337032.all_alt.T</td>\n",
       "      <td>451</td>\n",
       "      <td>rs1129171</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2337032</td>\n",
       "      <td>NaN</td>\n",
       "      <td>schizo</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.093925</td>\n",
       "      <td>1.458</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_2337032</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 143 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         snp_id         TSS  distance                    name     X  \\\n",
       "0  chr1_2069681  AL590822.2     51556  chr1.2069681.all_alt.T  1545   \n",
       "1  chr1_2069681  AL590822.2     51556      chr1.2069681.Ref.C  1545   \n",
       "2  chr1_2069681  AL590822.2     51556  chr1.2069681.all_alt.T  1545   \n",
       "3  chr1_2069681  AL590822.2     51556      chr1.2069681.Ref.C  1545   \n",
       "4  chr1_2337032       PEX10      8204  chr1.2337032.all_alt.T   451   \n",
       "\n",
       "  Linked_SNP   Chr  Position  Index_SNP        Disease  ... dbscSNV.ada_score  \\\n",
       "0  rs3753242  chr1   2069681  rs3753242  Schizophrenia  ...               NaN   \n",
       "1  rs3753242  chr1   2069681  rs3753242  Schizophrenia  ...               NaN   \n",
       "2  rs3753242  chr1   2069681  rs3753242  Schizophrenia  ...               NaN   \n",
       "3  rs3753242  chr1   2069681  rs3753242  Schizophrenia  ...               NaN   \n",
       "4  rs1129171  chr1   2337032        NaN         schizo  ...               NaN   \n",
       "\n",
       "  dbscSNV.rf_score  RawScore  PHRED num_disease disease_interest  \\\n",
       "0              NaN -0.131842  1.130           1            False   \n",
       "1              NaN -0.131842  1.130           1            False   \n",
       "2              NaN -0.131842  1.130           1            False   \n",
       "3              NaN -0.131842  1.130           1            False   \n",
       "4              NaN -0.093925  1.458           1            False   \n",
       "\n",
       "   pass_thres_low pass_thres CADD_score_bool       rowname  \n",
       "0            True       True            True  chr1_2069681  \n",
       "1            True       True            True  chr1_2069681  \n",
       "2            True       True            True  chr1_2069681  \n",
       "3            True       True            True  chr1_2069681  \n",
       "4            True       True             NaN  chr1_2337032  \n",
       "\n",
       "[5 rows x 143 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_ngene_bed = gwas_bed.closest(TSS_bed,d=True)\n",
    "gwas_ngene_bed.saveas(os.path.join('../../psych_lib_ngene_TSS.bed'))\n",
    "gwas_ngene_df = gwas_ngene_bed.to_dataframe()\n",
    "gwas_ngene_df = gwas_ngene_df[['name','thickEnd','itemRgb']]\n",
    "gwas_ngene_df.columns = ['snp_id','TSS','distance']\n",
    "gwas_ngene_df = gwas_ngene_df.merge(lib_df, how='left', left_on='snp_id',right_on='rowname')\n",
    "gwas_ngene_df = gwas_ngene_df[gwas_ngene_df.distance < MAX_DIST]\n",
    "print(gwas_ngene_df.TSS.unique().shape)\n",
    "gwas_ngene_df[:5]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_to_annongene_dict = pd.Series(lib_df.GeneName.values, index=lib_df.rowname.values).to_dict()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_to_ngene_df = gwas_ngene_df[['snp_id','TSS']].drop_duplicates()#.snp_id.value_counts()\n",
    "snp_to_ngene_dict = pd.Series(snp_to_ngene_df.TSS.values, index=snp_to_ngene_df.snp_id.values).to_dict()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['APOPT1' 'RP11-73M18.2']\n"
     ]
    }
   ],
   "source": [
    "#one case with 2\n",
    "print(gwas_ngene_df[gwas_ngene_df.snp_id=='chr14_104029449'].TSS.unique())\n",
    "snp_to_ngene_dict['chr14_104029449'] = 'APOPT1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1755"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(snp_to_ngene_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "there are 1755 ngenes of the 2221 possible to be mapped for snps add to dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_df['ngene'] = lib_df.rowname.map(snp_to_ngene_dict)\n",
    "lib_df.ngene.fillna('',inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. get egenes haploreg (also ngenes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_table_header(header_line):\n",
    "    new_table_line = []\n",
    "    for x in header_line:\n",
    "        x = x.replace('<b>', '')\n",
    "        x = x.replace('</b>', '')\n",
    "        new_table_line.append(x.strip('>'))\n",
    "    return new_table_line\n",
    "\n",
    "def get_tables(rsid):\n",
    "    response = requests.get('https://pubs.broadinstitute.org/mammals/haploreg/detail_v4.1.php?query=&id=' + rsid)\n",
    "    table_lines =re.findall(r'<tr>(.*?)</tr>', str(response.content), re.M|re.I|re.S)\n",
    "\n",
    "    type_table = ''\n",
    "    type_to_headers = {}\n",
    "    table_dict = defaultdict(list)\n",
    "\n",
    "    for table_line in table_lines:\n",
    "    #     print(re.findall(r'<td(.*?)</td>',table_line))\n",
    "        table_line = re.findall(r'<td(.*?)</td>',table_line)\n",
    "        if len(table_line)==0:\n",
    "            continue\n",
    "\n",
    "        if ('<b>chr' in table_line[0]) or ('AFR' in table_line[0]):\n",
    "            type_table = 'freq'\n",
    "            type_to_headers[type_table] = ['chr_hg19', 'pos_hg19', 'chr_hg38', 'pos_hg38', 'ref','alt', 'freq_AFR', 'freq_AMR','freq_ASN','freq_EUR', 'constraint_GERP', 'constraint_SiPhy', 'functional_annon']\n",
    "        elif 'Source' in table_line[0]:\n",
    "            type_table = 'ngene'\n",
    "            type_to_headers[type_table] = make_table_header(table_line)\n",
    "        elif 'Epigenome' in table_line[0]:\n",
    "            type_table = 'epi'\n",
    "            type_to_headers[type_table] = make_table_header(table_line)\n",
    "        elif 'Cell ID' in table_line[0]:\n",
    "            type_table = 'chip'\n",
    "            type_to_headers[type_table] = make_table_header(table_line)\n",
    "        elif 'Trait' in table_line[0]:\n",
    "            type_table = 'grasp'\n",
    "            type_to_headers[type_table] = make_table_header(table_line)\n",
    "        elif 'Study ID' in table_line[0]:\n",
    "            type_table = 'eqtl'\n",
    "            type_to_headers[type_table] = ['Study ID',  'Paper Title',  'PMID',  'Tissue','Correlated gene',  'p-value']\n",
    "        elif 'Position Weight Matrix ID' in table_line[0]:\n",
    "            type_table = 'motif'\n",
    "            type_to_headers[type_table] =['motif','strand','ref','alt']\n",
    "        else: # content line\n",
    "    #         print(type_table)\n",
    "            table_dict[type_table].append([x.strip('>') for x in table_line])\n",
    "        \n",
    "    table_df_dict = {}\n",
    "    for key, list_table in table_dict.items():\n",
    "        header = type_to_headers[key]\n",
    "        if key == 'freq':\n",
    "            table_df_dict[key] = pd.DataFrame(list_table[:-1], columns = header)\n",
    "        else:\n",
    "            table_df_dict[key] = pd.DataFrame(list_table, columns = header)\n",
    "    return table_df_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "rsids = sorted(lib_df.Linked_SNP.unique())\n",
    "len(rsids),lib_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "rsid_to_tables = {}\n",
    "for idx, rsid in enumerate(rsids):\n",
    "    if (idx%100)==0:\n",
    "        print(idx,rsid)\n",
    "    rsid_to_tables[rsid] = get_tables(rsid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Testing\n",
    "rsid_to_tables['rs3824756'].keys()\n",
    "egenes_df = rsid_to_tables[rsid]['eqtl']\n",
    "egenes_df = egenes_df.groupby('Tissue').agg({'Correlated gene':'|'.join}).reset_index()\n",
    "egenes_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_results = {}\n",
    "for rsid in  rsids:\n",
    "    if 'ngene' in rsid_to_tables[rsid]:\n",
    "        negene = '|'.join(rsid_to_tables[rsid]['ngene']['Common name'].unique())\n",
    "    if 'eqtl' in rsid_to_tables[rsid]:\n",
    "        egenes_df = rsid_to_tables[rsid]['eqtl']\n",
    "        egenes_df = egenes_df.groupby('Tissue').agg({'Correlated gene':'|'.join}).reset_index()\n",
    "#         egenes =  rsid_to_tables[rsid]['eqtl']['Correlated gene'].unique()\n",
    "        tissue_to_egene = pd.Series(egenes_df['Correlated gene'].values, index=egenes_df.Tissue.values).to_dict()\n",
    "#         print(rsid, negene, tissue_to_egene.get('Nerve_Tibial'), tissue_to_egene.get('Whole Blood') , '|'.join(rsid_to_tables[rsid]['eqtl']['Correlated gene'].unique()))  \n",
    "        gene_results[rsid] = [negene, tissue_to_egene.get('Nerve_Tibial'), tissue_to_egene.get('Whole Blood') , '|'.join(rsid_to_tables[rsid]['eqtl']['Correlated gene'].unique())]\n",
    "    else:\n",
    "        gene_results[rsid] = [negene,'', '', 'no eqtl found']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_results_df = pd.DataFrame(gene_results).T.reset_index()\n",
    "\n",
    "gene_results_df.columns=['Linked_SNP','ngene_haploreg','egene_nerve','egene_blood','egene_all']\n",
    "gene_results_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_results_df.to_csv(os.path.join(save_dir, 'gene_results_df_haploreg.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_df_egene = lib_df.merge(gene_results_df,how='left', on='Linked_SNP')\n",
    "lib_df_egene.fillna('',inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_df_egene.to_csv(os.path.join(save_dir, 'lib_df_post_haploreg.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(lib_df_egene.egene_all.unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. get egenes gtex\n",
    "process\n",
    "- A. decide gtex tissues to use\n",
    "- B. find all eqtls associated with snp id using api\n",
    "- C. filter all egenes (from part #2 and #3 through rna filter)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3A choose tissues \n",
    "\n",
    "['Adipose_Subcutaneous', 'Adipose_Visceral_Omentum', 'Adrenal_Gland', 'Artery_Aorta', 'Artery_Coronary', 'Artery_Tibial', 'Bladder', 'Brain_Amygdala', 'Brain_Anterior_cingulate_cortex_BA24', 'Brain_Caudate_basal_ganglia', 'Brain_Cerebellar_Hemisphere', 'Brain_Cerebellum', 'Brain_Cortex', 'Brain_Frontal_Cortex_BA9', 'Brain_Hippocampus', 'Brain_Hypothalamus', 'Brain_Nucleus_accumbens_basal_ganglia', 'Brain_Putamen_basal_ganglia', 'Brain_Spinal_cord_cervical_c-1', 'Brain_Substantia_nigra', 'Breast_Mammary_Tissue', 'Cells_EBV-transformed_lymphocytes', 'Cells_Cultured_fibroblasts', 'Cells_Transformed_fibroblasts', 'Cervix_Ectocervix', 'Cervix_Endocervix', 'Colon_Sigmoid', 'Colon_Transverse', 'Esophagus_Gastroesophageal_Junction', 'Esophagus_Mucosa', 'Esophagus_Muscularis', 'Fallopian_Tube', 'Heart_Atrial_Appendage', 'Heart_Left_Ventricle', 'Kidney_Cortex', 'Kidney_Medulla', 'Liver', 'Lung', 'Minor_Salivary_Gland', 'Muscle_Skeletal', 'Nerve_Tibial', 'Ovary', 'Pancreas', 'Pituitary', 'Prostate', 'Skin_Not_Sun_Exposed_Suprapubic', 'Skin_Sun_Exposed_Lower_leg', 'Small_Intestine_Terminal_Ileum', 'Spleen', 'Stomach', 'Testis', 'Thyroid', 'Uterus', 'Vagina', 'Whole_Blood']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissues = ['Brain_Amygdala', 'Brain_Anterior_cingulate_cortex_BA24', \n",
    "           'Brain_Caudate_basal_ganglia', 'Brain_Cerebellar_Hemisphere', 'Brain_Cerebellum', 'Brain_Cortex', \n",
    "           'Brain_Frontal_Cortex_BA9', 'Brain_Hippocampus', 'Brain_Hypothalamus', 'Brain_Nucleus_accumbens_basal_ganglia',\n",
    "           'Brain_Putamen_basal_ganglia', 'Brain_Spinal_cord_cervical_c-1', 'Brain_Substantia_nigra',\n",
    "           'Nerve_Tibial','Whole_Blood']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3B. requests gtex api portal\n",
    " \n",
    " with rsid or variant\n",
    "- https://gtexportal.org/rest/v1/association/singleTissueEqtl?format=json&snpId=rs1129171&tissueSiteDetailId=Whole_Blood&datasetId=gtex_v8\n",
    "- url = 'https://gtexportal.org/rest/v1/association/singleTissueEqtl?format=json&tissueSiteDetailId={}&variantId={}&datasetId=gtex_v8'.format(tissue, variant_id)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 rs10022045 (6, 11)\n",
      "3 rs10050527 (8, 11)\n",
      "4 rs10058246 (9, 11)\n",
      "6 rs10083367 (15, 11)\n",
      "7 rs10083370 (21, 11)\n",
      "8 rs10101168 (42, 11)\n",
      "11 rs10129426 (57, 11)\n",
      "12 rs10131184 (60, 11)\n",
      "13 rs10135117 (65, 11)\n",
      "14 rs10135763 (66, 11)\n",
      "15 rs10136026 (72, 11)\n",
      "16 rs10136622 (77, 11)\n",
      "17 rs10137887 (82, 11)\n",
      "19 rs10142176 (86, 11)\n",
      "20 rs10148162 (91, 11)\n",
      "21 rs10148269 (94, 11)\n",
      "22 rs10149052 (97, 11)\n",
      "23 rs10149470 (112, 11)\n",
      "24 rs1014972 (113, 11)\n",
      "27 rs10191980 (125, 11)\n",
      "28 rs10192179 (137, 11)\n",
      "29 rs1019307 (140, 11)\n",
      "30 rs1019309 (143, 11)\n",
      "31 rs10214031 (145, 11)\n",
      "32 rs10220477 (149, 11)\n",
      "34 rs1024197 (153, 11)\n",
      "35 rs10267593 (155, 11)\n",
      "36 rs10268609 (157, 11)\n",
      "37 rs1027746 (159, 11)\n",
      "38 rs1040193 (161, 11)\n",
      "40 rs10404887 (172, 11)\n",
      "41 rs10414643 (183, 11)\n",
      "42 rs10414830 (205, 11)\n",
      "44 rs10419245 (224, 11)\n",
      "45 rs1042169 (239, 11)\n",
      "46 rs10423745 (240, 11)\n",
      "47 rs10425465 (241, 11)\n",
      "48 rs10433708 (247, 11)\n",
      "49 rs1043782 (265, 11)\n",
      "50 rs1044507 (284, 11)\n",
      "51 rs1045582 (290, 11)\n",
      "52 rs10456362 (358, 11)\n",
      "53 rs10456390 (380, 11)\n",
      "54 rs1046088 (382, 11)\n",
      "55 rs1046411 (424, 11)\n",
      "56 rs1046778 (435, 11)\n",
      "57 rs1047997 (495, 11)\n",
      "58 rs10481697 (501, 11)\n",
      "59 rs10481698 (507, 11)\n",
      "62 rs1050244 (510, 11)\n",
      "63 rs1050347 (515, 11)\n",
      "64 rs1053338 (521, 11)\n",
      "65 rs1053651 (535, 11)\n",
      "66 rs1054442 (548, 11)\n",
      "67 rs1057137 (551, 11)\n",
      "68 rs1062274 (552, 11)\n",
      "70 rs1067330 (554, 11)\n",
      "71 rs1067338 (556, 11)\n",
      "72 rs1067344 (557, 11)\n",
      "74 rs10748839 (599, 11)\n",
      "75 rs10749635 (605, 11)\n",
      "76 rs10760206 (609, 11)\n",
      "80 rs10779764 (618, 11)\n",
      "82 rs10783299 (630, 11)\n",
      "83 rs10786713 (671, 11)\n",
      "84 rs10786729 (710, 11)\n",
      "87 rs10797593 (711, 11)\n",
      "88 rs10805098 (717, 11)\n",
      "89 rs10805099 (723, 11)\n",
      "90 rs10805100 (729, 11)\n",
      "93 rs10821745 (730, 11)\n",
      "94 rs10822058 (733, 11)\n",
      "98 rs10830363 (751, 11)\n",
      "106 rs10838677 (754, 11)\n",
      "107 rs1084439 (756, 11)\n",
      "108 rs10852934 (767, 11)\n",
      "109 rs10864543 (778, 11)\n",
      "111 rs10883765 (782, 11)\n",
      "112 rs10883805 (821, 11)\n",
      "114 rs10885636 (822, 11)\n",
      "115 rs10888257 (827, 11)\n",
      "116 rs10896636 (842, 11)\n",
      "117 rs10898129 (843, 11)\n",
      "118 rs10920357 (846, 11)\n",
      "119 rs10926981 (856, 11)\n",
      "120 rs10935184 (880, 11)\n",
      "121 rs10938170 (886, 11)\n",
      "122 rs10938176 (892, 11)\n",
      "123 rs10941607 (893, 11)\n",
      "124 rs10941608 (894, 11)\n",
      "127 rs10952077 (895, 11)\n",
      "129 rs1095626 (898, 11)\n",
      "130 rs1095627 (901, 11)\n",
      "132 rs10967575 (905, 11)\n",
      "133 rs10973954 (916, 11)\n",
      "135 rs10985408 (921, 11)\n",
      "136 rs10985454 (926, 11)\n",
      "137 rs10988519 (940, 11)\n",
      "138 rs10992234 (948, 11)\n",
      "140 rs10995311 (952, 11)\n",
      "144 rs11012 (1127, 11)\n",
      "145 rs11018751 (1145, 11)\n",
      "147 rs1106409 (1156, 11)\n",
      "148 rs11075922 (1157, 11)\n",
      "149 rs11075924 (1159, 11)\n",
      "151 rs11078919 (1173, 11)\n",
      "152 rs11083979 (1184, 11)\n",
      "153 rs11085829 (1185, 11)\n",
      "155 rs11121827 (1194, 11)\n",
      "156 rs11121829 (1203, 11)\n",
      "158 rs11124578 (1204, 11)\n",
      "159 rs11126399 (1224, 11)\n",
      "160 rs1112730 (1281, 11)\n",
      "161 rs111334345 (1286, 11)\n",
      "162 rs11136093 (1310, 11)\n",
      "167 rs11164843 (1317, 11)\n",
      "170 rs11167605 (1367, 11)\n",
      "171 rs11168838 (1388, 11)\n",
      "172 rs11168839 (1404, 11)\n",
      "174 rs111776518 (1409, 11)\n",
      "175 rs11178984 (1410, 11)\n",
      "177 rs111896713 (1478, 11)\n",
      "179 rs11208754 (1479, 11)\n",
      "181 rs112099181 (1497, 11)\n",
      "182 rs11209951 (1502, 11)\n",
      "183 rs11209952 (1507, 11)\n",
      "184 rs11210916 (1508, 11)\n",
      "189 rs1122326 (1515, 11)\n",
      "191 rs1124999 (1518, 11)\n",
      "192 rs112513913 (1523, 11)\n",
      "194 rs11263858 (1525, 11)\n",
      "195 rs11263861 (1526, 11)\n",
      "196 rs11264118 (1527, 11)\n",
      "198 rs112814824 (1529, 11)\n",
      "199 rs112827498 (1532, 11)\n",
      "200 rs1129171 (1541, 11)\n",
      "201 rs1129694 (1543, 11)\n",
      "202 rs113027608 (1544, 11)\n",
      "203 rs113059374 (1548, 11)\n",
      "205 rs113458968 (1549, 11)\n",
      "208 rs113872928 (1551, 11)\n",
      "209 rs113934837 (1552, 11)\n",
      "211 rs1139652 (1591, 11)\n",
      "212 rs1139653 (1621, 11)\n",
      "215 rs1141095 (1629, 11)\n",
      "220 rs1147169 (1633, 11)\n",
      "223 rs114927745 (1650, 11)\n",
      "225 rs1150694 (1753, 11)\n",
      "226 rs115070660 (1754, 11)\n",
      "227 rs1150754 (1812, 11)\n",
      "229 rs11509880 (1814, 11)\n",
      "231 rs1152588 (1817, 11)\n",
      "232 rs1152591 (1820, 11)\n",
      "233 rs115320070 (1821, 11)\n",
      "235 rs115428878 (1823, 11)\n",
      "236 rs11544238 (1830, 11)\n",
      "238 rs11550286 (1831, 11)\n",
      "239 rs11550287 (1832, 11)\n",
      "240 rs115564630 (1833, 11)\n",
      "241 rs11556769 (1847, 11)\n",
      "242 rs1155751 (1854, 11)\n",
      "243 rs11570051 (1861, 11)\n",
      "244 rs11570078 (1869, 11)\n",
      "245 rs11570115 (1875, 11)\n",
      "249 rs11586191 (1877, 11)\n",
      "250 rs11586985 (1883, 11)\n",
      "251 rs11592442 (1887, 11)\n",
      "253 rs11602109 (1900, 11)\n",
      "255 rs11606396 (1913, 11)\n",
      "256 rs116100977 (1915, 11)\n",
      "257 rs11612500 (1921, 11)\n",
      "258 rs11615336 (1927, 11)\n",
      "260 rs11635624 (1938, 11)\n",
      "261 rs11636934 (1942, 11)\n",
      "262 rs11637666 (1982, 11)\n",
      "263 rs11638290 (2026, 11)\n",
      "264 rs11638554 (2073, 11)\n",
      "269 rs11649804 (2079, 11)\n",
      "270 rs11651280 (2080, 11)\n",
      "271 rs11651497 (2091, 11)\n",
      "273 rs11658786 (2104, 11)\n",
      "275 rs116855242 (2105, 11)\n",
      "277 rs11692886 (2110, 11)\n",
      "281 rs1170443 (2120, 11)\n",
      "282 rs11706826 (2123, 11)\n",
      "283 rs117114519 (2124, 11)\n",
      "285 rs11717383 (2152, 11)\n",
      "289 rs117465668 (2153, 11)\n",
      "291 rs11759977 (2154, 11)\n",
      "292 rs11770612 (2155, 11)\n",
      "295 rs11780207 (2177, 11)\n",
      "298 rs11790613 (2195, 11)\n",
      "299 rs11792859 (2201, 11)\n",
      "304 rs118061771 (2202, 11)\n",
      "306 rs118132287 (2203, 11)\n",
      "307 rs11865690 (2237, 11)\n",
      "308 rs1187088 (2239, 11)\n",
      "309 rs11878568 (2251, 11)\n",
      "310 rs11888760 (2255, 11)\n",
      "311 rs11895190 (2277, 11)\n",
      "312 rs11906612 (2289, 11)\n",
      "313 rs1193508 (2292, 11)\n",
      "314 rs11935648 (2298, 11)\n",
      "315 rs11943813 (2300, 11)\n",
      "317 rs11970439 (2353, 11)\n",
      "318 rs11986274 (2373, 11)\n",
      "320 rs12033462 (2375, 11)\n",
      "321 rs12033824 (2377, 11)\n",
      "322 rs12035943 (2379, 11)\n",
      "323 rs12050284 (2380, 11)\n",
      "324 rs12050285 (2381, 11)\n",
      "325 rs12053233 (2393, 11)\n",
      "327 rs12060189 (2394, 11)\n",
      "330 rs12080583 (2398, 11)\n",
      "332 rs12096473 (2401, 11)\n",
      "333 rs12107103 (2407, 11)\n",
      "334 rs12107104 (2411, 11)\n",
      "336 rs12121412 (2412, 11)\n",
      "337 rs12123251 (2425, 11)\n",
      "338 rs12124014 (2436, 11)\n",
      "339 rs1213048 (2439, 11)\n",
      "340 rs12132780 (2450, 11)\n",
      "341 rs12134663 (2456, 11)\n",
      "342 rs12135232 (2462, 11)\n",
      "343 rs12146541 (2477, 11)\n",
      "345 rs12150214 (2489, 11)\n",
      "346 rs12150298 (2503, 11)\n",
      "347 rs12150603 (2515, 11)\n",
      "348 rs12153967 (2517, 11)\n",
      "350 rs12159200 (2521, 11)\n",
      "352 rs12193504 (2541, 11)\n",
      "355 rs12206964 (2542, 11)\n",
      "356 rs12207025 (2564, 11)\n",
      "358 rs12237867 (2579, 11)\n",
      "360 rs12264415 (2618, 11)\n",
      "362 rs12292340 (2636, 11)\n",
      "363 rs12292372 (2654, 11)\n",
      "364 rs1229967 (2656, 11)\n",
      "365 rs1230019 (2658, 11)\n",
      "366 rs1230025 (2660, 11)\n",
      "368 rs12337180 (2663, 11)\n",
      "369 rs12340544 (2669, 11)\n",
      "372 rs12405994 (2677, 11)\n",
      "373 rs12407660 (2680, 11)\n",
      "374 rs12410520 (2683, 11)\n",
      "376 rs12412727 (2684, 11)\n",
      "377 rs12414691 (2697, 11)\n",
      "378 rs12447826 (2727, 11)\n",
      "379 rs12450474 (2729, 11)\n",
      "380 rs12451122 (2757, 11)\n",
      "381 rs12453172 (2787, 11)\n",
      "382 rs12460764 (2806, 11)\n",
      "383 rs12465867 (2826, 11)\n",
      "384 rs12488303 (2879, 11)\n",
      "385 rs12489828 (2905, 11)\n",
      "387 rs12499239 (2906, 11)\n",
      "388 rs12529756 (2927, 11)\n",
      "390 rs12544187 (2929, 11)\n",
      "391 rs12546541 (2930, 11)\n",
      "392 rs1256044 (2933, 11)\n",
      "393 rs1256059 (2936, 11)\n",
      "394 rs1256065 (2939, 11)\n",
      "395 rs12599178 (2964, 11)\n",
      "396 rs12619296 (2966, 11)\n",
      "397 rs12625347 (2967, 11)\n",
      "398 rs1264325 (3025, 11)\n",
      "399 rs1264326 (3083, 11)\n",
      "400 rs12643956 (3088, 11)\n",
      "401 rs12645495 (3090, 11)\n",
      "402 rs12646523 (3092, 11)\n",
      "403 rs1264705 (3140, 11)\n",
      "404 rs1264706 (3188, 11)\n",
      "406 rs12659765 (3191, 11)\n",
      "407 rs12665974 (3194, 11)\n",
      "408 rs12668097 (3199, 11)\n",
      "409 rs12668156 (3205, 11)\n",
      "410 rs12678205 (3224, 11)\n",
      "411 rs1269056 (3227, 11)\n",
      "412 rs12699332 (3230, 11)\n",
      "416 rs12709094 (3249, 11)\n",
      "417 rs12713635 (3261, 11)\n",
      "419 rs12720 (3262, 11)\n",
      "422 rs12749612 (3273, 11)\n",
      "423 rs12758337 (3278, 11)\n",
      "424 rs12764048 (3291, 11)\n",
      "425 rs12822891 (3297, 11)\n",
      "426 rs12824504 (3299, 11)\n",
      "440 rs1289002 (3300, 11)\n",
      "441 rs12890577 (3301, 11)\n",
      "443 rs12899981 (3341, 11)\n",
      "444 rs12906125 (3343, 11)\n",
      "445 rs12913702 (3388, 11)\n",
      "447 rs12918956 (3391, 11)\n",
      "454 rs12940986 (3403, 11)\n",
      "456 rs12976696 (3405, 11)\n",
      "457 rs12986980 (3407, 11)\n",
      "459 rs13015451 (3436, 11)\n",
      "461 rs13165 (3438, 11)\n",
      "462 rs131804 (3452, 11)\n",
      "463 rs131808 (3465, 11)\n",
      "464 rs13195636 (3501, 11)\n",
      "465 rs13198809 (3539, 11)\n",
      "467 rs13199906 (3576, 11)\n",
      "468 rs13202688 (3591, 11)\n",
      "469 rs13206501 (3604, 11)\n",
      "470 rs13209332 (3640, 11)\n",
      "471 rs13210634 (3674, 11)\n",
      "472 rs13212318 (3711, 11)\n",
      "473 rs13212562 (3752, 11)\n",
      "474 rs13212651 (3788, 11)\n",
      "475 rs13219181 (3806, 11)\n",
      "476 rs13220817 (3826, 11)\n",
      "481 rs13294677 (3837, 11)\n",
      "482 rs13294710 (3854, 11)\n",
      "483 rs13300897 (3871, 11)\n",
      "484 rs13306278 (3872, 11)\n",
      "485 rs13306561 (3879, 11)\n",
      "486 rs13336260 (3907, 11)\n",
      "487 rs133369 (3989, 11)\n",
      "488 rs133375 (4075, 11)\n",
      "489 rs13342397 (4077, 11)\n",
      "490 rs1335507 (4085, 11)\n",
      "491 rs13379476 (4091, 11)\n",
      "492 rs13379477 (4095, 11)\n",
      "493 rs13379478 (4098, 11)\n",
      "494 rs13388224 (4099, 11)\n",
      "495 rs1339899 (4124, 11)\n",
      "496 rs1347551 (4133, 11)\n",
      "497 rs134887 (4198, 11)\n",
      "498 rs134888 (4262, 11)\n",
      "499 rs134889 (4326, 11)\n",
      "502 rs1370407 (4328, 11)\n",
      "505 rs138108 (4338, 11)\n",
      "507 rs13811 (4340, 11)\n",
      "511 rs1386768 (4344, 11)\n",
      "512 rs138843 (4357, 11)\n",
      "515 rs139302 (4368, 11)\n",
      "516 rs139357 (4376, 11)\n",
      "519 rs139784051 (4392, 11)\n",
      "522 rs1409839 (4393, 11)\n",
      "524 rs141298312 (4406, 11)\n",
      "525 rs14157 (4424, 11)\n",
      "527 rs142218582 (4431, 11)\n",
      "531 rs1435524 (4434, 11)\n",
      "532 rs1435525 (4437, 11)\n",
      "533 rs1435526 (4441, 11)\n",
      "534 rs1435527 (4444, 11)\n",
      "535 rs1438662 (4447, 11)\n",
      "536 rs1438663 (4448, 11)\n",
      "539 rs1445581 (4450, 11)\n",
      "540 rs145226413 (4451, 11)\n",
      "544 rs1460943 (4456, 11)\n",
      "550 rs147797700 (4457, 11)\n",
      "551 rs1480380 (4499, 11)\n",
      "557 rs148968669 (4500, 11)\n",
      "560 rs149299 (4588, 11)\n",
      "561 rs149393118 (4595, 11)\n",
      "562 rs1495100 (4610, 11)\n",
      "563 rs1495101 (4624, 11)\n",
      "564 rs1495102 (4638, 11)\n",
      "565 rs149843164 (4640, 11)\n",
      "566 rs149956 (4742, 11)\n",
      "569 rs152280 (4764, 11)\n",
      "570 rs1523919 (4765, 11)\n",
      "577 rs1536500 (4769, 11)\n",
      "578 rs1537514 (4777, 11)\n",
      "579 rs1537515 (4785, 11)\n",
      "580 rs1537516 (4793, 11)\n",
      "581 rs1546737 (4838, 11)\n",
      "582 rs1548884 (4841, 11)\n",
      "584 rs1550538 (4844, 11)\n",
      "585 rs1560344 (4848, 11)\n",
      "586 rs1560884 (4850, 11)\n",
      "587 rs156407 (4905, 11)\n",
      "588 rs1564078 (4906, 11)\n",
      "591 rs1565920 (4921, 11)\n",
      "592 rs1565922 (4935, 11)\n",
      "594 rs1570114 (4936, 11)\n",
      "595 rs15931 (4939, 11)\n",
      "596 rs159960 (4944, 11)\n",
      "599 rs1610640 (5012, 11)\n",
      "600 rs1610645 (5082, 11)\n",
      "601 rs1610718 (5150, 11)\n",
      "602 rs1611203 (5212, 11)\n",
      "603 rs1611204 (5281, 11)\n",
      "604 rs1611205 (5350, 11)\n",
      "605 rs1611206 (5418, 11)\n",
      "606 rs1611211 (5486, 11)\n",
      "607 rs1626899 (5503, 11)\n",
      "608 rs1646010 (5511, 11)\n",
      "610 rs16835708 (5514, 11)\n",
      "611 rs16835791 (5517, 11)\n",
      "613 rs16846026 (5518, 11)\n",
      "614 rs16853941 (5524, 11)\n",
      "615 rs16854783 (5526, 11)\n",
      "619 rs16887343 (5545, 11)\n",
      "620 rs16894116 (5596, 11)\n",
      "621 rs16895067 (5619, 11)\n",
      "622 rs16895070 (5642, 11)\n",
      "623 rs169158 (5661, 11)\n",
      "624 rs16957696 (5678, 11)\n",
      "625 rs16957831 (5689, 11)\n",
      "626 rs16973716 (5691, 11)\n",
      "633 rs17032850 (5694, 11)\n",
      "634 rs17037388 (5701, 11)\n",
      "635 rs170731 (5703, 11)\n",
      "638 rs1708059 (5704, 11)\n",
      "640 rs17097836 (5711, 11)\n",
      "641 rs17097837 (5719, 11)\n",
      "647 rs17158313 (5721, 11)\n",
      "648 rs1716184 (5735, 11)\n",
      "649 rs1716651 (5750, 11)\n",
      "650 rs17173608 (5756, 11)\n",
      "651 rs17184114 (5761, 11)\n",
      "652 rs17184255 (5785, 11)\n",
      "653 rs17188226 (5817, 11)\n",
      "655 rs17194627 (5849, 11)\n",
      "656 rs171983 (5860, 11)\n",
      "659 rs17215340 (5871, 11)\n",
      "661 rs17221829 (5888, 11)\n",
      "662 rs1724704 (5891, 11)\n",
      "663 rs17259177 (5893, 11)\n",
      "664 rs1727332 (5910, 11)\n",
      "666 rs17286349 (5920, 11)\n",
      "667 rs17300048 (5963, 11)\n",
      "668 rs1730040 (5967, 11)\n",
      "669 rs17301375 (5968, 11)\n",
      "670 rs17367289 (5973, 11)\n",
      "671 rs17367504 (5979, 11)\n",
      "672 rs17367629 (5986, 11)\n",
      "673 rs1736891 (6093, 11)\n",
      "674 rs1737006 (6161, 11)\n",
      "677 rs174528 (6175, 11)\n",
      "678 rs174529 (6189, 11)\n",
      "679 rs174530 (6205, 11)\n",
      "680 rs174533 (6225, 11)\n",
      "681 rs174534 (6235, 11)\n",
      "682 rs174535 (6253, 11)\n",
      "683 rs174536 (6273, 11)\n",
      "684 rs174537 (6292, 11)\n",
      "685 rs174538 (6309, 11)\n",
      "686 rs174541 (6330, 11)\n",
      "687 rs174544 (6350, 11)\n",
      "688 rs174546 (6370, 11)\n",
      "689 rs174561 (6390, 11)\n",
      "690 rs174562 (6410, 11)\n",
      "691 rs174568 (6429, 11)\n",
      "692 rs174570 (6433, 11)\n",
      "693 rs174574 (6450, 11)\n",
      "694 rs174576 (6468, 11)\n",
      "695 rs174577 (6485, 11)\n",
      "696 rs174578 (6502, 11)\n",
      "697 rs174580 (6518, 11)\n",
      "698 rs174583 (6536, 11)\n",
      "699 rs174599 (6552, 11)\n",
      "700 rs174600 (6564, 11)\n",
      "701 rs17479770 (6565, 11)\n",
      "702 rs17546378 (6570, 11)\n",
      "703 rs17601029 (6611, 11)\n",
      "704 rs17617406 (6626, 11)\n",
      "705 rs176185 (6631, 11)\n",
      "706 rs176249 (6633, 11)\n",
      "707 rs17628118 (6635, 11)\n",
      "708 rs17628613 (6639, 11)\n",
      "710 rs17657783 (6640, 11)\n",
      "711 rs176644 (6642, 11)\n",
      "712 rs1768144 (6643, 11)\n",
      "713 rs1768237 (6644, 11)\n",
      "715 rs17750747 (6681, 11)\n",
      "716 rs17751184 (6718, 11)\n",
      "717 rs17763089 (6755, 11)\n",
      "718 rs17776473 (6756, 11)\n",
      "720 rs17835937 (6762, 11)\n",
      "721 rs1786171 (6767, 11)\n",
      "722 rs17866752 (6768, 11)\n",
      "723 rs1787984 (6769, 11)\n",
      "724 rs1805645 (6770, 11)\n",
      "727 rs1820353 (6787, 11)\n",
      "729 rs182187683 (6788, 11)\n",
      "730 rs1834451 (6791, 11)\n",
      "732 rs184054 (6793, 11)\n",
      "736 rs1846416 (6795, 11)\n",
      "739 rs186307979 (6796, 11)\n",
      "741 rs1866881 (6799, 11)\n",
      "742 rs1872924 (6811, 11)\n",
      "743 rs1874228 (6822, 11)\n",
      "744 rs1877031 (6835, 11)\n",
      "745 rs1878579 (6836, 11)\n",
      "747 rs1883324 (6837, 11)\n",
      "749 rs1884066 (6842, 11)\n",
      "750 rs1884352 (6845, 11)\n",
      "751 rs188624478 (6848, 11)\n",
      "752 rs1889292 (6859, 11)\n",
      "753 rs1889293 (6868, 11)\n",
      "755 rs1890185 (6907, 11)\n",
      "757 rs1894399 (6908, 11)\n",
      "758 rs190289 (6913, 11)\n",
      "762 rs1930048 (6914, 11)\n",
      "764 rs193423 (6915, 11)\n",
      "765 rs1935949 (6916, 11)\n",
      "766 rs193779 (6918, 11)\n",
      "768 rs1947025 (6930, 11)\n",
      "769 rs1955606 (6935, 11)\n",
      "770 rs1955608 (6938, 11)\n",
      "771 rs1955609 (6941, 11)\n",
      "772 rs1971546 (6964, 11)\n",
      "773 rs1972907 (6966, 11)\n",
      "774 rs1978550 (6972, 11)\n",
      "775 rs1981960 (6980, 11)\n",
      "776 rs1981961 (6988, 11)\n",
      "777 rs198357 (7007, 11)\n",
      "778 rs198394 (7018, 11)\n",
      "779 rs198396 (7027, 11)\n",
      "780 rs198397 (7042, 11)\n",
      "781 rs198399 (7057, 11)\n",
      "782 rs198402 (7076, 11)\n",
      "783 rs198404 (7087, 11)\n",
      "784 rs198407 (7098, 11)\n",
      "785 rs198408 (7112, 11)\n",
      "786 rs198409 (7131, 11)\n",
      "787 rs1985437 (7155, 11)\n",
      "788 rs1986972 (7180, 11)\n",
      "789 rs1989867 (7200, 11)\n",
      "790 rs1990620 (7203, 11)\n",
      "791 rs1990621 (7206, 11)\n",
      "792 rs199514 (7325, 11)\n",
      "796 rs200483 (7366, 11)\n",
      "797 rs200497 (7409, 11)\n",
      "801 rs200948 (7453, 11)\n",
      "802 rs200949 (7494, 11)\n",
      "806 rs2019065 (7524, 11)\n",
      "808 rs2021722 (7573, 11)\n",
      "809 rs202665 (7608, 11)\n",
      "810 rs2027130 (7614, 11)\n",
      "811 rs2030151 (7616, 11)\n",
      "814 rs2037814 (7619, 11)\n",
      "815 rs2039531 (7623, 11)\n",
      "816 rs2044693 (7635, 11)\n",
      "817 rs2053345 (7639, 11)\n",
      "818 rs2056193 (7640, 11)\n",
      "819 rs2059922 (7642, 11)\n",
      "821 rs2066470 (7648, 11)\n",
      "822 rs2067007 (7649, 11)\n",
      "823 rs206786 (7651, 11)\n",
      "824 rs2071041 (7701, 11)\n",
      "825 rs2071278 (7754, 11)\n",
      "826 rs2071507 (7794, 11)\n",
      "827 rs2077566 (7797, 11)\n",
      "828 rs2079340 (7802, 11)\n",
      "830 rs2087563 (7812, 11)\n",
      "832 rs2092563 (7828, 11)\n",
      "833 rs209473 (7833, 11)\n",
      "834 rs2103665 (7834, 11)\n",
      "835 rs211448 (7843, 11)\n",
      "839 rs213239 (7873, 11)\n",
      "840 rs2138862 (7874, 11)\n",
      "841 rs2142189 (7876, 11)\n",
      "842 rs2144061 (7881, 11)\n",
      "843 rs214469 (7883, 11)\n",
      "844 rs2144909 (7889, 11)\n",
      "845 rs2145615 (7890, 11)\n",
      "846 rs2148582 (7896, 11)\n",
      "847 rs2151654 (7904, 11)\n",
      "848 rs2151655 (7913, 11)\n",
      "852 rs2167079 (7936, 11)\n",
      "853 rs2185502 (7937, 11)\n",
      "856 rs2213856 (7939, 11)\n",
      "857 rs2213857 (7941, 11)\n",
      "858 rs221789 (7960, 11)\n",
      "859 rs221803 (7978, 11)\n",
      "860 rs221922 (7987, 11)\n",
      "861 rs2223937 (7993, 11)\n",
      "862 rs223323 (8010, 11)\n",
      "863 rs223332 (8027, 11)\n",
      "864 rs223345 (8047, 11)\n",
      "865 rs223346 (8065, 11)\n",
      "866 rs2233474 (8081, 11)\n",
      "867 rs223385 (8102, 11)\n",
      "868 rs223462 (8119, 11)\n",
      "869 rs223463 (8136, 11)\n",
      "870 rs223464 (8153, 11)\n",
      "871 rs2237228 (8166, 11)\n",
      "872 rs2237326 (8169, 11)\n",
      "876 rs2239038 (8171, 11)\n",
      "877 rs2240694 (8239, 11)\n",
      "878 rs2240695 (8305, 11)\n",
      "879 rs2241353 (8307, 11)\n",
      "881 rs2242625 (8309, 11)\n",
      "883 rs2247862 (8325, 11)\n",
      "884 rs225092 (8330, 11)\n",
      "885 rs225095 (8336, 11)\n",
      "886 rs225100 (8342, 11)\n",
      "887 rs225106 (8348, 11)\n",
      "888 rs225109 (8354, 11)\n",
      "889 rs225110 (8360, 11)\n",
      "890 rs225131 (8363, 11)\n",
      "891 rs225132 (8366, 11)\n",
      "892 rs2252259 (8372, 11)\n",
      "894 rs2255632 (8377, 11)\n",
      "895 rs226242 (8382, 11)\n",
      "896 rs226249 (8387, 11)\n",
      "897 rs226251 (8392, 11)\n",
      "899 rs2270374 (8397, 11)\n",
      "900 rs2270376 (8415, 11)\n",
      "901 rs2271308 (8429, 11)\n",
      "902 rs2272730 (8439, 11)\n",
      "903 rs2272991 (8462, 11)\n",
      "904 rs2274267 (8483, 11)\n",
      "905 rs2274730 (8487, 11)\n",
      "906 rs2274976 (8491, 11)\n",
      "907 rs2275271 (8532, 11)\n",
      "908 rs2276834 (8559, 11)\n",
      "909 rs2280375 (8561, 11)\n",
      "910 rs2280397 (8562, 11)\n",
      "911 rs2280881 (8569, 11)\n",
      "916 rs2285910 (8585, 11)\n",
      "917 rs2288921 (8597, 11)\n",
      "918 rs2290238 (8599, 11)\n",
      "921 rs2292974 (8603, 11)\n",
      "922 rs2292975 (8607, 11)\n",
      "924 rs2294890 (8612, 11)\n",
      "925 rs2295406 (8630, 11)\n",
      "926 rs2295408 (8647, 11)\n",
      "927 rs2295409 (8664, 11)\n",
      "928 rs2296568 (8680, 11)\n",
      "930 rs2297786 (8719, 11)\n",
      "931 rs2299309 (8723, 11)\n",
      "933 rs2301669 (8742, 11)\n",
      "934 rs2301814 (8762, 11)\n",
      "935 rs2302704 (8768, 11)\n",
      "936 rs2302705 (8774, 11)\n",
      "937 rs2303384 (8797, 11)\n",
      "938 rs2304631 (8835, 11)\n",
      "939 rs230492 (8838, 11)\n",
      "940 rs230495 (8841, 11)\n",
      "941 rs230504 (8844, 11)\n",
      "942 rs230505 (8847, 11)\n",
      "944 rs230534 (8850, 11)\n",
      "945 rs2313171 (8865, 11)\n",
      "946 rs2319462 (8883, 11)\n",
      "948 rs2342558 (8887, 11)\n",
      "949 rs2343302 (8900, 11)\n",
      "950 rs2353436 (8902, 11)\n",
      "951 rs2361422 (8905, 11)\n",
      "952 rs2362965 (8915, 11)\n",
      "953 rs2362972 (8918, 11)\n",
      "954 rs237900 (8923, 11)\n",
      "955 rs2389463 (8925, 11)\n",
      "957 rs2402960 (8926, 11)\n",
      "960 rs2426109 (8932, 11)\n",
      "961 rs2428682 (8933, 11)\n",
      "963 rs2442722 (8972, 11)\n",
      "964 rs2448685 (8976, 11)\n",
      "965 rs2460898 (8977, 11)\n",
      "966 rs2460906 (8978, 11)\n",
      "967 rs2466255 (8979, 11)\n",
      "968 rs2466256 (8980, 11)\n",
      "969 rs24772 (8981, 11)\n",
      "970 rs2478539 (8987, 11)\n",
      "971 rs2478543 (8993, 11)\n",
      "972 rs2486065 (9000, 11)\n",
      "973 rs2493135 (9006, 11)\n",
      "974 rs2494598 (9010, 11)\n",
      "975 rs2494730 (9013, 11)\n",
      "976 rs2494733 (9017, 11)\n",
      "977 rs2498796 (9020, 11)\n",
      "978 rs2498797 (9023, 11)\n",
      "979 rs2498803 (9028, 11)\n",
      "982 rs2517578 (9087, 11)\n",
      "983 rs2517953 (9098, 11)\n",
      "984 rs2517954 (9111, 11)\n",
      "985 rs2517955 (9122, 11)\n",
      "986 rs2517956 (9135, 11)\n",
      "987 rs2517957 (9147, 11)\n",
      "988 rs2517958 (9159, 11)\n",
      "989 rs2526882 (9169, 11)\n",
      "991 rs2535334 (9228, 11)\n",
      "992 rs2563283 (9293, 11)\n",
      "993 rs2563335 (9362, 11)\n",
      "997 rs2565061 (9364, 11)\n",
      "998 rs2568960 (9369, 11)\n",
      "999 rs2592120 (9370, 11)\n",
      "1000 rs2592195 (9372, 11)\n",
      "1001 rs2596505 (9432, 11)\n",
      "1003 rs2638550 (9433, 11)\n",
      "1004 rs2646011 (9435, 11)\n",
      "1008 rs2675959 (9443, 11)\n",
      "1009 rs2675964 (9444, 11)\n",
      "1010 rs2675965 (9445, 11)\n",
      "1011 rs2678113 (9446, 11)\n",
      "1012 rs2678119 (9447, 11)\n",
      "1013 rs2682406 (9450, 11)\n",
      "1014 rs2693542 (9453, 11)\n",
      "1017 rs272827 (9454, 11)\n",
      "1018 rs272833 (9455, 11)\n",
      "1019 rs2731005 (9456, 11)\n",
      "1020 rs2731006 (9457, 11)\n",
      "1025 rs2747054 (9498, 11)\n",
      "1026 rs2764261 (9499, 11)\n",
      "1027 rs2765453 (9500, 11)\n",
      "1028 rs2799077 (9568, 11)\n",
      "1029 rs2799079 (9636, 11)\n",
      "1030 rs28157 (9648, 11)\n",
      "1031 rs2815752 (9653, 11)\n",
      "1032 rs2815753 (9658, 11)\n",
      "1033 rs281791 (9674, 11)\n",
      "1037 rs28360499 (9715, 11)\n",
      "1038 rs28362669 (9733, 11)\n",
      "1039 rs28365850 (9742, 11)\n",
      "1040 rs28398007 (9745, 11)\n",
      "1041 rs28428768 (9766, 11)\n",
      "1042 rs28434312 (9768, 11)\n",
      "1046 rs2845845 (9769, 11)\n",
      "1047 rs28473763 (9787, 11)\n",
      "1048 rs284853 (9790, 11)\n",
      "1050 rs28498032 (9796, 11)\n",
      "1051 rs28500170 (9797, 11)\n",
      "1052 rs2850599 (9817, 11)\n",
      "1053 rs28528780 (9818, 11)\n",
      "1054 rs2854002 (9883, 11)\n",
      "1056 rs2854827 (9953, 11)\n",
      "1057 rs2856650 (9979, 11)\n",
      "1058 rs2856837 (9981, 11)\n",
      "1060 rs28636952 (9983, 11)\n",
      "1061 rs28676999 (9995, 11)\n",
      "1062 rs28681082 (10015, 11)\n",
      "1063 rs2869492 (10035, 11)\n",
      "1064 rs28712119 (10044, 11)\n",
      "1065 rs28728306 (10047, 11)\n",
      "1067 rs28732976 (10052, 11)\n",
      "1069 rs2877501 (10054, 11)\n",
      "1070 rs28817965 (10060, 11)\n",
      "1073 rs28869534 (10061, 11)\n",
      "1076 rs2905426 (10082, 11)\n",
      "1077 rs2905427 (10103, 11)\n",
      "1078 rs2907 (10104, 11)\n",
      "1079 rs2912062 (10113, 11)\n",
      "1080 rs2934951 (10128, 11)\n",
      "1081 rs2934952 (10142, 11)\n",
      "1082 rs2934953 (10156, 11)\n",
      "1083 rs2934956 (10171, 11)\n",
      "1084 rs2941503 (10187, 11)\n",
      "1085 rs2941504 (10202, 11)\n",
      "1086 rs2941505 (10217, 11)\n",
      "1087 rs2941506 (10232, 11)\n",
      "1088 rs2952152 (10247, 11)\n",
      "1089 rs2955357 (10263, 11)\n",
      "1090 rs2955358 (10277, 11)\n",
      "1091 rs2955385 (10290, 11)\n",
      "1092 rs2961272 (10291, 11)\n",
      "1093 rs2961300 (10292, 11)\n",
      "1094 rs2965180 (10314, 11)\n",
      "1095 rs2965186 (10335, 11)\n",
      "1096 rs2965191 (10354, 11)\n",
      "1097 rs2965201 (10375, 11)\n",
      "1103 rs2980655 (10385, 11)\n",
      "1104 rs300716 (10387, 11)\n",
      "1105 rs3010090 (10390, 11)\n",
      "1106 rs301802 (10399, 11)\n",
      "1107 rs301805 (10408, 11)\n",
      "1108 rs301806 (10416, 11)\n",
      "1109 rs301807 (10424, 11)\n",
      "1110 rs301814 (10428, 11)\n",
      "1111 rs3020023 (10431, 11)\n",
      "1112 rs3020029 (10433, 11)\n",
      "1113 rs3020263 (10442, 11)\n",
      "1114 rs3020266 (10452, 11)\n",
      "1117 rs3091427 (10456, 11)\n",
      "1118 rs3092374 (10462, 11)\n",
      "1119 rs3094064 (10525, 11)\n",
      "1120 rs3094614 (10568, 11)\n",
      "1121 rs3096689 (10626, 11)\n",
      "1122 rs3096696 (10689, 11)\n",
      "1123 rs3097385 (10691, 11)\n",
      "1124 rs3097411 (10692, 11)\n",
      "1125 rs3120747 (10695, 11)\n",
      "1126 rs312483 (10699, 11)\n",
      "1127 rs3130347 (10758, 11)\n",
      "1128 rs3130380 (10822, 11)\n",
      "1129 rs3130825 (10850, 11)\n",
      "1130 rs3131044 (10917, 11)\n",
      "1131 rs3131300 (10979, 11)\n",
      "1132 rs3132668 (11021, 11)\n",
      "1133 rs3132672 (11062, 11)\n",
      "1134 rs3132674 (11104, 11)\n",
      "1135 rs3133155 (11107, 11)\n",
      "1136 rs3134605 (11156, 11)\n",
      "1137 rs3134940 (11218, 11)\n",
      "1138 rs3134944 (11276, 11)\n",
      "1139 rs3134945 (11334, 11)\n",
      "1140 rs3134946 (11392, 11)\n",
      "1141 rs3134947 (11450, 11)\n",
      "1142 rs316398 (11453, 11)\n",
      "1143 rs316409 (11454, 11)\n",
      "1144 rs319237 (11455, 11)\n",
      "1148 rs3213849 (11457, 11)\n",
      "1152 rs33997182 (11464, 11)\n",
      "1154 rs34107459 (11483, 11)\n",
      "1156 rs34243448 (11523, 11)\n",
      "1157 rs34268041 (11536, 11)\n",
      "1159 rs34294776 (11539, 11)\n",
      "1160 rs34330 (11540, 11)\n",
      "1161 rs34396849 (11577, 11)\n",
      "1164 rs34481365 (11589, 11)\n",
      "1166 rs34569561 (11634, 11)\n",
      "1167 rs34706883 (11671, 11)\n",
      "1169 rs34752946 (11673, 11)\n",
      "1171 rs34785014 (11677, 11)\n",
      "1172 rs34896432 (11697, 11)\n",
      "1174 rs34959638 (11715, 11)\n",
      "1175 rs34976037 (11716, 11)\n",
      "1177 rs35037868 (11753, 11)\n",
      "1178 rs35191208 (11758, 11)\n",
      "1179 rs352045 (11759, 11)\n",
      "1180 rs35236974 (11763, 11)\n",
      "1184 rs35324223 (11768, 11)\n",
      "1185 rs354042 (11769, 11)\n",
      "1187 rs35432211 (11788, 11)\n",
      "1190 rs35525740 (11830, 11)\n",
      "1191 rs35541316 (11831, 11)\n",
      "1192 rs35565446 (11868, 11)\n",
      "1193 rs35593468 (11873, 11)\n",
      "1195 rs35647154 (11897, 11)\n",
      "1196 rs35671072 (11912, 11)\n",
      "1198 rs35686634 (11913, 11)\n",
      "1199 rs35735140 (11925, 11)\n",
      "1200 rs35744819 (11963, 11)\n",
      "1201 rs35808061 (11978, 11)\n",
      "1203 rs35848276 (12014, 11)\n",
      "1204 rs35866672 (12016, 11)\n",
      "1205 rs35885918 (12036, 11)\n",
      "1206 rs36014129 (12052, 11)\n",
      "1208 rs36039295 (12053, 11)\n",
      "1209 rs36061954 (12055, 11)\n",
      "1210 rs36078 (12056, 11)\n",
      "1211 rs36118153 (12057, 11)\n",
      "1212 rs361227 (12063, 11)\n",
      "1213 rs370155 (12104, 11)\n",
      "1214 rs370812 (12110, 11)\n",
      "1216 rs3729802 (12129, 11)\n",
      "1217 rs3729936 (12131, 11)\n",
      "1218 rs3729948 (12133, 11)\n",
      "1219 rs3729953 (12135, 11)\n",
      "1220 rs3729986 (12138, 11)\n",
      "1221 rs3729989 (12144, 11)\n",
      "1223 rs3735993 (12154, 11)\n",
      "1224 rs3737965 (12158, 11)\n",
      "1225 rs3737966 (12167, 11)\n",
      "1226 rs3739121 (12172, 11)\n",
      "1227 rs3739547 (12183, 11)\n",
      "1228 rs3739548 (12194, 11)\n",
      "1229 rs3739550 (12196, 11)\n",
      "1230 rs3741194 (12219, 11)\n",
      "1231 rs3741619 (12224, 11)\n",
      "1232 rs3743735 (12241, 11)\n",
      "1233 rs3743741 (12272, 11)\n",
      "1234 rs3745937 (12285, 11)\n",
      "1235 rs3746665 (12296, 11)\n",
      "1236 rs3746666 (12307, 11)\n",
      "1237 rs3748376 (12348, 11)\n",
      "1238 rs3751903 (12360, 11)\n",
      "1240 rs3753582 (12367, 11)\n",
      "1241 rs3753584 (12375, 11)\n",
      "1242 rs3754080 (12376, 11)\n",
      "1243 rs3756334 (12439, 11)\n",
      "1244 rs3756342 (12507, 11)\n",
      "1245 rs3756599 (12510, 11)\n",
      "1246 rs3758335 (12527, 11)\n",
      "1247 rs3761853 (12528, 11)\n",
      "1248 rs3764353 (12539, 11)\n",
      "1250 rs3767474 (12540, 11)\n",
      "1252 rs3769483 (12549, 11)\n",
      "1253 rs37711 (12551, 11)\n",
      "1254 rs37717 (12553, 11)\n",
      "1255 rs3774933 (12555, 11)\n",
      "1256 rs3780447 (12560, 11)\n",
      "1257 rs3784074 (12566, 11)\n",
      "1258 rs3784075 (12570, 11)\n",
      "1259 rs3785109 (12587, 11)\n",
      "1260 rs3785113 (12598, 11)\n",
      "1261 rs3785127 (12609, 11)\n",
      "1262 rs3791046 (12610, 11)\n",
      "1263 rs3794809 (12639, 11)\n",
      "1264 rs3797040 (12641, 11)\n",
      "1265 rs3800231 (12642, 11)\n",
      "1266 rs3800316 (12653, 11)\n",
      "1267 rs3800318 (12670, 11)\n",
      "1269 rs38031 (12688, 11)\n",
      "1270 rs3803763 (12693, 11)\n",
      "1271 rs38041 (12713, 11)\n",
      "1273 rs3806825 (12715, 11)\n",
      "1274 rs3807865 (12718, 11)\n",
      "1275 rs3809449 (12727, 11)\n",
      "1276 rs3809594 (12728, 11)\n",
      "1278 rs3814546 (12745, 11)\n",
      "1279 rs3814547 (12765, 11)\n",
      "1280 rs3814548 (12782, 11)\n",
      "1283 rs3816334 (12797, 11)\n",
      "1284 rs3817160 (12808, 11)\n",
      "1285 rs3818003 (12823, 11)\n",
      "1286 rs3818085 (12829, 11)\n",
      "1288 rs3818717 (12840, 11)\n",
      "1289 rs3820192 (12848, 11)\n",
      "1290 rs3822346 (12911, 11)\n",
      "1291 rs3824288 (12928, 11)\n",
      "1292 rs3824756 (12931, 11)\n",
      "1293 rs3825697 (12934, 11)\n",
      "1295 rs3842748 (12935, 11)\n",
      "1296 rs3842753 (12936, 11)\n",
      "1297 rs3845486 (12938, 11)\n",
      "1298 rs3851005 (12957, 11)\n",
      "1299 rs3852788 (12959, 11)\n",
      "1302 rs3901849 (12976, 11)\n",
      "1303 rs3904682 (12986, 11)\n",
      "1305 rs3924855 (13018, 11)\n",
      "1306 rs39315 (13020, 11)\n",
      "1307 rs393329 (13041, 11)\n",
      "1309 rs39500 (13053, 11)\n",
      "1314 rs408320 (13059, 11)\n",
      "1315 rs411136 (13067, 11)\n",
      "1317 rs41266779 (13084, 11)\n",
      "1318 rs41270349 (13085, 11)\n",
      "1319 rs41271527 (13086, 11)\n",
      "1320 rs41271951 (13100, 11)\n",
      "1321 rs41275460 (13108, 11)\n",
      "1322 rs41275462 (13115, 11)\n",
      "1323 rs41275464 (13121, 11)\n",
      "1324 rs41299934 (13122, 11)\n",
      "1325 rs4140512 (13127, 11)\n",
      "1326 rs4141702 (13133, 11)\n",
      "1327 rs4141841 (13199, 11)\n",
      "1329 rs4147536 (13201, 11)\n",
      "1332 rs4151680 (13264, 11)\n",
      "1334 rs4236274 (13265, 11)\n",
      "1337 rs4239639 (13285, 11)\n",
      "1341 rs4244354 (13290, 11)\n",
      "1346 rs42769 (13310, 11)\n",
      "1347 rs4291 (13318, 11)\n",
      "1349 rs4325622 (13345, 11)\n",
      "1350 rs433061 (13447, 11)\n",
      "1352 rs4365348 (13461, 11)\n",
      "1353 rs4384978 (13464, 11)\n",
      "1354 rs4395073 (13471, 11)\n",
      "1355 rs439558 (13692, 11)\n",
      "1356 rs440119 (13694, 11)\n",
      "1357 rs4402589 (13726, 11)\n",
      "1358 rs4427857 (13751, 11)\n",
      "1359 rs442862 (13757, 11)\n",
      "1360 rs4459619 (13758, 11)\n",
      "1362 rs4496901 (13761, 11)\n",
      "1363 rs4511180 (13763, 11)\n",
      "1364 rs4547554 (13787, 11)\n",
      "1366 rs45504202 (13793, 11)\n",
      "1367 rs45509595 (13830, 11)\n",
      "1368 rs45608437 (13836, 11)\n",
      "1369 rs4607924 (13844, 11)\n",
      "1370 rs461964 (13849, 11)\n",
      "1371 rs4644136 (13851, 11)\n",
      "1374 rs4649957 (13853, 11)\n",
      "1375 rs46522 (13881, 11)\n",
      "1376 rs4653160 (13883, 11)\n",
      "1377 rs465474 (13887, 11)\n",
      "1378 rs465506 (13892, 11)\n",
      "1379 rs4666012 (13894, 11)\n",
      "1380 rs4670209 (13896, 11)\n",
      "1381 rs4679827 (13902, 11)\n",
      "1382 rs4692783 (13907, 11)\n",
      "1383 rs4692784 (13912, 11)\n",
      "1384 rs4692785 (13917, 11)\n",
      "1385 rs4698874 (13934, 11)\n",
      "1386 rs4711107 (13952, 11)\n",
      "1387 rs4711154 (13973, 11)\n",
      "1388 rs4711155 (13995, 11)\n",
      "1389 rs4711429 (13999, 11)\n",
      "1390 rs4713919 (14004, 11)\n",
      "1393 rs4727621 (14008, 11)\n",
      "1395 rs4733768 (14014, 11)\n",
      "1396 rs4736342 (14023, 11)\n",
      "1401 rs4766480 (14028, 11)\n",
      "1402 rs4784051 (14036, 11)\n",
      "1403 rs4785741 (14042, 11)\n",
      "1404 rs4785962 (14074, 11)\n",
      "1405 rs4785964 (14111, 11)\n",
      "1407 rs4794817 (14123, 11)\n",
      "1408 rs4795390 (14136, 11)\n",
      "1409 rs4806711 (14138, 11)\n",
      "1411 rs4820433 (14153, 11)\n",
      "1412 rs4846040 (14162, 11)\n",
      "1413 rs4846041 (14171, 11)\n",
      "1414 rs4846043 (14182, 11)\n",
      "1415 rs4846047 (14192, 11)\n",
      "1416 rs4846049 (14203, 11)\n",
      "1417 rs4846056 (14212, 11)\n",
      "1418 rs4847378 (14218, 11)\n",
      "1419 rs4848303 (14220, 11)\n",
      "1422 rs4861021 (14226, 11)\n",
      "1424 rs4897288 (14227, 11)\n",
      "1430 rs4908760 (14231, 11)\n",
      "1431 rs4919666 (14233, 11)\n",
      "1432 rs4919669 (14240, 11)\n",
      "1433 rs4924832 (14254, 11)\n",
      "1434 rs4934646 (14256, 11)\n",
      "1436 rs4950882 (14263, 11)\n",
      "1437 rs4950928 (14269, 11)\n",
      "1438 rs4950931 (14276, 11)\n",
      "1439 rs4968361 (14277, 11)\n",
      "1442 rs499188 (14288, 11)\n",
      "1443 rs50034 (14305, 11)\n",
      "1444 rs5007118 (14309, 11)\n",
      "1445 rs5011437 (14312, 11)\n",
      "1446 rs5011438 (14315, 11)\n",
      "1447 rs5011439 (14317, 11)\n",
      "1448 rs5019570 (14320, 11)\n",
      "1449 rs5019571 (14321, 11)\n",
      "1450 rs5051 (14327, 11)\n",
      "1451 rs508977 (14328, 11)\n",
      "1452 rs509360 (14332, 11)\n",
      "1453 rs509817 (14333, 11)\n",
      "1454 rs518632 (14334, 11)\n",
      "1455 rs519565 (14335, 11)\n",
      "1457 rs530614 (14340, 11)\n",
      "1461 rs541923 (14341, 11)\n",
      "1462 rs55634431 (14347, 11)\n",
      "1463 rs55661783 (14348, 11)\n",
      "1464 rs55668272 (14353, 11)\n",
      "1465 rs55675218 (14364, 11)\n",
      "1466 rs55689128 (14401, 11)\n",
      "1467 rs55703821 (14420, 11)\n",
      "1470 rs55751606 (14442, 11)\n",
      "1471 rs55752627 (14443, 11)\n",
      "1474 rs55848773 (14444, 11)\n",
      "1475 rs55849753 (14445, 11)\n",
      "1476 rs55867221 (14453, 11)\n",
      "1477 rs55885412 (14456, 11)\n",
      "1478 rs55903932 (14459, 11)\n",
      "1479 rs55929441 (14467, 11)\n",
      "1480 rs55957413 (14499, 11)\n",
      "1481 rs55962869 (14517, 11)\n",
      "1482 rs55971843 (14518, 11)\n",
      "1483 rs55990971 (14522, 11)\n",
      "1486 rs56071379 (14535, 11)\n",
      "1487 rs56083896 (14536, 11)\n",
      "1488 rs56092809 (14537, 11)\n",
      "1489 rs56108505 (14538, 11)\n",
      "1490 rs56160221 (14539, 11)\n",
      "1492 rs56169202 (14540, 11)\n",
      "1493 rs56205728 (14553, 11)\n",
      "1496 rs56259105 (14554, 11)\n",
      "1498 rs56303487 (14577, 11)\n",
      "1499 rs56324992 (14587, 11)\n",
      "1500 rs56331443 (14588, 11)\n",
      "1502 rs56405707 (14622, 11)\n",
      "1503 rs565876 (14624, 11)\n",
      "1504 rs56824938 (14654, 11)\n",
      "1506 rs56894701 (14658, 11)\n",
      "1507 rs570546 (14660, 11)\n",
      "1509 rs57178676 (14663, 11)\n",
      "1510 rs57249955 (14672, 11)\n",
      "1511 rs57310340 (14676, 11)\n",
      "1512 rs57368576 (14684, 11)\n",
      "1513 rs573878 (14685, 11)\n",
      "1514 rs57433322 (14699, 11)\n",
      "1515 rs57434514 (14709, 11)\n",
      "1516 rs5751069 (14724, 11)\n",
      "1518 rs5758264 (14739, 11)\n",
      "1519 rs5758266 (14754, 11)\n",
      "1520 rs5758566 (14841, 11)\n",
      "1521 rs5770728 (14858, 11)\n",
      "1522 rs5770750 (14876, 11)\n",
      "1523 rs57728924 (14888, 11)\n",
      "1524 rs57838764 (14900, 11)\n",
      "1525 rs580016 (14907, 11)\n",
      "1527 rs5816505 (14939, 11)\n",
      "1528 rs5827177 (14940, 11)\n",
      "1529 rs58367317 (14945, 11)\n",
      "1531 rs585283 (14946, 11)\n",
      "1532 rs5853820 (14949, 11)\n",
      "1534 rs5875151 (15054, 11)\n",
      "1536 rs5896 (15068, 11)\n",
      "1537 rs59082627 (15069, 11)\n",
      "1539 rs59173939 (15074, 11)\n",
      "1540 rs591917 (15075, 11)\n",
      "1541 rs59304093 (15087, 11)\n",
      "1542 rs593641 (15088, 11)\n",
      "1543 rs59375726 (15094, 11)\n",
      "1544 rs59498392 (15114, 11)\n",
      "1545 rs59576053 (15153, 11)\n",
      "1546 rs598373 (15170, 11)\n",
      "1547 rs59856565 (15171, 11)\n",
      "1548 rs598946 (15180, 11)\n",
      "1550 rs5996094 (15235, 11)\n",
      "1551 rs6001974 (15237, 11)\n",
      "1552 rs6002000 (15243, 11)\n",
      "1553 rs6002548 (15305, 11)\n",
      "1554 rs6002555 (15365, 11)\n",
      "1555 rs601338 (15388, 11)\n",
      "1556 rs602885 (15389, 11)\n",
      "1558 rs603542 (15392, 11)\n",
      "1559 rs6037708 (15400, 11)\n",
      "1560 rs60459288 (15403, 11)\n",
      "1561 rs604980 (15405, 11)\n",
      "1562 rs6052209 (15414, 11)\n",
      "1563 rs6052217 (15423, 11)\n",
      "1564 rs6052218 (15432, 11)\n",
      "1565 rs6052219 (15441, 11)\n",
      "1566 rs6052222 (15450, 11)\n",
      "1568 rs60550390 (15456, 11)\n",
      "1569 rs60558877 (15474, 11)\n",
      "1573 rs610060 (15502, 11)\n",
      "1574 rs61014003 (15529, 11)\n",
      "1575 rs6107386 (15540, 11)\n",
      "1576 rs6107393 (15551, 11)\n",
      "1577 rs61240589 (15578, 11)\n",
      "1580 rs61381693 (15579, 11)\n",
      "1581 rs61432137 (15580, 11)\n",
      "1582 rs61519741 (15598, 11)\n",
      "1584 rs61747867 (15602, 11)\n",
      "1585 rs61773952 (15607, 11)\n",
      "1586 rs61773953 (15612, 11)\n",
      "1587 rs61776071 (15618, 11)\n",
      "1596 rs61846516 (15619, 11)\n",
      "1597 rs61854284 (15620, 11)\n",
      "1599 rs61882674 (15626, 11)\n",
      "1600 rs61884264 (15632, 11)\n",
      "1602 rs62032510 (15637, 11)\n",
      "1603 rs62070344 (15664, 11)\n",
      "1604 rs62100765 (15665, 11)\n",
      "1605 rs62144168 (15679, 11)\n",
      "1606 rs62150473 (15680, 11)\n",
      "1608 rs62238588 (15699, 11)\n",
      "1612 rs62305755 (15701, 11)\n",
      "1615 rs62402024 (15721, 11)\n",
      "1616 rs62402025 (15741, 11)\n",
      "1617 rs62505473 (15743, 11)\n",
      "1619 rs62574275 (15747, 11)\n",
      "1621 rs630303 (15780, 11)\n",
      "1622 rs630394 (15783, 11)\n",
      "1623 rs633800 (15807, 11)\n",
      "1624 rs634304 (15808, 11)\n",
      "1626 rs637094 (15809, 11)\n",
      "1628 rs6435048 (15818, 11)\n",
      "1629 rs6437074 (15833, 11)\n",
      "1630 rs6441198 (15842, 11)\n",
      "1631 rs6457839 (15845, 11)\n",
      "1632 rs6460895 (15848, 11)\n",
      "1633 rs6460896 (15851, 11)\n",
      "1635 rs647080 (15866, 11)\n",
      "1636 rs6478608 (15870, 11)\n",
      "1638 rs6498070 (15873, 11)\n",
      "1639 rs6502633 (15886, 11)\n",
      "1641 rs653765 (15889, 11)\n",
      "1642 rs6540443 (15900, 11)\n",
      "1643 rs6546829 (15918, 11)\n",
      "1647 rs6551277 (15922, 11)\n",
      "1648 rs6551278 (15926, 11)\n",
      "1649 rs655240 (15940, 11)\n",
      "1651 rs656623 (15950, 11)\n",
      "1652 rs6573718 (15954, 11)\n",
      "1653 rs6573719 (15958, 11)\n",
      "1654 rs6573726 (15964, 11)\n",
      "1655 rs6573733 (15969, 11)\n",
      "1656 rs6573735 (15972, 11)\n",
      "1657 rs6573743 (15975, 11)\n",
      "1658 rs6573744 (15979, 11)\n",
      "1659 rs6573747 (15983, 11)\n",
      "1663 rs657586 (15984, 11)\n",
      "1665 rs6577488 (15988, 11)\n",
      "1667 rs6617 (16021, 11)\n",
      "1672 rs66586255 (16032, 11)\n",
      "1673 rs6666223 (16040, 11)\n",
      "1674 rs6669395 (16041, 11)\n",
      "1675 rs6680385 (16050, 11)\n",
      "1676 rs6681483 (16055, 11)\n",
      "1679 rs6691979 (16057, 11)\n",
      "1681 rs6706235 (16078, 11)\n",
      "1682 rs67101035 (16114, 11)\n",
      "1683 rs6716776 (16132, 11)\n",
      "1684 rs6722536 (16135, 11)\n",
      "1685 rs673005 (16139, 11)\n",
      "1686 rs6730590 (16141, 11)\n",
      "1687 rs67340775 (16179, 11)\n",
      "1689 rs673882 (16201, 11)\n",
      "1690 rs6745545 (16203, 11)\n",
      "1691 rs675828 (16204, 11)\n",
      "1693 rs67652222 (16238, 11)\n",
      "1695 rs67899457 (16256, 11)\n",
      "1696 rs6806287 (16265, 11)\n",
      "1697 rs6806702 (16274, 11)\n",
      "1698 rs6830685 (16276, 11)\n",
      "1699 rs683671 (16277, 11)\n",
      "1700 rs684080 (16278, 11)\n",
      "1701 rs6842442 (16283, 11)\n",
      "1702 rs6871683 (16285, 11)\n",
      "1704 rs6889338 (16287, 11)\n",
      "1705 rs6890748 (16288, 11)\n",
      "1706 rs6890923 (16289, 11)\n",
      "1709 rs6901575 (16357, 11)\n",
      "1710 rs6902389 (16369, 11)\n",
      "1711 rs6902392 (16381, 11)\n",
      "1712 rs6903015 (16393, 11)\n",
      "1713 rs6903652 (16449, 11)\n",
      "1714 rs6903823 (16505, 11)\n",
      "1716 rs6908156 (16517, 11)\n",
      "1719 rs6913660 (16534, 11)\n",
      "1720 rs6921772 (16540, 11)\n",
      "1721 rs6923139 (16552, 11)\n",
      "1722 rs6926869 (16553, 11)\n",
      "1726 rs6939727 (16554, 11)\n",
      "1727 rs6939978 (16566, 11)\n",
      "1728 rs6942483 (16570, 11)\n",
      "1729 rs6949073 (16573, 11)\n",
      "1732 rs698269 (16594, 11)\n",
      "1734 rs6996860 (16613, 11)\n",
      "1736 rs7019254 (16630, 11)\n",
      "1737 rs7021 (16644, 11)\n",
      "1740 rs702691 (16660, 11)\n",
      "1741 rs7033443 (16677, 11)\n",
      "1743 rs7046540 (16684, 11)\n",
      "1744 rs7067970 (16726, 11)\n",
      "1747 rs709206 (16739, 11)\n",
      "1748 rs7094595 (16741, 11)\n",
      "1749 rs709937 (16751, 11)\n",
      "1750 rs710172 (16753, 11)\n",
      "1753 rs7119426 (16776, 11)\n",
      "1754 rs7120113 (16783, 11)\n",
      "1755 rs7123729 (16789, 11)\n",
      "1756 rs7128102 (16797, 11)\n",
      "1758 rs7142669 (16803, 11)\n",
      "1760 rs7145295 (16808, 11)\n",
      "1761 rs7146682 (16822, 11)\n",
      "1762 rs7148456 (16841, 11)\n",
      "1763 rs7148903 (16842, 11)\n",
      "1764 rs7149112 (16845, 11)\n",
      "1765 rs7152947 (16847, 11)\n",
      "1769 rs7156852 (16852, 11)\n",
      "1770 rs7159871 (16854, 11)\n",
      "1772 rs7170068 (16865, 11)\n",
      "1774 rs7174985 (16869, 11)\n",
      "1775 rs7187202 (16892, 11)\n",
      "1776 rs718741 (16906, 11)\n",
      "1777 rs719095 (16907, 11)\n",
      "1778 rs7192848 (16909, 11)\n",
      "1779 rs7200757 (16938, 11)\n",
      "1780 rs7204974 (16955, 11)\n",
      "1781 rs7210978 (16957, 11)\n",
      "1782 rs7214152 (16959, 11)\n",
      "1783 rs7214988 (16960, 11)\n",
      "1785 rs7216572 (16970, 11)\n",
      "1786 rs7222358 (16972, 11)\n",
      "1791 rs7243008 (16973, 11)\n",
      "1792 rs7249925 (16985, 11)\n",
      "1793 rs72500859 (16990, 11)\n",
      "1794 rs7251295 (17001, 11)\n",
      "1795 rs72630624 (17022, 11)\n",
      "1797 rs72696119 (17024, 11)\n",
      "1798 rs72698842 (17034, 11)\n",
      "1799 rs72708145 (17040, 11)\n",
      "1805 rs72771404 (17042, 11)\n",
      "1806 rs72772787 (17045, 11)\n",
      "1810 rs72817596 (17058, 11)\n",
      "1811 rs728238 (17064, 11)\n",
      "1812 rs72839449 (17081, 11)\n",
      "1814 rs7284768 (17086, 11)\n",
      "1816 rs7292804 (17091, 11)\n",
      "1817 rs7293100 (17096, 11)\n",
      "1820 rs72986630 (17099, 11)\n",
      "1821 rs7299940 (17104, 11)\n",
      "1823 rs73050128 (17106, 11)\n",
      "1825 rs73084575 (17116, 11)\n",
      "1826 rs73086540 (17127, 11)\n",
      "1827 rs73086588 (17138, 11)\n",
      "1828 rs73086595 (17149, 11)\n",
      "1829 rs73086597 (17160, 11)\n",
      "1830 rs73088416 (17171, 11)\n",
      "1831 rs73088421 (17182, 11)\n",
      "1833 rs73169087 (17183, 11)\n",
      "1834 rs73169089 (17184, 11)\n",
      "1840 rs733672 (17189, 11)\n",
      "1841 rs73396802 (17230, 11)\n",
      "1848 rs735765 (17294, 11)\n",
      "1849 rs73598006 (17311, 11)\n",
      "1850 rs73615919 (17320, 11)\n",
      "1853 rs737321 (17321, 11)\n",
      "1854 rs73802964 (17322, 11)\n",
      "1855 rs73805839 (17323, 11)\n",
      "1856 rs7383287 (17347, 11)\n",
      "1857 rs73978969 (17349, 11)\n",
      "1858 rs7406982 (17362, 11)\n",
      "1861 rs743572 (17404, 11)\n",
      "1862 rs74395031 (17405, 11)\n",
      "1864 rs74580703 (17406, 11)\n",
      "1865 rs746011 (17431, 11)\n",
      "1866 rs74609824 (17432, 11)\n",
      "1867 rs7462053 (17441, 11)\n",
      "1868 rs74629553 (17456, 11)\n",
      "1869 rs746694 (17496, 11)\n",
      "1870 rs747685 (17498, 11)\n",
      "1872 rs751627 (17514, 11)\n",
      "1873 rs7521145 (17515, 11)\n",
      "1876 rs7521538 (17523, 11)\n",
      "1881 rs7527402 (17524, 11)\n",
      "1884 rs7533315 (17531, 11)\n",
      "1886 rs7534676 (17541, 11)\n",
      "1887 rs7536869 (17543, 11)\n",
      "1889 rs754184 (17545, 11)\n",
      "1890 rs7542594 (17547, 11)\n",
      "1891 rs7542608 (17552, 11)\n",
      "1893 rs7543548 (17559, 11)\n",
      "1896 rs75477292 (17560, 11)\n",
      "1899 rs7550347 (17565, 11)\n",
      "1901 rs75559897 (17573, 11)\n",
      "1902 rs756138 (17579, 11)\n",
      "1903 rs756139 (17585, 11)\n",
      "1904 rs75629461 (17586, 11)\n",
      "1906 rs75657440 (17587, 11)\n",
      "1907 rs75685335 (17591, 11)\n",
      "1908 rs756999 (17610, 11)\n",
      "1909 rs7570547 (17639, 11)\n",
      "1910 rs7570730 (17641, 11)\n",
      "1911 rs7574378 (17653, 11)\n",
      "1912 rs757791 (17655, 11)\n",
      "1914 rs7579359 (17666, 11)\n",
      "1915 rs7579996 (17668, 11)\n",
      "1916 rs75800607 (17670, 11)\n",
      "1917 rs758324 (17687, 11)\n",
      "1919 rs7584565 (17716, 11)\n",
      "1920 rs7584810 (17721, 11)\n",
      "1921 rs7587117 (17732, 11)\n",
      "1923 rs7599591 (17744, 11)\n",
      "1924 rs76028451 (17745, 11)\n",
      "1925 rs7604588 (17764, 11)\n",
      "1926 rs7605146 (17770, 11)\n",
      "1929 rs7614981 (17796, 11)\n",
      "1930 rs7618915 (17842, 11)\n",
      "1933 rs762995 (17883, 11)\n",
      "1937 rs763254 (17884, 11)\n",
      "1938 rs763263 (17946, 11)\n",
      "1940 rs76337233 (17951, 11)\n",
      "1941 rs76344302 (17955, 11)\n",
      "1942 rs7636227 (17980, 11)\n",
      "1943 rs7637291 (17983, 11)\n",
      "1947 rs7639867 (17989, 11)\n",
      "1948 rs7644528 (17990, 11)\n",
      "1952 rs7655924 (17991, 11)\n",
      "1955 rs7666219 (17996, 11)\n",
      "1956 rs7673231 (17998, 11)\n",
      "1959 rs7688417 (18003, 11)\n",
      "1961 rs77012921 (18007, 11)\n",
      "1962 rs7701755 (18070, 11)\n",
      "1966 rs77138653 (18071, 11)\n",
      "1967 rs77157935 (18075, 11)\n",
      "1969 rs77250160 (18091, 11)\n",
      "1971 rs7729529 (18092, 11)\n",
      "1972 rs7739256 (18093, 11)\n",
      "1974 rs7751952 (18112, 11)\n",
      "1979 rs77772792 (18124, 11)\n",
      "1980 rs7781670 (18127, 11)\n",
      "1984 rs7788921 (18128, 11)\n",
      "1986 rs7804433 (18131, 11)\n",
      "1989 rs782210 (18133, 11)\n",
      "1990 rs782239 (18135, 11)\n",
      "1991 rs782246 (18137, 11)\n",
      "1992 rs7822958 (18198, 11)\n",
      "1993 rs783539 (18236, 11)\n",
      "1996 rs78532287 (18272, 11)\n",
      "1997 rs78537727 (18289, 11)\n",
      "2002 rs786613 (18291, 11)\n",
      "2005 rs78866909 (18309, 11)\n",
      "2006 rs7899084 (18351, 11)\n",
      "2012 rs79206825 (18360, 11)\n",
      "2016 rs79375669 (18377, 11)\n",
      "2017 rs7940442 (18379, 11)\n",
      "2018 rs7946010 (18388, 11)\n",
      "2023 rs79712000 (18392, 11)\n",
      "2026 rs797266 (18393, 11)\n",
      "2028 rs79757743 (18397, 11)\n",
      "2029 rs7978610 (18413, 11)\n",
      "2030 rs7979816 (18418, 11)\n",
      "2031 rs79810375 (18419, 11)\n",
      "2032 rs798744 (18436, 11)\n",
      "2034 rs80001635 (18444, 11)\n",
      "2036 rs8005740 (18448, 11)\n",
      "2038 rs8007762 (18451, 11)\n",
      "2041 rs801093 (18453, 11)\n",
      "2042 rs801168 (18522, 11)\n",
      "2043 rs801183 (18591, 11)\n",
      "2044 rs8012119 (18592, 11)\n",
      "2047 rs8016213 (18596, 11)\n",
      "2048 rs8016725 (18599, 11)\n",
      "2052 rs8021098 (18604, 11)\n",
      "2054 rs8024814 (18605, 11)\n",
      "2057 rs80338392 (18614, 11)\n",
      "2058 rs80346622 (18631, 11)\n",
      "2061 rs8039184 (18635, 11)\n",
      "2065 rs8047119 (18652, 11)\n",
      "2066 rs8052408 (18653, 11)\n",
      "2067 rs805257 (18704, 11)\n",
      "2068 rs805290 (18749, 11)\n",
      "2069 rs8057119 (18764, 11)\n",
      "2070 rs8064558 (18766, 11)\n",
      "2071 rs8066602 (18791, 11)\n",
      "2072 rs8066731 (18816, 11)\n",
      "2073 rs8067439 (18826, 11)\n",
      "2074 rs8067576 (18856, 11)\n",
      "2076 rs8075602 (18882, 11)\n",
      "2077 rs8077172 (18896, 11)\n",
      "2078 rs8081961 (18898, 11)\n",
      "2079 rs8084 (18945, 11)\n",
      "2080 rs8086812 (18946, 11)\n",
      "2081 rs8111617 (18947, 11)\n",
      "2083 rs813115 (18996, 11)\n",
      "2085 rs8139758 (19014, 11)\n",
      "2091 rs841116 (19016, 11)\n",
      "2092 rs853685 (19078, 11)\n",
      "2093 rs853694 (19140, 11)\n",
      "2094 rs8539 (19146, 11)\n",
      "2095 rs858819 (19148, 11)\n",
      "2096 rs861539 (19170, 11)\n",
      "2098 rs879606 (19181, 11)\n",
      "2099 rs884548 (19201, 11)\n",
      "2100 rs885912 (19250, 11)\n",
      "2101 rs886425 (19303, 11)\n",
      "2102 rs893280 (19304, 11)\n",
      "2103 rs900947 (19311, 11)\n",
      "2104 rs900948 (19319, 11)\n",
      "2105 rs900949 (19326, 11)\n",
      "2106 rs903501 (19340, 11)\n",
      "2107 rs903502 (19355, 11)\n",
      "2108 rs903503 (19370, 11)\n",
      "2109 rs903504 (19385, 11)\n",
      "2110 rs907088 (19400, 11)\n",
      "2111 rs907089 (19415, 11)\n",
      "2112 rs907090 (19430, 11)\n",
      "2113 rs907094 (19441, 11)\n",
      "2114 rs909865 (19442, 11)\n",
      "2115 rs914160 (19443, 11)\n",
      "2119 rs926300 (19460, 11)\n",
      "2120 rs9265985 (19541, 11)\n",
      "2121 rs9266066 (19606, 11)\n",
      "2123 rs9272271 (19656, 11)\n",
      "2124 rs9274659 (19667, 11)\n",
      "2125 rs9281054 (19720, 11)\n",
      "2126 rs9291710 (19723, 11)\n",
      "2127 rs9303274 (19737, 11)\n",
      "2128 rs931145 (19740, 11)\n",
      "2129 rs931620 (19741, 11)\n",
      "2131 rs931992 (19754, 11)\n",
      "2132 rs9320322 (19755, 11)\n",
      "2134 rs9323482 (19760, 11)\n",
      "2135 rs9323489 (19763, 11)\n",
      "2136 rs9323491 (19768, 11)\n",
      "2137 rs9323492 (19773, 11)\n",
      "2138 rs9323493 (19778, 11)\n",
      "2140 rs9348774 (19809, 11)\n",
      "2142 rs9357045 (19839, 11)\n",
      "2143 rs9368531 (19880, 11)\n",
      "2144 rs9368563 (19952, 11)\n",
      "2145 rs9379873 (19975, 11)\n",
      "2146 rs9379952 (19996, 11)\n",
      "2147 rs9379962 (20012, 11)\n",
      "2148 rs9380522 (20016, 11)\n",
      "2151 rs9400239 (20017, 11)\n",
      "2153 rs941653 (20026, 11)\n",
      "2156 rs9425758 (20031, 11)\n",
      "2158 rs943297 (20033, 11)\n",
      "2160 rs9461536 (20051, 11)\n",
      "2161 rs9461856 (20058, 11)\n",
      "2162 rs9467701 (20070, 11)\n",
      "2163 rs9467707 (20082, 11)\n",
      "2164 rs9467708 (20094, 11)\n",
      "2165 rs9468174 (20099, 11)\n",
      "2166 rs9468350 (20156, 11)\n",
      "2168 rs9481189 (20157, 11)\n",
      "2169 rs949142 (20177, 11)\n",
      "2173 rs960069 (20180, 11)\n",
      "2175 rs9614952 (20181, 11)\n",
      "2177 rs9652567 (20226, 11)\n",
      "2178 rs966544 (20228, 11)\n",
      "2179 rs9675194 (20243, 11)\n",
      "2180 rs9677635 (20255, 11)\n",
      "2181 rs9678430 (20257, 11)\n",
      "2183 rs9722212 (20274, 11)\n",
      "2184 rs9722901 (20291, 11)\n",
      "2187 rs9784311 (20293, 11)\n",
      "2188 rs978565 (20294, 11)\n",
      "2192 rs9789254 (20295, 11)\n",
      "2193 rs9809968 (20297, 11)\n",
      "2194 rs9811396 (20300, 11)\n",
      "2197 rs9861125 (20302, 11)\n",
      "2199 rs9868500 (20305, 11)\n",
      "2200 rs9873195 (20308, 11)\n",
      "2201 rs9880806 (20314, 11)\n",
      "2202 rs9881468 (20352, 11)\n",
      "2203 rs9882911 (20356, 11)\n",
      "2204 rs9890075 (20381, 11)\n",
      "2209 rs9905950 (20406, 11)\n",
      "2210 rs990871 (20411, 11)\n",
      "2211 rs9917108 (20432, 11)\n",
      "2220 rs9986596 (20499, 11)\n",
      "CPU times: user 12min 20s, sys: 38.4 s, total: 12min 58s\n",
      "Wall time: 2h 6min 1s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# going to take a ridiculously long time\n",
    "eqtl_gtex_df = pd.DataFrame(columns = ['chromosome', 'datasetId', 'gencodeId', 'geneSymbol', 'geneSymbolUpper',\n",
    "       'nes', 'pValue', 'pos', 'snpId', 'tissueSiteDetailId', 'variantId'])\n",
    "\n",
    "prev_shape = eqtl_gtex_df.shape\n",
    "for idx,variant_id in enumerate(rsids):\n",
    "    for tissue in tissues:    \n",
    "        url = 'https://gtexportal.org/rest/v1/association/singleTissueEqtl?format=json&snpId={}&tissueSiteDetailId={}&datasetId=gtex_v8'.format(variant_id,tissue)\n",
    "        response = requests.get(url).json()['singleTissueEqtl']\n",
    "        if len(response)>0:\n",
    "            response_df = pd.DataFrame(response)\n",
    "            eqtl_gtex_df = pd.concat([eqtl_gtex_df,response_df])\n",
    "    if eqtl_gtex_df.shape!=prev_shape:\n",
    "        print(idx, variant_id,eqtl_gtex_df.shape )\n",
    "        prev_shape = eqtl_gtex_df.shape\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(20499, 11) (1499,) 2221\n"
     ]
    }
   ],
   "source": [
    "print(eqtl_gtex_df.shape,eqtl_gtex_df.snpId.unique().shape,len(rsids))\n",
    "eqtl_gtex_df.to_csv(os.path.join(save_dir, 'eqtl_gtex_df.csv'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1499 of the 2221 MPRA tested snps have an eqtl for a total of 20499 egenes for the eqtls. \n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3C. rna filt\n",
    "now we need to filter those eqtls for rnaseq expression > 1 in the active cell types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Astrocytes', 'SL_D0', 'SL_D2', 'SLC_D0', 'SLC_D2', 'H9_D0', 'H9_D2',\n",
      "       'H9_D10', 'H9_D28'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "rna_df = pd.read_csv('/Users/mguo123/Documents/pan_omics_psych/data/interim/rna/tissue_tpm_sym.csv',index_col=0)\n",
    "print(rna_df.columns)\n",
    "def get_geneexp(gene, tissue,df = rna_df):\n",
    "    try:\n",
    "        return df.loc[gene,tissue]\n",
    "    except:\n",
    "        return 0\n",
    "\n",
    "def get_min_geneexp(gene, tissues =['Astrocytes', 'SL_D0', 'SLC_D0', 'H9_D0', 'H9_D2','H9_D10'] ,df = rna_df):\n",
    "    try:\n",
    "        return df.loc[gene,:].min()\n",
    "    except:\n",
    "        return 0\n",
    "    \n",
    "def get_max_geneexp(gene, tissues =['Astrocytes', 'SL_D0', 'SLC_D0', 'H9_D0', 'H9_D2','H9_D10'] ,df = rna_df):\n",
    "    try:\n",
    "        return df.loc[gene,:].max()\n",
    "    except:\n",
    "        return 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "# tissue_mapper_mpra2rna = dict(zip(['Astrocytes', 'SL', 'SLC',  'H9D0', 'H9D2',\n",
    "#        'H9D4', 'H9D10'], ['Astrocytes', 'SL_D0',  'SLC_D0','H9_D0', 'H9_D2',\n",
    "#        'H9_D2', 'H9_D10']))    \n",
    "# tissue_mapper_mpra2rna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>datasetId</th>\n",
       "      <th>gencodeId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>geneSymbolUpper</th>\n",
       "      <th>nes</th>\n",
       "      <th>pValue</th>\n",
       "      <th>pos</th>\n",
       "      <th>snpId</th>\n",
       "      <th>tissueSiteDetailId</th>\n",
       "      <th>variantId</th>\n",
       "      <th>rowname</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>bool_pass_exp</th>\n",
       "      <th>min_valid_exp</th>\n",
       "      <th>tissues_valid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.324001</td>\n",
       "      <td>2.472690e-07</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.324001</td>\n",
       "      <td>2.472690e-07</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.272681</td>\n",
       "      <td>7.182740e-09</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.272681</td>\n",
       "      <td>7.182740e-09</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.235531</td>\n",
       "      <td>1.680670e-08</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome datasetId           gencodeId geneSymbol geneSymbolUpper  \\\n",
       "0       chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "1       chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "2       chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "3       chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "4       chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "\n",
       "        nes        pValue       pos       snpId           tissueSiteDetailId  \\\n",
       "0 -0.324001  2.472690e-07  42071771  rs10022045               Brain_Amygdala   \n",
       "1 -0.324001  2.472690e-07  42071771  rs10022045               Brain_Amygdala   \n",
       "2 -0.272681  7.182740e-09  42071771  rs10022045  Brain_Caudate_basal_ganglia   \n",
       "3 -0.272681  7.182740e-09  42071771  rs10022045  Brain_Caudate_basal_ganglia   \n",
       "4 -0.235531  1.680670e-08  42071771  rs10022045                 Brain_Cortex   \n",
       "\n",
       "               variantId        rowname  Linked_SNP  bool_pass_exp  \\\n",
       "0  chr4_42071771_T_C_b38  chr4_42073788  rs10022045           True   \n",
       "1  chr4_42071771_T_C_b38  chr4_42073788  rs10022045           True   \n",
       "2  chr4_42071771_T_C_b38  chr4_42073788  rs10022045           True   \n",
       "3  chr4_42071771_T_C_b38  chr4_42073788  rs10022045           True   \n",
       "4  chr4_42071771_T_C_b38  chr4_42073788  rs10022045           True   \n",
       "\n",
       "   min_valid_exp                                      tissues_valid  \n",
       "0             21  Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "1             21  Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "2             21  Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "3             21  Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "4             21  Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  "
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_gtex_df_rna[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 20 s, sys: 22 ms, total: 20.1 s\n",
      "Wall time: 20.1 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "THRES = 1\n",
    "eqtl_gtex_df_rna = eqtl_gtex_df.merge(lib_df[['rowname', 'Linked_SNP']],how='left',left_on='snpId',right_on='Linked_SNP')\n",
    "eqtl_gtex_df_rna['bool_pass_exp'] = False\n",
    "eqtl_gtex_df_rna['min_valid_exp'] = 0\n",
    "eqtl_gtex_df_rna['tissues_valid'] = ''\n",
    "\n",
    "for idx, row in eqtl_gtex_df_rna.iterrows():\n",
    "    gene = row['geneSymbolUpper']\n",
    "    if gene in rna_df.index:\n",
    "        gene_exp_series = rna_df.loc[gene,:][rna_df.loc[gene,:]>THRES]\n",
    "        exp_arr = list(gene_exp_series.values)\n",
    "        tissue_arr = list(gene_exp_series.index)\n",
    "        if len(exp_arr)>0:\n",
    "            eqtl_gtex_df_rna.at[idx, 'min_valid_exp'] = min(exp_arr)\n",
    "            eqtl_gtex_df_rna.at[idx,'bool_pass_exp'] = True\n",
    "            eqtl_gtex_df_rna.at[idx, 'tissues_valid'] = '|'.join(tissue_arr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(26726, 16) (41673, 16)\n",
      "(1359,) (1499,)\n"
     ]
    }
   ],
   "source": [
    "eqtl_gtex_df_rna_filt = eqtl_gtex_df_rna[eqtl_gtex_df_rna.bool_pass_exp]\n",
    "print(eqtl_gtex_df_rna_filt.shape, eqtl_gtex_df_rna.shape)\n",
    "print(eqtl_gtex_df_rna_filt.snpId.unique().shape, eqtl_gtex_df_rna.snpId.unique().shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "eqtl_gtex_df_rna_filt.to_csv(os.path.join(save_dir, 'eqtl_gtex_df_rna_filt.csv'))"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "1359 of the 1499 eqtl mpra lib  (2221 total mpra tests) have an egene that passes the rna filter\n",
    "\n",
    "26726 egenes of 41673 egenes have been found that pass filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chromosome', 'datasetId', 'gencodeId', 'geneSymbol', 'geneSymbolUpper',\n",
       "       'nes', 'pValue', 'pos', 'snpId', 'tissueSiteDetailId', 'variantId',\n",
       "       'rowname', 'Linked_SNP', 'bool_pass_exp', 'min_valid_exp',\n",
       "       'tissues_valid'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_gtex_df_rna_filt.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3B. psych stuff only\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/mguo123/opt/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (63,65) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Index(['name', 'X', 'Linked_SNP', 'Chr', 'Position', 'Index_SNP', 'Disease',\n",
       "       'DHS_tissue', 'eGene', 'eQTL_tissue',\n",
       "       ...\n",
       "       'Sngl10000bp', 'dbscSNV.ada_score', 'dbscSNV.rf_score', 'RawScore',\n",
       "       'PHRED', 'num_disease', 'disease_interest', 'pass_thres_low',\n",
       "       'pass_thres', 'CADD_score_bool'],\n",
       "      dtype='object', length=139)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_df  = pd.read_csv('~/Google Drive/1_khavari/noncancer_project/psych_lib_info.csv',index_col=0)\n",
    "lib_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "anxiety 179\n",
      "attent 142\n",
      "autism 54\n",
      "bipolar 400\n",
      "depress 424\n",
      "ocd 116\n",
      "panic 61\n",
      "personality 39\n",
      "schizo 1067\n",
      "traum 101\n"
     ]
    }
   ],
   "source": [
    "diseases = ['anxiety', 'attent', 'autism',\n",
    "       'bipolar', 'depress', 'ocd', 'panic', 'personality', 'schizo', 'traum']\n",
    "dz_to_rsid_dict = {}\n",
    "for dz in diseases:\n",
    "    dz_to_rsid_dict[dz] = sorted(set(lib_df[lib_df[dz]==1].Linked_SNP.to_list()))\n",
    "    print(dz,len(dz_to_rsid_dict[dz]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['gwas_enrichment_shuffle/SL_D0_shuffle_100Times_actuallastline_diffloop_allATAC_loops_psych.txt',\n",
       " 'gwas_enrichment_shuffle/H9_D10_shuffle_100Times_actuallastline_diffloop_allATAC_loops_psych.txt',\n",
       " 'gwas_enrichment_shuffle/H9_D0_shuffle_100Times_actuallastline_diffloop_allATAC_loops_psych.txt',\n",
       " 'gwas_enrichment_shuffle/Astro_shuffle_100Times_actuallastline_diffloop_allATAC_loops_psych.txt',\n",
       " 'gwas_enrichment_shuffle/H9_D28_shuffle_100Times_actuallastline_diffloop_allATAC_loops_psych.txt',\n",
       " 'gwas_enrichment_shuffle/SLC_D0_shuffle_100Times_actuallastline_diffloop_allATAC_loops_psych.txt']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res_files = glob.glob('gwas_enrichment_shuffle/*psych.txt')\n",
    "res_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SL_D0\n",
      "H9_D10\n",
      "H9_D0\n",
      "Astro\n",
      "H9_D28\n",
      "SLC_D0\n"
     ]
    }
   ],
   "source": [
    "\n",
    "tissue_dz_pval_dict = dict()\n",
    "tissue_dz_or_dict = dict()\n",
    "for file in res_files:\n",
    "    tissue = os.path.basename(file).split('_shuffle')[0]\n",
    "    print(tissue)\n",
    "    df_orig = pd.read_table(file).fillna(0).drop('File',axis=1)\n",
    "    df = df_orig.div(df_orig.sum(axis=1), axis=0)\n",
    "    # subselect columns    \n",
    "    dz_to_pval = dict()\n",
    "    dz_to_or = dict()\n",
    "    for dz in diseases:\n",
    "        # print(dz)\n",
    "        dz_cols = dz_to_rsid_dict[dz]\n",
    "        # print(dz_cols)\n",
    "        dz_series = df[dz_cols].sum(axis=1)\n",
    "        pval = pd.Series(dz_series[100]<dz_series[:100]).sum()/100\n",
    "        oddsratio = dz_series[100]/dz_series[:100].mean()\n",
    "        dz_to_pval[dz] = pval\n",
    "        dz_to_or[dz] = oddsratio\n",
    "    \n",
    "    tissue_dz_pval_dict[tissue] = dz_to_pval\n",
    "    tissue_dz_or_dict[tissue] = dz_to_or"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SL_D0</th>\n",
       "      <th>H9_D10</th>\n",
       "      <th>H9_D0</th>\n",
       "      <th>Astro</th>\n",
       "      <th>H9_D28</th>\n",
       "      <th>SLC_D0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anxiety</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>attent</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>autism</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>bipolar</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>depress</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ocd</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>panic</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>personality</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>schizo</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>traum</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             SL_D0  H9_D10  H9_D0  Astro  H9_D28  SLC_D0\n",
       "anxiety        1.0    1.00    1.0   1.00    1.00     1.0\n",
       "attent         0.0    0.00    0.0   1.00    1.00     0.0\n",
       "autism         1.0    1.00    1.0   1.00    1.00     1.0\n",
       "bipolar        0.0    0.41    0.0   0.16    0.00     0.0\n",
       "depress        1.0    1.00    1.0   1.00    1.00     1.0\n",
       "ocd            0.0    0.93    0.0   1.00    1.00     0.0\n",
       "panic          1.0    1.00    1.0   0.00    1.00     1.0\n",
       "personality    1.0    1.00    1.0   1.00    0.32     1.0\n",
       "schizo         0.0    0.00    0.0   0.00    0.00     0.0\n",
       "traum          1.0    1.00    1.0   1.00    1.00     1.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tissue_dz_pval_psych_df = pd.DataFrame.from_dict(tissue_dz_pval_dict)\n",
    "display(tissue_dz_pval_psych_df)\n",
    "tissue_dz_pval_psych_df.to_csv(os.path.join(save_dir, 'tissue_dz_pval_psych_df.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SL_D0</th>\n",
       "      <th>H9_D10</th>\n",
       "      <th>H9_D0</th>\n",
       "      <th>Astro</th>\n",
       "      <th>H9_D28</th>\n",
       "      <th>SLC_D0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>anxiety</td>\n",
       "      <td>0.478711</td>\n",
       "      <td>0.681938</td>\n",
       "      <td>0.650068</td>\n",
       "      <td>0.780981</td>\n",
       "      <td>0.749333</td>\n",
       "      <td>0.837868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>attent</td>\n",
       "      <td>1.241426</td>\n",
       "      <td>1.275240</td>\n",
       "      <td>1.067224</td>\n",
       "      <td>0.810272</td>\n",
       "      <td>0.863916</td>\n",
       "      <td>1.058824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>autism</td>\n",
       "      <td>0.500870</td>\n",
       "      <td>0.146647</td>\n",
       "      <td>0.178139</td>\n",
       "      <td>0.036936</td>\n",
       "      <td>0.118317</td>\n",
       "      <td>0.103360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>bipolar</td>\n",
       "      <td>1.366600</td>\n",
       "      <td>1.033218</td>\n",
       "      <td>1.314912</td>\n",
       "      <td>1.034548</td>\n",
       "      <td>1.340208</td>\n",
       "      <td>1.265238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>depress</td>\n",
       "      <td>0.579233</td>\n",
       "      <td>0.519790</td>\n",
       "      <td>0.448986</td>\n",
       "      <td>0.445122</td>\n",
       "      <td>0.673251</td>\n",
       "      <td>0.447916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ocd</td>\n",
       "      <td>1.565776</td>\n",
       "      <td>0.787648</td>\n",
       "      <td>2.118382</td>\n",
       "      <td>0.763021</td>\n",
       "      <td>0.764552</td>\n",
       "      <td>1.532625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>panic</td>\n",
       "      <td>0.796175</td>\n",
       "      <td>0.779368</td>\n",
       "      <td>0.822677</td>\n",
       "      <td>1.193707</td>\n",
       "      <td>0.644264</td>\n",
       "      <td>0.793025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>personality</td>\n",
       "      <td>0.402055</td>\n",
       "      <td>0.686717</td>\n",
       "      <td>0.573936</td>\n",
       "      <td>0.798470</td>\n",
       "      <td>1.032437</td>\n",
       "      <td>0.313637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>schizo</td>\n",
       "      <td>1.400875</td>\n",
       "      <td>1.634911</td>\n",
       "      <td>1.455309</td>\n",
       "      <td>1.659355</td>\n",
       "      <td>1.548476</td>\n",
       "      <td>1.447566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>traum</td>\n",
       "      <td>0.299106</td>\n",
       "      <td>0.234424</td>\n",
       "      <td>0.344939</td>\n",
       "      <td>0.628152</td>\n",
       "      <td>0.063152</td>\n",
       "      <td>0.471222</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                SL_D0    H9_D10     H9_D0     Astro    H9_D28    SLC_D0\n",
       "anxiety      0.478711  0.681938  0.650068  0.780981  0.749333  0.837868\n",
       "attent       1.241426  1.275240  1.067224  0.810272  0.863916  1.058824\n",
       "autism       0.500870  0.146647  0.178139  0.036936  0.118317  0.103360\n",
       "bipolar      1.366600  1.033218  1.314912  1.034548  1.340208  1.265238\n",
       "depress      0.579233  0.519790  0.448986  0.445122  0.673251  0.447916\n",
       "ocd          1.565776  0.787648  2.118382  0.763021  0.764552  1.532625\n",
       "panic        0.796175  0.779368  0.822677  1.193707  0.644264  0.793025\n",
       "personality  0.402055  0.686717  0.573936  0.798470  1.032437  0.313637\n",
       "schizo       1.400875  1.634911  1.455309  1.659355  1.548476  1.447566\n",
       "traum        0.299106  0.234424  0.344939  0.628152  0.063152  0.471222"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tissue_dz_or_psych_df = pd.DataFrame.from_dict(tissue_dz_or_dict).fillna(1)\n",
    "display(tissue_dz_or_psych_df)\n",
    "tissue_dz_or_psych_df.to_csv(os.path.join(save_dir, 'tissue_dz_or_psych_df.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dz</th>\n",
       "      <th>tissue</th>\n",
       "      <th>or</th>\n",
       "      <th>pval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>anxiety</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>0.478711</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>attent</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>1.241426</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>autism</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>0.500870</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>bipolar</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>1.366600</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>depress</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>0.579233</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ocd</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>1.565776</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>panic</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>0.796175</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>personality</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>0.402055</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>schizo</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>1.400875</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>traum</td>\n",
       "      <td>SL_D0</td>\n",
       "      <td>0.299106</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>anxiety</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>0.681938</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>attent</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>1.275240</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>autism</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>0.146647</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>bipolar</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>1.033218</td>\n",
       "      <td>0.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>depress</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>0.519790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>ocd</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>0.787648</td>\n",
       "      <td>0.93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>panic</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>0.779368</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>personality</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>0.686717</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>schizo</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>1.634911</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>traum</td>\n",
       "      <td>H9_D10</td>\n",
       "      <td>0.234424</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>anxiety</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>0.650068</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>attent</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>1.067224</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>autism</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>0.178139</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>bipolar</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>1.314912</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>depress</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>0.448986</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>ocd</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>2.118382</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>panic</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>0.822677</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>personality</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>0.573936</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>schizo</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>1.455309</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>traum</td>\n",
       "      <td>H9_D0</td>\n",
       "      <td>0.344939</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>anxiety</td>\n",
       "      <td>Astro</td>\n",
       "      <td>0.780981</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>attent</td>\n",
       "      <td>Astro</td>\n",
       "      <td>0.810272</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>autism</td>\n",
       "      <td>Astro</td>\n",
       "      <td>0.036936</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>bipolar</td>\n",
       "      <td>Astro</td>\n",
       "      <td>1.034548</td>\n",
       "      <td>0.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>depress</td>\n",
       "      <td>Astro</td>\n",
       "      <td>0.445122</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>ocd</td>\n",
       "      <td>Astro</td>\n",
       "      <td>0.763021</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>panic</td>\n",
       "      <td>Astro</td>\n",
       "      <td>1.193707</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>personality</td>\n",
       "      <td>Astro</td>\n",
       "      <td>0.798470</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>schizo</td>\n",
       "      <td>Astro</td>\n",
       "      <td>1.659355</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>traum</td>\n",
       "      <td>Astro</td>\n",
       "      <td>0.628152</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>anxiety</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>0.749333</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>attent</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>0.863916</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>autism</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>0.118317</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>bipolar</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>1.340208</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44</td>\n",
       "      <td>depress</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>0.673251</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>ocd</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>0.764552</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>panic</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>0.644264</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>personality</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>1.032437</td>\n",
       "      <td>0.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>schizo</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>1.548476</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>49</td>\n",
       "      <td>traum</td>\n",
       "      <td>H9_D28</td>\n",
       "      <td>0.063152</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50</td>\n",
       "      <td>anxiety</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>0.837868</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>51</td>\n",
       "      <td>attent</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>1.058824</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>52</td>\n",
       "      <td>autism</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>0.103360</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>bipolar</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>1.265238</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>54</td>\n",
       "      <td>depress</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>0.447916</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55</td>\n",
       "      <td>ocd</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>1.532625</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>panic</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>0.793025</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>personality</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>0.313637</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>schizo</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>1.447566</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>traum</td>\n",
       "      <td>SLC_D0</td>\n",
       "      <td>0.471222</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             dz  tissue        or  pval\n",
       "0       anxiety   SL_D0  0.478711  1.00\n",
       "1        attent   SL_D0  1.241426  0.00\n",
       "2        autism   SL_D0  0.500870  1.00\n",
       "3       bipolar   SL_D0  1.366600  0.00\n",
       "4       depress   SL_D0  0.579233  1.00\n",
       "5           ocd   SL_D0  1.565776  0.00\n",
       "6         panic   SL_D0  0.796175  1.00\n",
       "7   personality   SL_D0  0.402055  1.00\n",
       "8        schizo   SL_D0  1.400875  0.00\n",
       "9         traum   SL_D0  0.299106  1.00\n",
       "10      anxiety  H9_D10  0.681938  1.00\n",
       "11       attent  H9_D10  1.275240  0.00\n",
       "12       autism  H9_D10  0.146647  1.00\n",
       "13      bipolar  H9_D10  1.033218  0.41\n",
       "14      depress  H9_D10  0.519790  1.00\n",
       "15          ocd  H9_D10  0.787648  0.93\n",
       "16        panic  H9_D10  0.779368  1.00\n",
       "17  personality  H9_D10  0.686717  1.00\n",
       "18       schizo  H9_D10  1.634911  0.00\n",
       "19        traum  H9_D10  0.234424  1.00\n",
       "20      anxiety   H9_D0  0.650068  1.00\n",
       "21       attent   H9_D0  1.067224  0.00\n",
       "22       autism   H9_D0  0.178139  1.00\n",
       "23      bipolar   H9_D0  1.314912  0.00\n",
       "24      depress   H9_D0  0.448986  1.00\n",
       "25          ocd   H9_D0  2.118382  0.00\n",
       "26        panic   H9_D0  0.822677  1.00\n",
       "27  personality   H9_D0  0.573936  1.00\n",
       "28       schizo   H9_D0  1.455309  0.00\n",
       "29        traum   H9_D0  0.344939  1.00\n",
       "30      anxiety   Astro  0.780981  1.00\n",
       "31       attent   Astro  0.810272  1.00\n",
       "32       autism   Astro  0.036936  1.00\n",
       "33      bipolar   Astro  1.034548  0.16\n",
       "34      depress   Astro  0.445122  1.00\n",
       "35          ocd   Astro  0.763021  1.00\n",
       "36        panic   Astro  1.193707  0.00\n",
       "37  personality   Astro  0.798470  1.00\n",
       "38       schizo   Astro  1.659355  0.00\n",
       "39        traum   Astro  0.628152  1.00\n",
       "40      anxiety  H9_D28  0.749333  1.00\n",
       "41       attent  H9_D28  0.863916  1.00\n",
       "42       autism  H9_D28  0.118317  1.00\n",
       "43      bipolar  H9_D28  1.340208  0.00\n",
       "44      depress  H9_D28  0.673251  1.00\n",
       "45          ocd  H9_D28  0.764552  1.00\n",
       "46        panic  H9_D28  0.644264  1.00\n",
       "47  personality  H9_D28  1.032437  0.32\n",
       "48       schizo  H9_D28  1.548476  0.00\n",
       "49        traum  H9_D28  0.063152  1.00\n",
       "50      anxiety  SLC_D0  0.837868  1.00\n",
       "51       attent  SLC_D0  1.058824  0.00\n",
       "52       autism  SLC_D0  0.103360  1.00\n",
       "53      bipolar  SLC_D0  1.265238  0.00\n",
       "54      depress  SLC_D0  0.447916  1.00\n",
       "55          ocd  SLC_D0  1.532625  0.00\n",
       "56        panic  SLC_D0  0.793025  1.00\n",
       "57  personality  SLC_D0  0.313637  1.00\n",
       "58       schizo  SLC_D0  1.447566  0.00\n",
       "59        traum  SLC_D0  0.471222  1.00"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tissue_dz_pval_psych_df_long = pd.melt(tissue_dz_pval_psych_df.reset_index(),id_vars='index')\n",
    "tissue_dz_pval_psych_df_long.columns = ['dz','tissue','pval']\n",
    "tissue_dz_or_psych_df_long = pd.melt(tissue_dz_or_psych_df.reset_index(),id_vars='index')\n",
    "tissue_dz_or_psych_df_long.columns = ['dz','tissue','or']\n",
    "tissue_dz_all_psych_df = tissue_dz_or_psych_df_long.merge(tissue_dz_pval_psych_df_long,on=['dz','tissue'])\n",
    "display(tissue_dz_all_psych_df)\n",
    "tissue_dz_all_psych_df.to_csv(os.path.join(save_dir, 'tissue_dz_psych_df_long.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>datasetId</th>\n",
       "      <th>gencodeId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>geneSymbolUpper</th>\n",
       "      <th>nes</th>\n",
       "      <th>pValue</th>\n",
       "      <th>pos</th>\n",
       "      <th>snpId</th>\n",
       "      <th>tissueSiteDetailId</th>\n",
       "      <th>variantId</th>\n",
       "      <th>rowname</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>bool_pass_exp</th>\n",
       "      <th>min_valid_exp</th>\n",
       "      <th>tissues_valid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.324001</td>\n",
       "      <td>2.472690e-07</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.324001</td>\n",
       "      <td>2.472690e-07</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Amygdala</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.272681</td>\n",
       "      <td>7.182740e-09</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.272681</td>\n",
       "      <td>7.182740e-09</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Caudate_basal_ganglia</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.235531</td>\n",
       "      <td>1.680670e-08</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.235531</td>\n",
       "      <td>1.680670e-08</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Cortex</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.209118</td>\n",
       "      <td>2.460570e-06</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.209118</td>\n",
       "      <td>2.460570e-06</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Hypothalamus</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.233575</td>\n",
       "      <td>3.805320e-10</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.233575</td>\n",
       "      <td>3.805320e-10</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.416739</td>\n",
       "      <td>2.218490e-09</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>chr4</td>\n",
       "      <td>gtex_v8</td>\n",
       "      <td>ENSG00000014824.13</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>-0.416739</td>\n",
       "      <td>2.218490e-09</td>\n",
       "      <td>42071771</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>Brain_Substantia_nigra</td>\n",
       "      <td>chr4_42071771_T_C_b38</td>\n",
       "      <td>chr4_42073788</td>\n",
       "      <td>rs10022045</td>\n",
       "      <td>True</td>\n",
       "      <td>21</td>\n",
       "      <td>Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome datasetId           gencodeId geneSymbol geneSymbolUpper  \\\n",
       "0        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "1        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "2        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "3        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "4        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "5        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "6        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "7        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "8        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "9        chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "10       chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "11       chr4   gtex_v8  ENSG00000014824.13    SLC30A9         SLC30A9   \n",
       "\n",
       "         nes        pValue       pos       snpId  \\\n",
       "0  -0.324001  2.472690e-07  42071771  rs10022045   \n",
       "1  -0.324001  2.472690e-07  42071771  rs10022045   \n",
       "2  -0.272681  7.182740e-09  42071771  rs10022045   \n",
       "3  -0.272681  7.182740e-09  42071771  rs10022045   \n",
       "4  -0.235531  1.680670e-08  42071771  rs10022045   \n",
       "5  -0.235531  1.680670e-08  42071771  rs10022045   \n",
       "6  -0.209118  2.460570e-06  42071771  rs10022045   \n",
       "7  -0.209118  2.460570e-06  42071771  rs10022045   \n",
       "8  -0.233575  3.805320e-10  42071771  rs10022045   \n",
       "9  -0.233575  3.805320e-10  42071771  rs10022045   \n",
       "10 -0.416739  2.218490e-09  42071771  rs10022045   \n",
       "11 -0.416739  2.218490e-09  42071771  rs10022045   \n",
       "\n",
       "                       tissueSiteDetailId              variantId  \\\n",
       "0                          Brain_Amygdala  chr4_42071771_T_C_b38   \n",
       "1                          Brain_Amygdala  chr4_42071771_T_C_b38   \n",
       "2             Brain_Caudate_basal_ganglia  chr4_42071771_T_C_b38   \n",
       "3             Brain_Caudate_basal_ganglia  chr4_42071771_T_C_b38   \n",
       "4                            Brain_Cortex  chr4_42071771_T_C_b38   \n",
       "5                            Brain_Cortex  chr4_42071771_T_C_b38   \n",
       "6                      Brain_Hypothalamus  chr4_42071771_T_C_b38   \n",
       "7                      Brain_Hypothalamus  chr4_42071771_T_C_b38   \n",
       "8   Brain_Nucleus_accumbens_basal_ganglia  chr4_42071771_T_C_b38   \n",
       "9   Brain_Nucleus_accumbens_basal_ganglia  chr4_42071771_T_C_b38   \n",
       "10                 Brain_Substantia_nigra  chr4_42071771_T_C_b38   \n",
       "11                 Brain_Substantia_nigra  chr4_42071771_T_C_b38   \n",
       "\n",
       "          rowname  Linked_SNP  bool_pass_exp  min_valid_exp  \\\n",
       "0   chr4_42073788  rs10022045           True             21   \n",
       "1   chr4_42073788  rs10022045           True             21   \n",
       "2   chr4_42073788  rs10022045           True             21   \n",
       "3   chr4_42073788  rs10022045           True             21   \n",
       "4   chr4_42073788  rs10022045           True             21   \n",
       "5   chr4_42073788  rs10022045           True             21   \n",
       "6   chr4_42073788  rs10022045           True             21   \n",
       "7   chr4_42073788  rs10022045           True             21   \n",
       "8   chr4_42073788  rs10022045           True             21   \n",
       "9   chr4_42073788  rs10022045           True             21   \n",
       "10  chr4_42073788  rs10022045           True             21   \n",
       "11  chr4_42073788  rs10022045           True             21   \n",
       "\n",
       "                                        tissues_valid  \n",
       "0   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "1   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "2   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "3   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "4   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "5   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "6   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "7   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "8   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "9   Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "10  Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  \n",
       "11  Astrocytes|SL_D0|SL_D2|SLC_D0|SLC_D2|H9_D0|H9_...  "
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_gtex_df_rna_filt[eqtl_gtex_df_rna_filt.snpId=='rs10022045']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "((13156, 4), (13156, 16))"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_gtex_df_rna_filt[['snpId','geneSymbolUpper','tissueSiteDetailId','geneSymbol']].drop_duplicates().shape, eqtl_gtex_df_rna_filt.drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['chromosome', 'datasetId', 'gencodeId', 'geneSymbol', 'geneSymbolUpper',\n",
      "       'nes', 'pValue', 'pos', 'snpId', 'tissueSiteDetailId', 'variantId',\n",
      "       'rowname', 'Linked_SNP', 'bool_pass_exp', 'min_valid_exp',\n",
      "       'tissues_valid'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(eqtl_gtex_df_rna_filt.columns)\n",
    "# eqtl_gtex_df_rna_filt[['snpId','geneSymbolUpper','tissueSiteDetailId','geneSymbol']].drop_duplicates().shape, eqtl_gtex_df_rna_filt.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [],
   "source": [
    "eqtl_gtex_df_rna_filt_by_snp = eqtl_gtex_df_rna_filt.drop_duplicates().groupby('snpId').agg({'geneSymbolUpper':'|'.join, 'tissueSiteDetailId':'|'.join,\n",
    "                                           'geneSymbol': lambda x: '|'.join(sorted(set(x))).upper()}).reset_index()\n",
    "eqtl_gtex_df_rna_filt_by_snp.fillna('',inplace=True)\n",
    "eqtl_gtex_df_rna_filt_by_snp.columns = ['Linked_SNP','all_genes','all_gtex_tissues','unique_egenes_gtex']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_df_gtex = lib_df_egene.merge(eqtl_gtex_df_rna_filt_by_snp,how='left', on='Linked_SNP').fillna('')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_rna_exp(row,col_to_filt):\n",
    "    genes = row[col_to_filt].split('|')\n",
    "    genes_pass_exp_filt = []\n",
    "    for gene in genes:\n",
    "        max_exp = get_max_geneexp(gene)\n",
    "        if max_exp>THRES:\n",
    "            genes_pass_exp_filt.append(gene)\n",
    "    return '|'.join(sorted(set(genes_pass_exp_filt)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_df_gtex['ngene'] = lib_df_gtex.apply(lambda x: filter_rna_exp(x, 'ngene'),axis=1)\n",
    "lib_df_gtex['ngene_haploreg'] = lib_df_gtex.apply(lambda x: filter_rna_exp(x, 'ngene_haploreg'),axis=1)\n",
    "lib_df_gtex['egene_nerve'] = lib_df_gtex.apply(lambda x: filter_rna_exp(x, 'egene_nerve'),axis=1)\n",
    "lib_df_gtex['egene_blood'] = lib_df_gtex.apply(lambda x: filter_rna_exp(x, 'egene_blood'),axis=1)\n",
    "lib_df_gtex['egene_all'] = lib_df_gtex.apply(lambda x: filter_rna_exp(x, 'egene_all'),axis=1)\n",
    "lib_df_gtex['all_genes_rna_filt'] = lib_df_gtex.apply(lambda x: filter_rna_exp(x, 'all_genes'),axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['name', 'X', 'Linked_SNP', 'Chr', 'Position', 'Index_SNP', 'Disease',\n",
       "       'DHS_tissue', 'eGene', 'eQTL_tissue',\n",
       "       ...\n",
       "       'rowname', 'ngene', 'ngene_haploreg', 'egene_nerve', 'egene_blood',\n",
       "       'egene_all', 'all_genes', 'all_gtex_tissues', 'unique_egenes_gtex',\n",
       "       'all_genes_rna_filt'],\n",
       "      dtype='object', length=149)"
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_df_gtex.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>X</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Index_SNP</th>\n",
       "      <th>Disease</th>\n",
       "      <th>DHS_tissue</th>\n",
       "      <th>eGene</th>\n",
       "      <th>eQTL_tissue</th>\n",
       "      <th>...</th>\n",
       "      <th>rowname</th>\n",
       "      <th>ngene</th>\n",
       "      <th>ngene_haploreg</th>\n",
       "      <th>egene_nerve</th>\n",
       "      <th>egene_blood</th>\n",
       "      <th>egene_all</th>\n",
       "      <th>all_genes</th>\n",
       "      <th>all_gtex_tissues</th>\n",
       "      <th>unique_egenes_gtex</th>\n",
       "      <th>all_genes_rna_filt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr1.111243375.all_alt.G</td>\n",
       "      <td>75004</td>\n",
       "      <td>rs12085135</td>\n",
       "      <td>chr1</td>\n",
       "      <td>111243375</td>\n",
       "      <td>rs12061304</td>\n",
       "      <td>panic disorder</td>\n",
       "      <td>t-helper_17_cell</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>chr1_111243375</td>\n",
       "      <td>KCNA3</td>\n",
       "      <td>KCNA3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1.111243375.Ref.C</td>\n",
       "      <td>75004</td>\n",
       "      <td>rs12085135</td>\n",
       "      <td>chr1</td>\n",
       "      <td>111243375</td>\n",
       "      <td>rs12061304</td>\n",
       "      <td>panic disorder</td>\n",
       "      <td>t-helper_17_cell</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>chr1_111243375</td>\n",
       "      <td>KCNA3</td>\n",
       "      <td>KCNA3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1.113938499.all_alt.C</td>\n",
       "      <td>1977</td>\n",
       "      <td>rs74395031</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113938499</td>\n",
       "      <td>rs115870915</td>\n",
       "      <td>type 2 diabetes,Obsessive Compulsive Disorder</td>\n",
       "      <td>astrocyte_of_the_spinal_cord</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>chr1_113938499</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1.113938499.Ref.T</td>\n",
       "      <td>1977</td>\n",
       "      <td>rs74395031</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113938499</td>\n",
       "      <td>rs115870915</td>\n",
       "      <td>type 2 diabetes,Obsessive Compulsive Disorder</td>\n",
       "      <td>astrocyte_of_the_spinal_cord</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>chr1_113938499</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1.11712109.all_alt.T</td>\n",
       "      <td>569</td>\n",
       "      <td>rs2103665</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11712109</td>\n",
       "      <td>rs115049252</td>\n",
       "      <td>schizo</td>\n",
       "      <td>spleen</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>chr1_11712109</td>\n",
       "      <td>FBXO44</td>\n",
       "      <td>FBXO2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>FBXO2</td>\n",
       "      <td>Whole_Blood</td>\n",
       "      <td>FBXO2</td>\n",
       "      <td>FBXO2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       name      X  Linked_SNP   Chr   Position    Index_SNP  \\\n",
       "0  chr1.111243375.all_alt.G  75004  rs12085135  chr1  111243375   rs12061304   \n",
       "1      chr1.111243375.Ref.C  75004  rs12085135  chr1  111243375   rs12061304   \n",
       "2  chr1.113938499.all_alt.C   1977  rs74395031  chr1  113938499  rs115870915   \n",
       "3      chr1.113938499.Ref.T   1977  rs74395031  chr1  113938499  rs115870915   \n",
       "4   chr1.11712109.all_alt.T    569   rs2103665  chr1   11712109  rs115049252   \n",
       "\n",
       "                                         Disease  \\\n",
       "0                                 panic disorder   \n",
       "1                                 panic disorder   \n",
       "2  type 2 diabetes,Obsessive Compulsive Disorder   \n",
       "3  type 2 diabetes,Obsessive Compulsive Disorder   \n",
       "4                                         schizo   \n",
       "\n",
       "                     DHS_tissue eGene eQTL_tissue  ...         rowname  \\\n",
       "0              t-helper_17_cell                    ...  chr1_111243375   \n",
       "1              t-helper_17_cell                    ...  chr1_111243375   \n",
       "2  astrocyte_of_the_spinal_cord                    ...  chr1_113938499   \n",
       "3  astrocyte_of_the_spinal_cord                    ...  chr1_113938499   \n",
       "4                        spleen                    ...   chr1_11712109   \n",
       "\n",
       "    ngene ngene_haploreg  egene_nerve egene_blood egene_all all_genes  \\\n",
       "0   KCNA3          KCNA3                                                \n",
       "1   KCNA3          KCNA3                                                \n",
       "2   MAGI3          MAGI3                                                \n",
       "3   MAGI3          MAGI3                                                \n",
       "4  FBXO44          FBXO2                                        FBXO2   \n",
       "\n",
       "  all_gtex_tissues unique_egenes_gtex  all_genes_rna_filt  \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "3                                                          \n",
       "4      Whole_Blood              FBXO2               FBXO2  \n",
       "\n",
       "[5 rows x 149 columns]"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_df_gtex.to_csv(os.path.join(save_dir, 'lib_df_gtex_rna_filt.csv'))\n",
    "\n",
    "lib_df_gtex[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "disease # egenes from gtex # egenes from haploreg # nearest genes\n",
      "anxiety 22\n",
      "attent 22\n",
      "autism 22\n",
      "bipolar 22\n",
      "depress 22\n",
      "ocd 22\n",
      "panic 22\n",
      "personality 22\n",
      "schizo 22\n",
      "traum 22\n"
     ]
    }
   ],
   "source": [
    " \n",
    "print('disease','# egenes from gtex','# egenes from haploreg','# nearest genes')\n",
    "for dz in diseases:\n",
    "    dz_df = lib_df_gtex[lib_df_gtex[dz]==1]\n",
    "\n",
    "    print(dz, len(dz_gene_default))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "disease # egenes from gtex # egenes from haploreg # nearest genes\n",
      "anxiety 26 30 28 39\n",
      "attent 122 124 61 53\n",
      "autism 1 1 4 5\n",
      "bipolar 273 256 142 134\n",
      "depress 141 153 96 100\n",
      "ocd 37 43 32 35\n",
      "panic 10 13 17 16\n",
      "personality 23 20 9 14\n",
      "schizo 587 622 392 387\n",
      "traum 20 27 19 22\n"
     ]
    }
   ],
   "source": [
    "###### number of egenes per dz (rna filtered)\n",
    "if not os.path.exists(save_dir+'/egene_haploreg'):\n",
    "    os.makedirs(save_dir+'/egene_haploreg')\n",
    "if not os.path.exists(save_dir+'/egene_gtex'):\n",
    "    os.makedirs(save_dir+'/egene_gtex')\n",
    "if not os.path.exists(save_dir+'/ngene_haploreg'):\n",
    "    os.makedirs(save_dir+'/ngene_haploreg')\n",
    "if not os.path.exists(save_dir+'/annon_gene_default'):\n",
    "    os.makedirs(save_dir+'/annon_gene_default')\n",
    "   \n",
    "\n",
    "def write_geneset(geneset,path):\n",
    "    with open(path, 'w') as f:\n",
    "        for gene in sorted(geneset):\n",
    "            f.write(gene.upper() + '\\n')\n",
    "    \n",
    "print('disease','# egenes from gtex','# egenes from haploreg','# nearest genes')\n",
    "for dz in diseases:\n",
    "    dz_df = lib_df_gtex[lib_df_gtex[dz]==1]\n",
    "    dz_egenes_gtex = sorted(set('|'.join(dz_df.unique_egenes_gtex.unique()).strip('|').split('|')))\n",
    "    dz_egenes_haploreg = sorted(set('|'.join(dz_df.egene_all.unique()).strip('|').split('|')))\n",
    "    dz_ngenes_haploreg = sorted(set('|'.join(dz_df.ngene_haploreg.unique()).strip('|').split('|')))\n",
    "    dz_gene_default = sorted(set('|'.join(dz_df.GeneName.unique()).strip('|').split('|')))\n",
    "    \n",
    "    write_geneset(dz_egenes_gtex,os.path.join(save_dir, 'egene_gtex',dz+'.txt'))\n",
    "    write_geneset(dz_egenes_haploreg,os.path.join(save_dir, 'egene_haploreg',dz+'.txt'))\n",
    "    write_geneset(dz_ngenes_haploreg,os.path.join(save_dir, 'ngene_haploreg',dz+'.txt'))\n",
    "    write_geneset(dz_gene_default,os.path.join(save_dir, 'annon_gene_default',dz+'.txt'))\n",
    "    \n",
    "    print(dz, len(dz_egenes_gtex), len(dz_egenes_haploreg), len(dz_ngenes_haploreg),len(dz_gene_default))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all 824 891 625 641\n"
     ]
    }
   ],
   "source": [
    "all_egenes_gtex = sorted(set('|'.join(lib_df_gtex.unique_egenes_gtex.unique()).strip('|').split('|')))\n",
    "all_egenes_haploreg = sorted(set('|'.join(lib_df_gtex.egene_all.unique()).strip('|').split('|')))\n",
    "all_ngenes_haploreg = sorted(set('|'.join(lib_df_gtex.ngene_haploreg.unique()).strip('|').split('|')))\n",
    "all_annon_gene_default = sorted(set('|'.join(lib_df_gtex.GeneName.unique()).strip('|').split('|')))\n",
    "print('all', len(all_egenes_gtex), len(all_egenes_haploreg), len(all_ngenes_haploreg),len(all_annon_gene_default))\n",
    "write_geneset(all_egenes_gtex,os.path.join(save_dir, 'egene_gtex','all'+'.txt'))\n",
    "write_geneset(all_egenes_haploreg,os.path.join(save_dir, 'egene_haploreg','all'+'.txt'))\n",
    "write_geneset(all_ngenes_haploreg,os.path.join(save_dir, 'ngene_haploreg','all'+'.txt'))\n",
    "write_geneset(all_annon_gene_default,os.path.join(save_dir, 'annon_gene_default','all'+'.txt'))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ##from 2D mpra hit #\n",
    "# disease # egenes from gtex # egenes from haploreg # nearest genes\n",
    "# anxiety 20 51 15\n",
    "# attent 56 67 25\n",
    "# autism 1 1 3\n",
    "# bipolar 161 174 58\n",
    "# depress 81 92 41\n",
    "# ocd 20 57 14\n",
    "# panic 4 3 7\n",
    "# personality 12 37 3\n",
    "# schizo 607 629 165\n",
    "# traum 5 6 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. get cgenes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(save_dir+'/cgene'):\n",
    "    os.makedirs(save_dir+'/cgene')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## get list of snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4440, 4)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/mguo123/opt/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (63,65) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n",
      "/Users/mguo123/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \"\"\"\n"
     ]
    }
   ],
   "source": [
    "snp_list_file = '../../psych_lib_info.csv'\n",
    "snp_list_df = pd.read_csv(snp_list_file,index_col=0)\n",
    "snp_list_bed_df = snp_list_df[['Chr','Position','name']]\n",
    "snp_list_bed_df.columns = ['chr', 'start', 'name']\n",
    "snp_list_bed_df['stop'] = snp_list_bed_df.start + 1\n",
    "snp_list_bed_df  = snp_list_bed_df[['chr','start', 'stop','name']]\n",
    "print(snp_list_bed_df.shape)\n",
    "snp_list_bed_df[:5]\n",
    "snp_list_bed = pybedtools.BedTool.from_dataframe(snp_list_bed_df).sort()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4A get snp to anchors mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SL_D2_sort.bed',\n",
       " 'SLC_D2_sort.bed',\n",
       " 'H9_D28_sort.bed',\n",
       " 'H9_D10_sort.bed',\n",
       " 'H9_D2_sort.bed',\n",
       " 'Astrocytes_sort.bed',\n",
       " 'SL_D0_sort.bed',\n",
       " 'SLC_D0_sort.bed',\n",
       " 'H9_D0_sort.bed']"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anchors_dir = '/Users/mguo123/Documents/pan_omics_psych/data/interim/merged/anchors_bed_sort/'\n",
    "os.listdir(anchors_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SL_D2 mpra_enrichment/cgene/SL_D2_snp_anchor_annon.bed\n",
      "SLC_D2 mpra_enrichment/cgene/SLC_D2_snp_anchor_annon.bed\n",
      "H9_D28 mpra_enrichment/cgene/H9_D28_snp_anchor_annon.bed\n",
      "H9_D10 mpra_enrichment/cgene/H9_D10_snp_anchor_annon.bed\n",
      "H9_D2 mpra_enrichment/cgene/H9_D2_snp_anchor_annon.bed\n",
      "Astrocytes mpra_enrichment/cgene/Astrocytes_snp_anchor_annon.bed\n",
      "SL_D0 mpra_enrichment/cgene/SL_D0_snp_anchor_annon.bed\n",
      "SLC_D0 mpra_enrichment/cgene/SLC_D0_snp_anchor_annon.bed\n",
      "H9_D0 mpra_enrichment/cgene/H9_D0_snp_anchor_annon.bed\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir(anchors_dir):\n",
    "    tissue = file.split('_sort')[0]\n",
    "    output_bedfile = os.path.join(save_dir,'cgene', tissue + '_snp_anchor_annon.bed')\n",
    "    print(tissue, output_bedfile)\n",
    "    db_bed = pybedtools.BedTool(os.path.join(anchors_dir, tissue+'_sort.bed'))\n",
    "    output_bed = snp_list_bed.intersect(db_bed, wo=True, sorted=True, names='db')\n",
    "    output_bed.saveas(output_bedfile)\n",
    "    output_bed.to_dataframe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4B get cgene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissue_mapping = {'Astrocytes':'AST1',\n",
    "    'H9_D0':'H9D0','H9_D2':'H9D2','H9_D4':'H9D4','H9_D10':'H9D10','H9_D28':'H9D28',\n",
    "    'SL_D0':'SL','SLC_D0':'SLC', 'SL_D2':'SL_D2','SLC_D2':'SLC_D2', \n",
    "}\n",
    "tissues = sorted(tissue_mapping.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "loop_dir = '/Users/mguo123/Documents/pan_omics_psych/data/interim/merged/loops'\n",
    "os.listdir(loop_dir)\n",
    "def make_loop_df(loop_file):\n",
    "    \"\"\"\n",
    "    from hichip_df make hichip df that's bidirectionary target <-> source\n",
    "\n",
    "\n",
    "    Arguments\n",
    "        loop_file : <str> like '../data/interim/merged/loops/GDSD3.loops.csv'\n",
    "\n",
    "    return: loop_df_bi <DataFrame> with colummns ['target','source','count']\n",
    "\n",
    "    \"\"\"\n",
    "    # loop_file = '../data/interim/merged/loops/GDSD3.loops.csv'\n",
    "    loop_df = pd.read_csv(loop_file,index_col=0)\n",
    "    loop_df_rev = loop_df.copy()\n",
    "    loop_df_rev.columns = ['target','source','count']\n",
    "    loop_df_bi = pd.concat([loop_df, loop_df_rev],sort=False)\n",
    "    return loop_df_bi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_cgenes(tissue):\n",
    "    print(tissue)\n",
    "    anchor_promoter_file = '/Users/mguo123/Documents/pan_omics_psych/data/interim/annon/promoter_anchors/promoter_'+tissue + '_annon.bed'\n",
    "    if not os.path.exists(anchor_promoter_file):\n",
    "        print('anchor_promoter_file not exists', anchor_promoter_file)\n",
    "        return\n",
    "        \n",
    "    anchor_promoter_annon_df =  pybedtools.BedTool(anchor_promoter_file).to_dataframe()\n",
    "    anchor_promoter_annon_df.columns = ['chr_p','start_p','stop_p','TSS','chr_f','start_f','stop_f','anchor','overlap']\n",
    "\n",
    "    anchor_annon_df_filt = anchor_promoter_annon_df.sort_values('overlap', ascending=False).drop_duplicates(['TSS','anchor']).sort_index()\n",
    "    anchor_annon_df_filt = anchor_annon_df_filt[['anchor','TSS']]\n",
    "    \n",
    "    \n",
    "    snp_annon_file = os.path.join(save_dir,'cgene',tissue + '_snp_anchor_annon.bed')\n",
    "    snp_annon_df =  pybedtools.BedTool(snp_annon_file).to_dataframe()\n",
    "    snp_annon_df = snp_annon_df[['thickEnd', 'name']]\n",
    "    snp_annon_df.columns = ['p_loc', 'name']\n",
    "    \n",
    "    snp_df_anchor = snp_annon_df.merge(anchor_annon_df_filt,how='inner', left_on='p_loc',right_on='anchor')\n",
    "    \n",
    "    \n",
    "    loop_file = os.path.join(loop_dir, tissue+'.loops.csv')\n",
    "    # read in loops, then \"annotate\" the source TSS regions the loop anchors\n",
    "    loop_df_bi = make_loop_df(loop_file)\n",
    "    loop_df_bi = loop_df_bi.merge(anchor_annon_df_filt,how='inner',left_on='source',right_on='anchor' )\n",
    "\n",
    "    snp_df_loop = snp_annon_df.merge(loop_df_bi,how='inner', left_on='p_loc',right_on='target')\n",
    "\n",
    "    snp_anchor_info = snp_df_anchor.groupby('name').agg({'TSS':[lambda x: '|'.join(x), 'count']})\n",
    "    snp_anchor_info.columns = ['cgenes_anchor', 'cgene_count_anchor']\n",
    "    snp_anchor_info.reset_index(inplace=True)\n",
    "    snp_loop_info = snp_df_loop.groupby('name').agg({'TSS':[lambda x: '|'.join(x), 'count']})\n",
    "    snp_loop_info.columns = ['cgenes_loop', 'cgene_count_loop']\n",
    "    snp_loop_info.reset_index(inplace=True)\n",
    "    snp_info_tissue = snp_anchor_info.merge(snp_loop_info, how='outer',on='name')\n",
    "    snp_info_tissue.cgenes_anchor.fillna('',inplace=True)\n",
    "    snp_info_tissue.cgene_count_anchor.fillna(0,inplace=True)\n",
    "    snp_info_tissue.cgenes_loop.fillna('',inplace=True)\n",
    "    snp_info_tissue.cgene_count_loop.fillna(0,inplace=True)\n",
    "    snp_info_tissue.to_csv(os.path.join(save_dir,'cgene', tissue+'_snp_cgene_info.csv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Astrocytes\n",
      "H9_D0\n",
      "H9_D10\n",
      "H9_D2\n",
      "H9_D28\n",
      "H9_D4\n",
      "anchor_promoter_file not exists /Users/mguo123/Documents/pan_omics_psych/data/interim/annon/promoter_anchors/promoter_H9_D4_annon.bed\n",
      "SLC_D0\n",
      "SLC_D2\n",
      "SL_D0\n",
      "SL_D2\n",
      "CPU times: user 1min 16s, sys: 2.9 s, total: 1min 19s\n",
      "Wall time: 32.8 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for tissue in tissues: \n",
    "    get_cgenes(tissue)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NOT DONE: BUT TRY TO GET THE cgenes and egenes that match in the overall table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['mpra_enrichment/cgene/SL_D2_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/H9_D0_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/Astrocytes_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/SLC_D2_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/H9_D2_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/SL_D0_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/SLC_D0_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/H9_D28_snp_cgene_info.csv',\n",
       " 'mpra_enrichment/cgene/H9_D10_snp_cgene_info.csv']"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "glob.glob(os.path.join(save_dir, 'cgene/*csv'))\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissue_mapper = {\n",
    "'H9_D0': 'H9D0',\n",
    "'Astrocytes' :'AST1',\n",
    "'H9_D2' :'H9D2',\n",
    "'SL_D0' :'SL',\n",
    "'SL_D2' :'SLD2',\n",
    "'SLC_D0': 'SLC',\n",
    "'SLC_D2': 'SLCD2',\n",
    "'H9_D10': 'H9D10',\n",
    "'H9_D28': 'H9D28'\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['name', 'X', 'Linked_SNP', 'Chr', 'Position', 'Index_SNP', 'Disease',\n",
       "       'DHS_tissue', 'eGene', 'eQTL_tissue',\n",
       "       ...\n",
       "       'rowname', 'ngene', 'ngene_haploreg', 'egene_nerve', 'egene_blood',\n",
       "       'egene_all', 'all_genes', 'all_gtex_tissues', 'unique_egenes_gtex',\n",
       "       'all_genes_rna_filt'],\n",
       "      dtype='object', length=149)"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_df_gtex.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "def agree_cgene_egene(row,col_to_check='unique_egenes_gtex'):\n",
    "    egene_set = set(row[col_to_check].split('|'))\n",
    "    try:\n",
    "        cgene_set = set(row.cgenes_anchor.split('|') + row.cgenes_loop.split('|'))\n",
    "        agree_set = sorted(egene_set.intersection(cgene_set))\n",
    "        return agree_set\n",
    "    except:\n",
    "        return []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rowname</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>ngene</th>\n",
       "      <th>anxiety</th>\n",
       "      <th>attent</th>\n",
       "      <th>autism</th>\n",
       "      <th>bipolar</th>\n",
       "      <th>depress</th>\n",
       "      <th>ocd</th>\n",
       "      <th>panic</th>\n",
       "      <th>...</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>Index_SNP</th>\n",
       "      <th>ngene_haploreg</th>\n",
       "      <th>egene_nerve</th>\n",
       "      <th>egene_blood</th>\n",
       "      <th>egene_all</th>\n",
       "      <th>all_genes</th>\n",
       "      <th>all_gtex_tissues</th>\n",
       "      <th>unique_egenes_gtex</th>\n",
       "      <th>all_genes_rna_filt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr1_111243375</td>\n",
       "      <td></td>\n",
       "      <td>KCNA3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>rs12085135</td>\n",
       "      <td>rs12061304</td>\n",
       "      <td>KCNA3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1_111243375</td>\n",
       "      <td></td>\n",
       "      <td>KCNA3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>rs12085135</td>\n",
       "      <td>rs12061304</td>\n",
       "      <td>KCNA3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1_113938499</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs74395031</td>\n",
       "      <td>rs115870915</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1_113938499</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs74395031</td>\n",
       "      <td>rs115870915</td>\n",
       "      <td>MAGI3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1_11712109</td>\n",
       "      <td>FBXO44</td>\n",
       "      <td>FBXO44</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs2103665</td>\n",
       "      <td>rs115049252</td>\n",
       "      <td>FBXO2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>FBXO2|FBXO2</td>\n",
       "      <td>Whole_Blood|Whole_Blood</td>\n",
       "      <td>FBXO2</td>\n",
       "      <td>FBXO2|FBXO2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4435</td>\n",
       "      <td>chr9_98824232</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs10481697</td>\n",
       "      <td>rs7867155</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4436</td>\n",
       "      <td>chr9_98824252</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs10481698</td>\n",
       "      <td>rs7867155</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4437</td>\n",
       "      <td>chr9_98824252</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs10481698</td>\n",
       "      <td>rs7867155</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4438</td>\n",
       "      <td>chr9_98824656</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs12340544</td>\n",
       "      <td>rs7867155</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4439</td>\n",
       "      <td>chr9_98824656</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>rs12340544</td>\n",
       "      <td>rs7867155</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4440 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             rowname GeneName   ngene  anxiety  attent  autism  bipolar  \\\n",
       "0     chr1_111243375            KCNA3        0       0       0        0   \n",
       "1     chr1_111243375            KCNA3        0       0       0        0   \n",
       "2     chr1_113938499    MAGI3   MAGI3        0       0       0        0   \n",
       "3     chr1_113938499    MAGI3   MAGI3        0       0       0        0   \n",
       "4      chr1_11712109   FBXO44  FBXO44        0       0       0        0   \n",
       "...              ...      ...     ...      ...     ...     ...      ...   \n",
       "4435   chr9_98824232                         1       0       0        0   \n",
       "4436   chr9_98824252                         1       0       0        0   \n",
       "4437   chr9_98824252                         1       0       0        0   \n",
       "4438   chr9_98824656                         1       0       0        0   \n",
       "4439   chr9_98824656                         1       0       0        0   \n",
       "\n",
       "      depress  ocd  panic  ...  Linked_SNP    Index_SNP  ngene_haploreg  \\\n",
       "0           0    0      1  ...  rs12085135   rs12061304           KCNA3   \n",
       "1           0    0      1  ...  rs12085135   rs12061304           KCNA3   \n",
       "2           0    1      0  ...  rs74395031  rs115870915           MAGI3   \n",
       "3           0    1      0  ...  rs74395031  rs115870915           MAGI3   \n",
       "4           0    0      0  ...   rs2103665  rs115049252           FBXO2   \n",
       "...       ...  ...    ...  ...         ...          ...             ...   \n",
       "4435        0    0      0  ...  rs10481697    rs7867155                   \n",
       "4436        0    0      0  ...  rs10481698    rs7867155                   \n",
       "4437        0    0      0  ...  rs10481698    rs7867155                   \n",
       "4438        0    0      0  ...  rs12340544    rs7867155                   \n",
       "4439        0    0      0  ...  rs12340544    rs7867155                   \n",
       "\n",
       "     egene_nerve egene_blood egene_all    all_genes         all_gtex_tissues  \\\n",
       "0                                                                              \n",
       "1                                                                              \n",
       "2                                                                              \n",
       "3                                                                              \n",
       "4                                       FBXO2|FBXO2  Whole_Blood|Whole_Blood   \n",
       "...          ...         ...       ...          ...                      ...   \n",
       "4435                                                                           \n",
       "4436                                                                           \n",
       "4437                                                                           \n",
       "4438                                                                           \n",
       "4439                                                                           \n",
       "\n",
       "     unique_egenes_gtex all_genes_rna_filt  \n",
       "0                                           \n",
       "1                                           \n",
       "2                                           \n",
       "3                                           \n",
       "4                 FBXO2        FBXO2|FBXO2  \n",
       "...                 ...                ...  \n",
       "4435                                        \n",
       "4436                                        \n",
       "4437                                        \n",
       "4438                                        \n",
       "4439                                        \n",
       "\n",
       "[4440 rows x 23 columns]"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lib_df_gtex[['rowname', 'GeneName', 'ngene', 'anxiety', 'attent', 'autism',\n",
    "       'bipolar', 'depress', 'ocd', 'panic', 'personality', 'schizo', 'traum',\n",
    "        'Linked_SNP', 'Index_SNP', 'ngene_haploreg', 'egene_nerve',\n",
    "       'egene_blood', 'egene_all', 'all_genes', 'all_gtex_tissues',\n",
    "       'unique_egenes_gtex', 'all_genes_rna_filt']]# 'tissue','num_dz'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NOT DONE AFTER THIS POINT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SL_D2 SLD2\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'DataFrame' object has no attribute 'tissue'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-169-0496d0513726>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;31m# get cgenes that are\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m     \u001b[0msnp_all_results_tissue\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlib_df_gtex\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mlib_df_gtex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtissue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcontains\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_str\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msnp_all_results\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msnp_all_results_tissue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0msnp_all_results_tissue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m   5177\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_info_axis\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_can_hold_identifiers_and_holds_name\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5178\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5179\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5180\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5181\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'tissue'"
     ]
    }
   ],
   "source": [
    "snp_all_results_cgene_egene = pd.DataFrame()\n",
    "# dataframe to store snps by tissue that agree by tissue specific cgene and egene\n",
    "for cgene_file in glob.glob(os.path.join(save_dir, 'cgene/*csv')):\n",
    "    tissue = os.path.basename(cgene_file).split('_snp')[0]\n",
    "    tissue_str = tissue_mapper[tissue]\n",
    "    print(tissue, tissue_str)\n",
    "    \n",
    "    # get cgenes that are \n",
    "    snp_all_results_tissue = lib_df_gtex[lib_df_gtex.tissue.str.contains(tissue_str)]\n",
    "    print(snp_all_results.shape, snp_all_results_tissue.shape)\n",
    "    if snp_all_results_tissue.shape[0]==0:\n",
    "        continue\n",
    "        \n",
    "    # get cgene info\n",
    "    cgene_df = pd.read_csv(cgene_file,index_col=0)\n",
    "    cgene_df['rowname'] = cgene_df.name.apply(lambda x: '_'.join(x.split('.')[:2]))\n",
    "    cgene_df.drop('name',axis=1,inplace=True)\n",
    "    snp_all_results_tissue = snp_all_results_tissue.merge(cgene_df, how='inner',on='rowname')\n",
    "    \n",
    "    # get agreement set of genes that \n",
    "    agree_set = snp_all_results_tissue.apply(lambda x: agree_cgene_egene(x, col_to_check='unique_egenes_gtex'),axis=1)\n",
    "    snp_all_results_tissue['cgene_egene_agree'] = agree_set.apply(lambda x: '|'.join(x))\n",
    "    snp_all_results_tissue['cgene_egene_agree_count'] = agree_set.apply(len)\n",
    "    snp_all_results_tissue = snp_all_results_tissue[snp_all_results_tissue.cgene_egene_agree_count>0]\n",
    "    snp_all_results_tissue['tissue_cgene_egene'] = tissue_str\n",
    "    \n",
    "    snp_all_results_cgene_egene_tissue = pd.concat([snp_all_results_cgene_egene_tissue, snp_all_results_tissue],axis=0)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_all_results_cgene_egene_tissue.to_csv('D_mpraanalyze_barcode_allelic/snp_all_results_cgene_egene_tissue_gtex.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(235, 32)"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_all_results_cgene_egene_tissue.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_str(str_arr):\n",
    "    all_genes = []\n",
    "    for str_x in str_arr:\n",
    "        all_genes+=str_x.split('|')\n",
    "    return '|'.join(sorted(set(all_genes)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(61, 19)"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_all_results_cgene_egene = snp_all_results_cgene_egene_tissue.groupby(['rowname', 'GeneName', 'ngene', 'tissue', 'anxiety', 'attent', 'autism',\n",
    "       'bipolar', 'depress', 'ocd', 'panic', 'personality', 'schizo', 'traum',\n",
    "       'num_dz', 'Linked_SNP', 'Index_SNP']).agg({\n",
    "    'cgene_egene_agree':merge_str\n",
    "}).reset_index()\n",
    "snp_all_results_cgene_egene['cgene_egene_agree_num'] = snp_all_results_cgene_egene.cgene_egene_agree.apply(lambda x: len(x.split('|')))\n",
    "snp_all_results_cgene_egene.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_all_results_cgene_egene.to_csv('D_mpraanalyze_barcode_allelic/snp_all_results_cgene_egene.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rowname</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>ngene</th>\n",
       "      <th>tissue</th>\n",
       "      <th>anxiety</th>\n",
       "      <th>attent</th>\n",
       "      <th>autism</th>\n",
       "      <th>bipolar</th>\n",
       "      <th>depress</th>\n",
       "      <th>ocd</th>\n",
       "      <th>panic</th>\n",
       "      <th>personality</th>\n",
       "      <th>schizo</th>\n",
       "      <th>traum</th>\n",
       "      <th>num_dz</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>Index_SNP</th>\n",
       "      <th>cgene_egene_agree</th>\n",
       "      <th>cgene_egene_agree_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr10_104598322</td>\n",
       "      <td>CYP17A1-AS1</td>\n",
       "      <td></td>\n",
       "      <td>H9D4|H9D0|H9D10|HEK293</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>rs10786713</td>\n",
       "      <td>rs7085104,rs11191424,rs11191419</td>\n",
       "      <td>ARL3|AS3MT|SFXN2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr10_104941112</td>\n",
       "      <td></td>\n",
       "      <td>NT5C2</td>\n",
       "      <td>H9D4|SLC|H9D2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>rs35525740</td>\n",
       "      <td>rs7914558</td>\n",
       "      <td>ARL3|AS3MT|CALHM2|CNNM2</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr10_82156982</td>\n",
       "      <td></td>\n",
       "      <td>FAM213A</td>\n",
       "      <td>AST1|H9D2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs12414691</td>\n",
       "      <td>rs7074356</td>\n",
       "      <td>TSPAN14</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr11_47267079</td>\n",
       "      <td>ACP2</td>\n",
       "      <td>NR1H3</td>\n",
       "      <td>SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs10838677</td>\n",
       "      <td></td>\n",
       "      <td>MADD</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11_47360053</td>\n",
       "      <td>MYBPC3</td>\n",
       "      <td></td>\n",
       "      <td>SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs3729948</td>\n",
       "      <td></td>\n",
       "      <td>MADD</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>chr6_28219661</td>\n",
       "      <td>ZKSCAN4</td>\n",
       "      <td>ZKSCAN4</td>\n",
       "      <td>H9D0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs9986596</td>\n",
       "      <td>rs853679</td>\n",
       "      <td>PGBD1|ZKSCAN3|ZKSCAN4|ZKSCAN8|ZNF165|ZSCAN31</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57</td>\n",
       "      <td>chr6_28234597</td>\n",
       "      <td>RP5-874C20.3</td>\n",
       "      <td></td>\n",
       "      <td>H9D4|SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs2799077</td>\n",
       "      <td>rs853679</td>\n",
       "      <td>PGBD1|ZKSCAN3|ZKSCAN4|ZSCAN23|ZSCAN31</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>chr6_28235176</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>HEK293|SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs2799079</td>\n",
       "      <td>rs853679</td>\n",
       "      <td>PGBD1|ZKSCAN3|ZKSCAN4|ZKSCAN8|ZSCAN23|ZSCAN31</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>chr6_28304384</td>\n",
       "      <td></td>\n",
       "      <td>ZKSCAN3</td>\n",
       "      <td>H9D4|SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs67340775</td>\n",
       "      <td>rs13197574,rs13213152,rs13217619,rs13214023</td>\n",
       "      <td>PGBD1|ZKSCAN3|ZKSCAN4|ZKSCAN8|ZNF165|ZSCAN31|Z...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>chr6_31319525</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>H9D4|SLC|HEK293|H9D10|H9D2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs9266066</td>\n",
       "      <td>rs2596500</td>\n",
       "      <td>C4A|CCHCR1|FLOT1|HLA-C|IER3|LY6G5B|MICB|MSH5|N...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            rowname      GeneName    ngene                      tissue  \\\n",
       "0   chr10_104598322   CYP17A1-AS1               H9D4|H9D0|H9D10|HEK293   \n",
       "1   chr10_104941112                  NT5C2               H9D4|SLC|H9D2   \n",
       "2    chr10_82156982                FAM213A                   AST1|H9D2   \n",
       "3    chr11_47267079          ACP2    NR1H3                         SLC   \n",
       "4    chr11_47360053        MYBPC3                                  SLC   \n",
       "..              ...           ...      ...                         ...   \n",
       "56    chr6_28219661       ZKSCAN4  ZKSCAN4                        H9D0   \n",
       "57    chr6_28234597  RP5-874C20.3                             H9D4|SLC   \n",
       "58    chr6_28235176                                         HEK293|SLC   \n",
       "59    chr6_28304384                ZKSCAN3                    H9D4|SLC   \n",
       "60    chr6_31319525         HLA-B    HLA-B  H9D4|SLC|HEK293|H9D10|H9D2   \n",
       "\n",
       "    anxiety  attent  autism  bipolar  depress  ocd  panic  personality  \\\n",
       "0         0       0       0        1        0    0      0            0   \n",
       "1         0       1       0        1        0    0      0            0   \n",
       "2         0       0       0        0        0    0      0            1   \n",
       "3         0       0       0        0        0    0      0            0   \n",
       "4         0       0       0        0        0    0      0            0   \n",
       "..      ...     ...     ...      ...      ...  ...    ...          ...   \n",
       "56        0       0       0        0        1    0      0            0   \n",
       "57        0       0       0        0        1    0      0            0   \n",
       "58        0       0       0        0        1    0      0            0   \n",
       "59        0       0       0        0        0    0      0            0   \n",
       "60        0       0       0        0        0    0      0            0   \n",
       "\n",
       "    schizo  traum  num_dz  Linked_SNP  \\\n",
       "0        1      0       2  rs10786713   \n",
       "1        1      0       3  rs35525740   \n",
       "2        0      0       1  rs12414691   \n",
       "3        1      0       1  rs10838677   \n",
       "4        1      0       1   rs3729948   \n",
       "..     ...    ...     ...         ...   \n",
       "56       0      0       1   rs9986596   \n",
       "57       0      0       1   rs2799077   \n",
       "58       0      0       1   rs2799079   \n",
       "59       1      0       1  rs67340775   \n",
       "60       1      0       1   rs9266066   \n",
       "\n",
       "                                      Index_SNP  \\\n",
       "0               rs7085104,rs11191424,rs11191419   \n",
       "1                                     rs7914558   \n",
       "2                                     rs7074356   \n",
       "3                                                 \n",
       "4                                                 \n",
       "..                                          ...   \n",
       "56                                     rs853679   \n",
       "57                                     rs853679   \n",
       "58                                     rs853679   \n",
       "59  rs13197574,rs13213152,rs13217619,rs13214023   \n",
       "60                                    rs2596500   \n",
       "\n",
       "                                    cgene_egene_agree  cgene_egene_agree_num  \n",
       "0                                    ARL3|AS3MT|SFXN2                      3  \n",
       "1                             ARL3|AS3MT|CALHM2|CNNM2                      4  \n",
       "2                                             TSPAN14                      1  \n",
       "3                                                MADD                      1  \n",
       "4                                                MADD                      1  \n",
       "..                                                ...                    ...  \n",
       "56       PGBD1|ZKSCAN3|ZKSCAN4|ZKSCAN8|ZNF165|ZSCAN31                      6  \n",
       "57              PGBD1|ZKSCAN3|ZKSCAN4|ZSCAN23|ZSCAN31                      5  \n",
       "58      PGBD1|ZKSCAN3|ZKSCAN4|ZKSCAN8|ZSCAN23|ZSCAN31                      6  \n",
       "59  PGBD1|ZKSCAN3|ZKSCAN4|ZKSCAN8|ZNF165|ZSCAN31|Z...                      7  \n",
       "60  C4A|CCHCR1|FLOT1|HLA-C|IER3|LY6G5B|MICB|MSH5|N...                     11  \n",
       "\n",
       "[61 rows x 19 columns]"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_all_results_cgene_egene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "disease # egenes from gtex # egenes from haploreg # nearest genes\n",
      "anxiety 9\n",
      "attent 4\n",
      "autism 1\n",
      "bipolar 26\n",
      "depress 7\n",
      "ocd 14\n",
      "panic 1\n",
      "personality 8\n",
      "schizo 104\n",
      "traum 1\n"
     ]
    }
   ],
   "source": [
    "if not os.path.exists('D_mpraanalyze_barcode_allelic/egene_cgene_agree'):\n",
    "    os.makedirs('D_mpraanalyze_barcode_allelic/egene_cgene_agree')\n",
    "\n",
    "print('disease','# egenes from gtex','# egenes from haploreg','# nearest genes')\n",
    "for dz in diseases:\n",
    "    dz_df = snp_all_results_cgene_egene[snp_all_results_cgene_egene[dz]==1]\n",
    "    dz_egenes = sorted(set('|'.join(dz_df.cgene_egene_agree.unique()).strip('|').split('|')))\n",
    "\n",
    "    write_geneset(dz_egenes,os.path.join('D_mpraanalyze_barcode_allelic/egene_cgene_agree',dz+'.txt'))\n",
    "\n",
    "    print(dz, len(dz_egenes))#, len(dz_egenes_haploreg), len(dz_ngenes_haploreg))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>cgene_egene_agree</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rowname</th>\n",
       "      <th>tissue</th>\n",
       "      <th>anxiety</th>\n",
       "      <th>attent</th>\n",
       "      <th>autism</th>\n",
       "      <th>bipolar</th>\n",
       "      <th>depress</th>\n",
       "      <th>ocd</th>\n",
       "      <th>panic</th>\n",
       "      <th>personality</th>\n",
       "      <th>schizo</th>\n",
       "      <th>traum</th>\n",
       "      <th>num_dz</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>Index_SNP</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>chr10_104598322</td>\n",
       "      <td>H9D4|H9D0|H9D10|HEK293</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>rs10786713</td>\n",
       "      <td>rs7085104,rs11191424,rs11191419</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr10_104941112</td>\n",
       "      <td>H9D4|SLC|H9D2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>rs35525740</td>\n",
       "      <td>rs7914558</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr10_82156982</td>\n",
       "      <td>AST1|H9D2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs12414691</td>\n",
       "      <td>rs7074356</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr11_47267079</td>\n",
       "      <td>SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs10838677</td>\n",
       "      <td></td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr11_47360053</td>\n",
       "      <td>SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs3729948</td>\n",
       "      <td></td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr6_28219661</td>\n",
       "      <td>H9D0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs9986596</td>\n",
       "      <td>rs853679</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr6_28234597</td>\n",
       "      <td>H9D4|SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs2799077</td>\n",
       "      <td>rs853679</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr6_28235176</td>\n",
       "      <td>HEK293|SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs2799079</td>\n",
       "      <td>rs853679</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr6_28304384</td>\n",
       "      <td>H9D4|SLC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs67340775</td>\n",
       "      <td>rs13197574,rs13213152,rs13217619,rs13214023</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>chr6_31319525</td>\n",
       "      <td>H9D4|SLC|HEK293|H9D10|H9D2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>rs9266066</td>\n",
       "      <td>rs2596500</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                   cgene_egene_agree\n",
       "rowname         tissue                     anxiety attent autism bipolar depress ocd panic personality schizo traum num_dz Linked_SNP Index_SNP                                                     \n",
       "chr10_104598322 H9D4|H9D0|H9D10|HEK293     0       0      0      1       0       0   0     0           1      0     2      rs10786713 rs7085104,rs11191424,rs11191419                              4\n",
       "chr10_104941112 H9D4|SLC|H9D2              0       1      0      1       0       0   0     0           1      0     3      rs35525740 rs7914558                                                    2\n",
       "chr10_82156982  AST1|H9D2                  0       0      0      0       0       0   0     1           0      0     1      rs12414691 rs7074356                                                    2\n",
       "chr11_47267079  SLC                        0       0      0      0       0       0   0     0           1      0     1      rs10838677                                                              4\n",
       "chr11_47360053  SLC                        0       0      0      0       0       0   0     0           1      0     1      rs3729948                                                               4\n",
       "...                                                                                                                                                                                              ...\n",
       "chr6_28219661   H9D0                       0       0      0      0       1       0   0     0           0      0     1      rs9986596  rs853679                                                     2\n",
       "chr6_28234597   H9D4|SLC                   0       0      0      0       1       0   0     0           0      0     1      rs2799077  rs853679                                                     4\n",
       "chr6_28235176   HEK293|SLC                 0       0      0      0       1       0   0     0           0      0     1      rs2799079  rs853679                                                     4\n",
       "chr6_28304384   H9D4|SLC                   0       0      0      0       0       0   0     0           1      0     1      rs67340775 rs13197574,rs13213152,rs13217619,rs13214023                  4\n",
       "chr6_31319525   H9D4|SLC|HEK293|H9D10|H9D2 0       0      0      0       0       0   0     0           1      0     1      rs9266066  rs2596500                                                    2\n",
       "\n",
       "[61 rows x 1 columns]"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_all_results_cgene_egene_tissue.groupby(['rowname', 'tissue', 'anxiety', 'attent', 'autism',\n",
    "       'bipolar', 'depress', 'ocd', 'panic', 'personality', 'schizo', 'traum',\n",
    "       'num_dz', 'Linked_SNP', 'Index_SNP']).agg({\n",
    "    'cgene_egene_agree':'count'\n",
    "})\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# def test hits from PMID: 30737407\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "hits_str = '''\n",
    "rs10795\n",
    "rs12136320\n",
    "rs12912934\n",
    "\n",
    "DNM1P51\n",
    "rs16937\n",
    "\n",
    "rs1801311\n",
    "\n",
    "rs223390\n",
    "rs2269524\n",
    "\n",
    "rs2270363\n",
    "\n",
    "CDIP1\n",
    "rs2304204\n",
    "rs2385395\n",
    "rs2711116\n",
    "rs281759\n",
    "\n",
    "rs2856268\n",
    "\n",
    "rs2974999\n",
    "rs340836\n",
    "rs3814880\n",
    "rs3822346\n",
    "\n",
    "PCDHA8\n",
    "PCDHA2\n",
    "PCDHA7\n",
    "rs4786494\n",
    "rs5751195\n",
    "\n",
    "rs6002621\n",
    "\n",
    "rs61660810\n",
    "rs6992091\n",
    "rs7014953\n",
    "rs72748702\n",
    "\n",
    "GOLGA2P7\n",
    "rs778593\n",
    "\n",
    "PCDHA8\n",
    "ZMAT2\n",
    "WDR55\n",
    "rs78532287\n",
    "\n",
    "rs796364\n",
    "\n",
    "rs8135801\n",
    "\n",
    "rs9362397'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hits =  [x for x in hits_str.split('\\n') if x.startswith('rs')]\n",
    "len(hits)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs3822346\n",
      "rs78532287\n"
     ]
    }
   ],
   "source": [
    "for hit in hits:\n",
    "    if hit in list(lib_df.Linked_SNP):\n",
    "        print(hit)\n",
    "    index_snp_df =  lib_df[lib_df.Index_SNP.fillna('').str.contains(hit)]\n",
    "    if index_snp_df.shape[0]>0:\n",
    "        print(hit)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
