{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 7C_drug_gene_tables\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Paul,\n",
    "We have a good group of key IMA-associated faculty lined up for your presentation tomorrow. Look forward to a productive discussion on the therapeutic implications of your large-scale non-coding mutational analysis.\n",
    "\n",
    "\n",
    "To facilitate discussion (and only if it is convenient for you), it would be helpful if you could organize your presentation along the following lines:\n",
    "- Background\n",
    "- Key findings (esp. the identification of ~1500 druggable targets across ~50 diseases)\n",
    "- Implications of your findings for drug repurposing\n",
    "- Implications of your findings for drug discovery\n",
    "\n",
    "In particular, I suggest separate discussions on 3 and 4 above, because the f/u will be different in the two cases (both relevant to the IMA).\n",
    "Thank you,\n",
    "Chaitan\n",
    "\n",
    "\n",
    "want list of gene names with \n",
    "- gene name\n",
    "- gene evidence (MPRA eQTL,  coding, rare/mendelian)\n",
    "- directionality of linked SNP (loss or gain)\n",
    "- class of gene (i.e. GPCR)\n",
    "- tissue distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os, glob"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get drug data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['rowname', 'Linked_SNP', 'Chr', 'Position', 'Ref/Alt', 'Index_SNP',\n",
       "       'ConsScore', 'ConsDetail', 'diseases', 'num_dz', 'gwas_pval', 'source',\n",
       "       'gene_annon', 'ngene_haploreg', 'egene_alltissue_gtex',\n",
       "       'egene_neurotissue_gtex', 'egene_neurotissue_mapgene_gtex',\n",
       "       'egene_neurotissue_maptissue_gtex', 'atac_tissues', 'bool_in_atac_pk',\n",
       "       'hichip_tissues', 'bool_in_hichip_pk', 'broken_motifs', 'gained_motifs',\n",
       "       'drug_target_gene', 'drug_target_pertname', 'drug_target_moa',\n",
       "       'drug_target_clinical_phase', 'drug_target_indication',\n",
       "       'drug_cmap_gene_up', 'drug_cmap_pertname_up', 'drug_cmap_gene_dn',\n",
       "       'drug_cmap_pertname_dn', 'tissue_asatac', 'bool_is_asatac',\n",
       "       'tissue_ashichip', 'bool_is_ashichip', 'mpra_tissue', 'mpra_pval_str',\n",
       "       'mpra_pval_mean', 'mpra_logfc_str', 'mpra_logfc_mean', 'bool_mpra_sig',\n",
       "       'cgene_egene_agree', 'cgene_egene_agree_num', 'bool_cgene_egene_agree',\n",
       "       'ukbb_drug', 'ukbb_phenotype', 'ukbb_phenotype_beta', 'SNP_id',\n",
       "       'psychencode_egenes', 'egenes_all_sources', 'abc_genes',\n",
       "       'abc_gene_tissues'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_df =  pd.read_csv('figure_tables/mpra_res_df_UPDATED_ABC.csv',index_col=0).fillna('')\n",
    "\n",
    "data_df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.DataFrame({'temp_c': [17.0, 25.0]},\n",
    "#                   index=['Portland', 'Berkeley'])\n",
    "# df.assign(temp_f=lambda x: x.temp_c * 9 / 5 + 32)\n",
    "\n",
    "def explore_genelist_disease(column, df = data_df):\n",
    "    df_sel = (df[['diseases',column, 'mpra_logfc_mean']]#,'drug_target_pertname']]\n",
    "     .assign(diseases= lambda x: x.diseases.str.split('|'))\n",
    "     .explode('diseases')\n",
    "     .assign(gene= lambda x: x[column].apply(lambda x: sorted(set(x.split('|')))\n",
    "                                                                        ))\n",
    "     .explode('gene')\n",
    "     .query('gene!=\"\"')\n",
    "     .drop(column, axis=1)\n",
    "     .groupby('gene')\n",
    "        .agg({'diseases': lambda arr: '|'.join(sorted(set(arr))),\n",
    "              'mpra_logfc_mean': 'mean'})\n",
    "     .reset_index()\n",
    "    )\n",
    "    return df_sel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>diseases</th>\n",
       "      <th>mpra_logfc_mean</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ADH1C</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-0.794781</td>\n",
       "      <td>direct_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>CHRNA2</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>0.665546</td>\n",
       "      <td>direct_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>BPD|MDD|SCZ</td>\n",
       "      <td>-0.383639</td>\n",
       "      <td>direct_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>DHODH</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-1.038550</td>\n",
       "      <td>direct_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ITPR3</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>0.608879</td>\n",
       "      <td>direct_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.927891</td>\n",
       "      <td>direct_target</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>SV2A</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>-0.555829</td>\n",
       "      <td>direct_target</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene     diseases  mpra_logfc_mean           type\n",
       "0   ADH1C          MDD        -0.794781  direct_target\n",
       "1  CHRNA2      BPD|SCZ         0.665546  direct_target\n",
       "2  CYP2D6  BPD|MDD|SCZ        -0.383639  direct_target\n",
       "3   DHODH          MDD        -1.038550  direct_target\n",
       "4   ITPR3          SCZ         0.608879  direct_target\n",
       "5   NFKB1          SCZ        -0.927891  direct_target\n",
       "6    SV2A      BPD|SCZ        -0.555829  direct_target"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "236\n",
      "135\n"
     ]
    }
   ],
   "source": [
    "# data_df_sel = (data_df[['diseases','drug_target_gene']]#,'drug_target_pertname']]\n",
    "#  .assign(diseases= lambda x: x.diseases.str.split('|'))\n",
    "#  .explode('diseases')\n",
    "#  .assign(drug_target_gene= lambda x: x.drug_target_gene.apply(lambda x: sorted(set(x.split('|')))\n",
    "#                                                                     ))\n",
    "#  .explode('drug_target_gene')\n",
    "#  .query('drug_target_gene!=\"\"')\n",
    "# #  .groupby\n",
    "# )\n",
    "# data_df_sel\n",
    "target_info = explore_genelist_disease('drug_target_gene')\n",
    "target_info['type'] = 'direct_target'\n",
    "display(target_info)\n",
    "cmap_up = explore_genelist_disease('drug_cmap_gene_up')\n",
    "cmap_up['type'] = 'cmap_up'\n",
    "print(len(cmap_up))\n",
    "cmap_dn = explore_genelist_disease('drug_cmap_gene_dn')\n",
    "cmap_dn['type'] = 'cmap_dn'\n",
    "print(len(cmap_dn))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(313, 4) (313,)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>diseases</th>\n",
       "      <th>mpra_logfc_mean</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ABCB9</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>0.297346</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ABCC8</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>-0.933245</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ABT1</td>\n",
       "      <td>MDD|SCZ</td>\n",
       "      <td>-2.045156</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ACO2</td>\n",
       "      <td>MDD</td>\n",
       "      <td>0.616614</td>\n",
       "      <td>cmap_dn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ADAMTS2</td>\n",
       "      <td>MDD</td>\n",
       "      <td>0.837381</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>308</td>\n",
       "      <td>ZNF276</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>1.101667</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>309</td>\n",
       "      <td>ZNF391</td>\n",
       "      <td>BPD|MDD</td>\n",
       "      <td>0.961567</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>310</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.855839</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>311</td>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>BPD|MDD|SCZ</td>\n",
       "      <td>0.278635</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>312</td>\n",
       "      <td>ZSCAN9</td>\n",
       "      <td>BPD|MDD|SCZ</td>\n",
       "      <td>0.343997</td>\n",
       "      <td>cmap_up</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>313 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene     diseases  mpra_logfc_mean             type\n",
       "0      ABCB9      BPD|SCZ         0.297346          cmap_up\n",
       "1      ABCC8      BPD|SCZ        -0.933245  cmap_up|cmap_dn\n",
       "2       ABT1      MDD|SCZ        -2.045156          cmap_up\n",
       "3       ACO2          MDD         0.616614          cmap_dn\n",
       "4    ADAMTS2          MDD         0.837381          cmap_up\n",
       "..       ...          ...              ...              ...\n",
       "308   ZNF276          SCZ         1.101667          cmap_up\n",
       "309   ZNF391      BPD|MDD         0.961567          cmap_up\n",
       "310   ZNF592          SCZ        -0.855839          cmap_up\n",
       "311  ZSCAN31  BPD|MDD|SCZ         0.278635          cmap_up\n",
       "312   ZSCAN9  BPD|MDD|SCZ         0.343997          cmap_up\n",
       "\n",
       "[313 rows x 4 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_drug_info = (pd.concat([target_info, cmap_up, cmap_dn])\n",
    "                 .groupby(['gene','diseases','mpra_logfc_mean'])\n",
    "                 .agg({'type':'|'.join})\n",
    "                 .reset_index()\n",
    "                 .sort_values('gene')\n",
    "                )\n",
    "print(all_drug_info.shape, all_drug_info.gene.unique().shape)\n",
    "all_drug_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## annotate with other drug information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# omim genes\n",
    "omim_genes=\"\"\"\n",
    "ACO2\n",
    "AKT1\n",
    "ANK3\n",
    "AP3B2\n",
    "ARC\n",
    "ARL3\n",
    "ATXN7\n",
    "BANF1\n",
    "CDK10\n",
    "CEP85L\n",
    "CHRNA2\n",
    "CLP1\n",
    "CNNM2\n",
    "CNOT1\n",
    "DCC\n",
    "DDHD2\n",
    "EP300\n",
    "ERBB2\n",
    "FGFR1\n",
    "FGFR3\n",
    "FIBP\n",
    "GNAI2\n",
    "HARS\n",
    "HLA-A\n",
    "HLA-DQB1\n",
    "IRF3\n",
    "MADD\n",
    "MARS\n",
    "MFN2\n",
    "MTHFD1\n",
    "MTHFR\n",
    "MYRF\n",
    "NEK1\n",
    "NT5C2\n",
    "PARK7\n",
    "PGAP3\n",
    "PIGQ\n",
    "PRMT7\n",
    "RERE\n",
    "RNASEH2C\n",
    "RSRC1\n",
    "SLC30A9\n",
    "SPTLC1\n",
    "SUFU\n",
    "SZT2\n",
    "TKT\n",
    "TMX2\n",
    "TPRKB\n",
    "VARS2\n",
    "WASF1\n",
    "WDR73\n",
    "\"\"\".split()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(omim_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "coding_genes = \"\"\"\n",
    "C4A\n",
    "CACNA1G\n",
    "DAGLA\n",
    "MAGI2\n",
    "STAG1\n",
    "SV2A\n",
    "XPO7\n",
    "\"\"\".split()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(coding_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    272\n",
      "True      41\n",
      "Name: in_omim, dtype: int64\n",
      "False    312\n",
      "True       1\n",
      "Name: is_coding, dtype: int64\n",
      "cmap_up                  171\n",
      "cmap_dn                   76\n",
      "cmap_up|cmap_dn           59\n",
      "direct_target|cmap_up      6\n",
      "direct_target              1\n",
      "Name: type, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "all_drug_info['in_omim'] = all_drug_info.gene.isin(omim_genes)\n",
    "all_drug_info['is_coding'] = all_drug_info.gene.isin(coding_genes)\n",
    "print(all_drug_info.in_omim.value_counts())\n",
    "print(all_drug_info.is_coding.value_counts())\n",
    "print(all_drug_info.type.value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(47, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>diseases</th>\n",
       "      <th>mpra_logfc_mean</th>\n",
       "      <th>type</th>\n",
       "      <th>in_omim</th>\n",
       "      <th>is_coding</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ACO2</td>\n",
       "      <td>MDD</td>\n",
       "      <td>0.616614</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ADH1C</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-0.794781</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.266337</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>AP3B2</td>\n",
       "      <td>PTSD</td>\n",
       "      <td>-0.951833</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>ARC</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.942587</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>ARL3</td>\n",
       "      <td>ADHD|BPD|SCZ</td>\n",
       "      <td>-0.103592</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>ATXN7</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.488048</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>BANF1</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>0.687733</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>CDK10</td>\n",
       "      <td>OCD</td>\n",
       "      <td>1.019088</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>CHRNA2</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>0.665546</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46</td>\n",
       "      <td>CLP1</td>\n",
       "      <td>MDD|SCZ</td>\n",
       "      <td>-2.920013</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>47</td>\n",
       "      <td>CNNM2</td>\n",
       "      <td>ADHD|BPD|SCZ</td>\n",
       "      <td>-0.347241</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48</td>\n",
       "      <td>CNOT1</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>1.148830</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>58</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>BPD|MDD|SCZ</td>\n",
       "      <td>-0.383639</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>DCC</td>\n",
       "      <td>MDD</td>\n",
       "      <td>0.057961</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>DHODH</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-1.038550</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74</td>\n",
       "      <td>EP300</td>\n",
       "      <td>MDD</td>\n",
       "      <td>0.616614</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>77</td>\n",
       "      <td>ERBB2</td>\n",
       "      <td>BPD</td>\n",
       "      <td>-0.479691</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>91</td>\n",
       "      <td>FGFR1</td>\n",
       "      <td>GAD</td>\n",
       "      <td>-1.093601</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>92</td>\n",
       "      <td>FGFR3</td>\n",
       "      <td>OCD</td>\n",
       "      <td>-1.847831</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>93</td>\n",
       "      <td>FIBP</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>0.687733</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>106</td>\n",
       "      <td>GNAI2</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>1.028166</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>111</td>\n",
       "      <td>HARS</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-1.143711</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>113</td>\n",
       "      <td>HLA-A</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>-1.138186</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>120</td>\n",
       "      <td>HLA-DQB1</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>0.273492</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>130</td>\n",
       "      <td>IRF3</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-1.704387</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>132</td>\n",
       "      <td>ITPR3</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>0.608879</td>\n",
       "      <td>direct_target</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>143</td>\n",
       "      <td>MADD</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>0.484664</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>149</td>\n",
       "      <td>MARS</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.611888</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>155</td>\n",
       "      <td>MFN2</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.091425</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>164</td>\n",
       "      <td>MTHFD1</td>\n",
       "      <td>MDD</td>\n",
       "      <td>1.051424</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>166</td>\n",
       "      <td>MTHFR</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>0.677840</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>174</td>\n",
       "      <td>NEK1</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.848821</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>177</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.927891</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>182</td>\n",
       "      <td>NT5C2</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-1.340069</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>192</td>\n",
       "      <td>PARK7</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-0.175536</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>206</td>\n",
       "      <td>PIGQ</td>\n",
       "      <td>ADHD</td>\n",
       "      <td>-0.839988</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>216</td>\n",
       "      <td>PRMT7</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.081837</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>226</td>\n",
       "      <td>RERE</td>\n",
       "      <td>MDD|SCZ</td>\n",
       "      <td>0.539125</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>238</td>\n",
       "      <td>RSRC1</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-0.619963</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>252</td>\n",
       "      <td>SLC30A9</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-0.366326</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>258</td>\n",
       "      <td>SPTLC1</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>0.019732</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>265</td>\n",
       "      <td>SV2A</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>-0.555829</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>266</td>\n",
       "      <td>SZT2</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.792680</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>273</td>\n",
       "      <td>TKT</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>-0.837160</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>278</td>\n",
       "      <td>TMX2</td>\n",
       "      <td>MDD|SCZ</td>\n",
       "      <td>-2.920013</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>297</td>\n",
       "      <td>WDR73</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.855839</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene      diseases  mpra_logfc_mean                   type  in_omim  \\\n",
       "3        ACO2           MDD         0.616614                cmap_dn     True   \n",
       "5       ADH1C           MDD        -0.794781  direct_target|cmap_up    False   \n",
       "8        AKT1           SCZ        -0.266337        cmap_up|cmap_dn     True   \n",
       "10      AP3B2          PTSD        -0.951833                cmap_up     True   \n",
       "12        ARC           SCZ        -0.942587                cmap_up     True   \n",
       "14       ARL3  ADHD|BPD|SCZ        -0.103592        cmap_up|cmap_dn     True   \n",
       "20      ATXN7           SCZ        -0.488048        cmap_up|cmap_dn     True   \n",
       "24      BANF1           SCZ         0.687733                cmap_dn     True   \n",
       "37      CDK10           OCD         1.019088                cmap_up     True   \n",
       "43     CHRNA2       BPD|SCZ         0.665546  direct_target|cmap_up     True   \n",
       "46       CLP1       MDD|SCZ        -2.920013                cmap_up     True   \n",
       "47      CNNM2  ADHD|BPD|SCZ        -0.347241                cmap_up     True   \n",
       "48      CNOT1       BPD|SCZ         1.148830                cmap_up     True   \n",
       "58     CYP2D6   BPD|MDD|SCZ        -0.383639  direct_target|cmap_up    False   \n",
       "61        DCC           MDD         0.057961                cmap_up     True   \n",
       "66      DHODH           MDD        -1.038550  direct_target|cmap_up    False   \n",
       "74      EP300           MDD         0.616614                cmap_up     True   \n",
       "77      ERBB2           BPD        -0.479691                cmap_dn     True   \n",
       "91      FGFR1           GAD        -1.093601        cmap_up|cmap_dn     True   \n",
       "92      FGFR3           OCD        -1.847831                cmap_up     True   \n",
       "93       FIBP           SCZ         0.687733                cmap_dn     True   \n",
       "106     GNAI2           SCZ         1.028166                cmap_dn     True   \n",
       "111      HARS           SCZ        -1.143711                cmap_dn     True   \n",
       "113     HLA-A       BPD|SCZ        -1.138186                cmap_up     True   \n",
       "120  HLA-DQB1           SCZ         0.273492                cmap_up     True   \n",
       "130      IRF3           SCZ        -1.704387                cmap_up     True   \n",
       "132     ITPR3           SCZ         0.608879          direct_target    False   \n",
       "143      MADD           SCZ         0.484664                cmap_up     True   \n",
       "149      MARS           SCZ        -0.611888                cmap_up     True   \n",
       "155      MFN2           SCZ        -0.091425                cmap_dn     True   \n",
       "164    MTHFD1           MDD         1.051424                cmap_dn     True   \n",
       "166     MTHFR           SCZ         0.677840                cmap_up     True   \n",
       "174      NEK1           SCZ        -0.848821                cmap_up     True   \n",
       "177     NFKB1           SCZ        -0.927891  direct_target|cmap_up    False   \n",
       "182     NT5C2           SCZ        -1.340069        cmap_up|cmap_dn     True   \n",
       "192     PARK7           MDD        -0.175536                cmap_dn     True   \n",
       "206      PIGQ          ADHD        -0.839988        cmap_up|cmap_dn     True   \n",
       "216     PRMT7           SCZ        -0.081837                cmap_up     True   \n",
       "226      RERE       MDD|SCZ         0.539125                cmap_dn     True   \n",
       "238     RSRC1           MDD        -0.619963                cmap_up     True   \n",
       "252   SLC30A9           MDD        -0.366326        cmap_up|cmap_dn     True   \n",
       "258    SPTLC1       BPD|SCZ         0.019732                cmap_dn     True   \n",
       "265      SV2A       BPD|SCZ        -0.555829  direct_target|cmap_up    False   \n",
       "266      SZT2           SCZ        -0.792680                cmap_up     True   \n",
       "273       TKT       BPD|SCZ        -0.837160                cmap_dn     True   \n",
       "278      TMX2       MDD|SCZ        -2.920013                cmap_dn     True   \n",
       "297     WDR73           SCZ        -0.855839        cmap_up|cmap_dn     True   \n",
       "\n",
       "     is_coding  \n",
       "3        False  \n",
       "5        False  \n",
       "8        False  \n",
       "10       False  \n",
       "12       False  \n",
       "14       False  \n",
       "20       False  \n",
       "24       False  \n",
       "37       False  \n",
       "43       False  \n",
       "46       False  \n",
       "47       False  \n",
       "48       False  \n",
       "58       False  \n",
       "61       False  \n",
       "66       False  \n",
       "74       False  \n",
       "77       False  \n",
       "91       False  \n",
       "92       False  \n",
       "93       False  \n",
       "106      False  \n",
       "111      False  \n",
       "113      False  \n",
       "120      False  \n",
       "130      False  \n",
       "132      False  \n",
       "143      False  \n",
       "149      False  \n",
       "155      False  \n",
       "164      False  \n",
       "166      False  \n",
       "174      False  \n",
       "177      False  \n",
       "182      False  \n",
       "192      False  \n",
       "206      False  \n",
       "216      False  \n",
       "226      False  \n",
       "238      False  \n",
       "252      False  \n",
       "258      False  \n",
       "265       True  \n",
       "266      False  \n",
       "273      False  \n",
       "278      False  \n",
       "297      False  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_drug_info_filt = all_drug_info[( all_drug_info.type.str.contains('direct'))|all_drug_info.in_omim|all_drug_info.is_coding]\n",
    "print(all_drug_info_filt.shape)\n",
    "all_drug_info_filt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# add in gene class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3035, 15)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>gene</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>idgFamily</th>\n",
       "      <th>idgTDL</th>\n",
       "      <th>jensenScore</th>\n",
       "      <th>novelty</th>\n",
       "      <th>knowledgeAvailability</th>\n",
       "      <th>pubTatorScore</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>drug_list</th>\n",
       "      <th>interaction_list</th>\n",
       "      <th>source_list</th>\n",
       "      <th>raw_info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>HTR1E</td>\n",
       "      <td>5-hydroxytryptamine receptor 1E</td>\n",
       "      <td>G-protein coupled receptor for 5-hydroxytrypta...</td>\n",
       "      <td>GPCR</td>\n",
       "      <td>Tchem</td>\n",
       "      <td>35.946685</td>\n",
       "      <td>-1.568858</td>\n",
       "      <td>24.464814</td>\n",
       "      <td>1.495159</td>\n",
       "      <td>3354</td>\n",
       "      <td>['OLANZAPINE', '1-NAPHTHYLPIPERAZINE', 'NARATR...</td>\n",
       "      <td>[nan, 'antagonist', 'agonist', 'agonist', 'ago...</td>\n",
       "      <td>['TdgClinicalTrial', 'GuideToPharmacologyInter...</td>\n",
       "      <td>[{'gene_name': 'HTR1E', 'gene_claim_name': 'P2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>EMP2</td>\n",
       "      <td>Epithelial membrane protein 2</td>\n",
       "      <td>Functions as a key regulator of cell membrane ...</td>\n",
       "      <td>Non-IDG</td>\n",
       "      <td>Tbio</td>\n",
       "      <td>210.602988</td>\n",
       "      <td>-2.308556</td>\n",
       "      <td>37.256058</td>\n",
       "      <td>1.599810</td>\n",
       "      <td>2013</td>\n",
       "      <td>['HORMONES']</td>\n",
       "      <td>[nan]</td>\n",
       "      <td>['NCI']</td>\n",
       "      <td>[{'gene_name': 'EMP2', 'gene_claim_name': 'EMP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>DUSP1</td>\n",
       "      <td>Dual specificity protein phosphatase 1</td>\n",
       "      <td>Dual specificity phosphatase that dephosphoryl...</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>Tchem</td>\n",
       "      <td>810.925362</td>\n",
       "      <td>-2.882599</td>\n",
       "      <td>47.969158</td>\n",
       "      <td>2.515725</td>\n",
       "      <td>1843</td>\n",
       "      <td>['HYDROXYUREA', 'VASOPRESSIN', 'ANTISENSE OLIG...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>['NCI', 'NCI', 'NCI']</td>\n",
       "      <td>[{'gene_name': 'DUSP1', 'gene_claim_name': 'DU...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>IL9</td>\n",
       "      <td>Interleukin-9</td>\n",
       "      <td>Supports IL-2 independent and IL-4 independent...</td>\n",
       "      <td>Non-IDG</td>\n",
       "      <td>Tbio</td>\n",
       "      <td>334.361390</td>\n",
       "      <td>-2.508110</td>\n",
       "      <td>20.035321</td>\n",
       "      <td>2.940012</td>\n",
       "      <td>3578</td>\n",
       "      <td>['MEDI-528', 'ENOKIZUMAB', 'MEDI-528']</td>\n",
       "      <td>['inhibitor', 'inhibitor', nan]</td>\n",
       "      <td>['TTD', 'ChemblInteractions', 'TdgClinicalTrial']</td>\n",
       "      <td>[{'gene_name': 'IL9', 'gene_claim_name': 'TTDC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>SELE</td>\n",
       "      <td>E-selectin</td>\n",
       "      <td>Cell-surface glycoprotein having a role in imm...</td>\n",
       "      <td>Non-IDG</td>\n",
       "      <td>Tchem</td>\n",
       "      <td>2672.113112</td>\n",
       "      <td>-3.449258</td>\n",
       "      <td>37.448817</td>\n",
       "      <td>3.350202</td>\n",
       "      <td>6401</td>\n",
       "      <td>['BIMOSIAMOSE', 'BIMOSIAMOSE', 'BIMOSIAMOSE', ...</td>\n",
       "      <td>[nan, 'inhibitor', 'inhibitor', 'antagonist', ...</td>\n",
       "      <td>['TdgClinicalTrial', 'TTD', 'ChemblInteraction...</td>\n",
       "      <td>[{'gene_name': 'SELE', 'gene_claim_name': 'P16...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id   gene                                    name  \\\n",
       "0   5  HTR1E         5-hydroxytryptamine receptor 1E   \n",
       "1   9   EMP2           Epithelial membrane protein 2   \n",
       "2  11  DUSP1  Dual specificity protein phosphatase 1   \n",
       "3  15    IL9                           Interleukin-9   \n",
       "4  22   SELE                              E-selectin   \n",
       "\n",
       "                                         description idgFamily idgTDL  \\\n",
       "0  G-protein coupled receptor for 5-hydroxytrypta...      GPCR  Tchem   \n",
       "1  Functions as a key regulator of cell membrane ...   Non-IDG   Tbio   \n",
       "2  Dual specificity phosphatase that dephosphoryl...    Enzyme  Tchem   \n",
       "3  Supports IL-2 independent and IL-4 independent...   Non-IDG   Tbio   \n",
       "4  Cell-surface glycoprotein having a role in imm...   Non-IDG  Tchem   \n",
       "\n",
       "   jensenScore   novelty  knowledgeAvailability  pubTatorScore  entrez_id  \\\n",
       "0    35.946685 -1.568858              24.464814       1.495159       3354   \n",
       "1   210.602988 -2.308556              37.256058       1.599810       2013   \n",
       "2   810.925362 -2.882599              47.969158       2.515725       1843   \n",
       "3   334.361390 -2.508110              20.035321       2.940012       3578   \n",
       "4  2672.113112 -3.449258              37.448817       3.350202       6401   \n",
       "\n",
       "                                           drug_list  \\\n",
       "0  ['OLANZAPINE', '1-NAPHTHYLPIPERAZINE', 'NARATR...   \n",
       "1                                       ['HORMONES']   \n",
       "2  ['HYDROXYUREA', 'VASOPRESSIN', 'ANTISENSE OLIG...   \n",
       "3             ['MEDI-528', 'ENOKIZUMAB', 'MEDI-528']   \n",
       "4  ['BIMOSIAMOSE', 'BIMOSIAMOSE', 'BIMOSIAMOSE', ...   \n",
       "\n",
       "                                    interaction_list  \\\n",
       "0  [nan, 'antagonist', 'agonist', 'agonist', 'ago...   \n",
       "1                                              [nan]   \n",
       "2                                    [nan, nan, nan]   \n",
       "3                    ['inhibitor', 'inhibitor', nan]   \n",
       "4  [nan, 'inhibitor', 'inhibitor', 'antagonist', ...   \n",
       "\n",
       "                                         source_list  \\\n",
       "0  ['TdgClinicalTrial', 'GuideToPharmacologyInter...   \n",
       "1                                            ['NCI']   \n",
       "2                              ['NCI', 'NCI', 'NCI']   \n",
       "3  ['TTD', 'ChemblInteractions', 'TdgClinicalTrial']   \n",
       "4  ['TdgClinicalTrial', 'TTD', 'ChemblInteraction...   \n",
       "\n",
       "                                            raw_info  \n",
       "0  [{'gene_name': 'HTR1E', 'gene_claim_name': 'P2...  \n",
       "1  [{'gene_name': 'EMP2', 'gene_claim_name': 'EMP...  \n",
       "2  [{'gene_name': 'DUSP1', 'gene_claim_name': 'DU...  \n",
       "3  [{'gene_name': 'IL9', 'gene_claim_name': 'TTDC...  \n",
       "4  [{'gene_name': 'SELE', 'gene_claim_name': 'P16...  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_df = pd.read_excel('/Users/mguo123/Google Drive/1_khavari/perturb_tf_project/pharos_targets_filtered2.xlsx',sheet_name='targets')\n",
    "\n",
    "\n",
    "print(drug_df.shape)\n",
    "drug_df[:5]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>diseases</th>\n",
       "      <th>mpra_logfc_mean</th>\n",
       "      <th>type</th>\n",
       "      <th>in_omim</th>\n",
       "      <th>is_coding</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>idgFamily</th>\n",
       "      <th>idgTDL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ACO2</td>\n",
       "      <td>MDD</td>\n",
       "      <td>0.616614</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ADH1C</td>\n",
       "      <td>MDD</td>\n",
       "      <td>-0.794781</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Alcohol dehydrogenase 1C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Enzyme</td>\n",
       "      <td>Tclin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.266337</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>RAC-alpha serine/threonine-protein kinase</td>\n",
       "      <td>AKT1-specific substrates have been recently id...</td>\n",
       "      <td>Kinase</td>\n",
       "      <td>Tchem</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>AP3B2</td>\n",
       "      <td>PTSD</td>\n",
       "      <td>-0.951833</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ARC</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.942587</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>SV2A</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>-0.555829</td>\n",
       "      <td>direct_target|cmap_up</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Synaptic vesicle glycoprotein 2A</td>\n",
       "      <td>Plays a role in the control of regulated secre...</td>\n",
       "      <td>Non-IDG</td>\n",
       "      <td>Tclin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>SZT2</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.792680</td>\n",
       "      <td>cmap_up</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>TKT</td>\n",
       "      <td>BPD|SCZ</td>\n",
       "      <td>-0.837160</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>65</td>\n",
       "      <td>TMX2</td>\n",
       "      <td>MDD|SCZ</td>\n",
       "      <td>-2.920013</td>\n",
       "      <td>cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66</td>\n",
       "      <td>WDR73</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>-0.855839</td>\n",
       "      <td>cmap_up|cmap_dn</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene diseases  mpra_logfc_mean                   type  in_omim  \\\n",
       "0    ACO2      MDD         0.616614                cmap_dn     True   \n",
       "1   ADH1C      MDD        -0.794781  direct_target|cmap_up    False   \n",
       "2    AKT1      SCZ        -0.266337        cmap_up|cmap_dn     True   \n",
       "3   AP3B2     PTSD        -0.951833                cmap_up     True   \n",
       "4     ARC      SCZ        -0.942587                cmap_up     True   \n",
       "..    ...      ...              ...                    ...      ...   \n",
       "62   SV2A  BPD|SCZ        -0.555829  direct_target|cmap_up    False   \n",
       "63   SZT2      SCZ        -0.792680                cmap_up     True   \n",
       "64    TKT  BPD|SCZ        -0.837160                cmap_dn     True   \n",
       "65   TMX2  MDD|SCZ        -2.920013                cmap_dn     True   \n",
       "66  WDR73      SCZ        -0.855839        cmap_up|cmap_dn     True   \n",
       "\n",
       "    is_coding                                       name  \\\n",
       "0       False                                        NaN   \n",
       "1       False                   Alcohol dehydrogenase 1C   \n",
       "2       False  RAC-alpha serine/threonine-protein kinase   \n",
       "3       False                                        NaN   \n",
       "4       False                                        NaN   \n",
       "..        ...                                        ...   \n",
       "62       True           Synaptic vesicle glycoprotein 2A   \n",
       "63      False                                        NaN   \n",
       "64      False                                        NaN   \n",
       "65      False                                        NaN   \n",
       "66      False                                        NaN   \n",
       "\n",
       "                                          description idgFamily idgTDL  \n",
       "0                                                 NaN       NaN    NaN  \n",
       "1                                                 NaN    Enzyme  Tclin  \n",
       "2   AKT1-specific substrates have been recently id...    Kinase  Tchem  \n",
       "3                                                 NaN       NaN    NaN  \n",
       "4                                                 NaN       NaN    NaN  \n",
       "..                                                ...       ...    ...  \n",
       "62  Plays a role in the control of regulated secre...   Non-IDG  Tclin  \n",
       "63                                                NaN       NaN    NaN  \n",
       "64                                                NaN       NaN    NaN  \n",
       "65                                                NaN       NaN    NaN  \n",
       "66                                                NaN       NaN    NaN  \n",
       "\n",
       "[67 rows x 10 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_drug_info_filt = all_drug_info_filt.merge(drug_df[['gene','name','description','idgFamily','idgTDL']], how='left',on='gene')\n",
    "all_drug_info_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_drug_info.to_csv('figure_tables/all_drug_info.csv')\n",
    "all_drug_info_filt.to_csv('figure_tables/all_drug_info_filt.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GPCRs\n",
    "\n",
    "manual add into spreadsheet\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(415, 33)\n",
      "Index(['Type', 'Family id', 'Family name', 'Target id', 'Target name',\n",
      "       'Subunits', 'Target systematic name', 'Target abbreviated name',\n",
      "       'synonyms', 'HGNC id', 'HGNC symbol', 'HGNC name',\n",
      "       'Human genetic localisation', 'Human nucleotide RefSeq',\n",
      "       'Human protein RefSeq', 'Human SwissProt', 'Human Entrez Gene',\n",
      "       'RGD id', 'RGD symbol', 'RGD name', 'Rat genetic localisation',\n",
      "       'Rat nucleotide RefSeq', 'Rat protein RefSeq', 'Rat SwissProt',\n",
      "       'Rat Entrez Gene', 'MGI id', 'MGI symbol', 'MGI name',\n",
      "       'Mouse genetic localisation', 'Mouse nucleotide RefSeq',\n",
      "       'Mouse protein RefSeq', 'Mouse SwissProt', 'Mouse Entrez Gene'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "gpcr_df = pd.read_csv('/Users/mguo123/Google Drive/0_altman/db/gpcr/GPCRTargets.csv')\n",
    "print(gpcr_df.shape)\n",
    "print(gpcr_df.columns)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ACKR1' 'ACKR2' 'ACKR3' 'ACKR4' 'ADCYAP1R1' 'ADGRA1' 'ADGRA2' 'ADGRA3'\n",
      " 'ADGRB1' 'ADGRB2' 'ADGRB3' 'ADGRD1' 'ADGRD2' 'ADGRE1' 'ADGRE2' 'ADGRE3'\n",
      " 'ADGRE4P' 'ADGRE5' 'ADGRF1' 'ADGRF2' 'ADGRF3' 'ADGRF4' 'ADGRF5' 'ADGRG1'\n",
      " 'ADGRG2' 'ADGRG3' 'ADGRG4' 'ADGRG5' 'ADGRG6' 'ADGRG7' 'ADGRL1' 'ADGRL2'\n",
      " 'ADGRL3' 'ADGRL4' 'ADGRV1' 'ADORA1' 'ADORA2A' 'ADORA2B' 'ADORA3' 'ADRA1A'\n",
      " 'ADRA1B' 'ADRA1D' 'ADRA2A' 'ADRA2B' 'ADRA2C' 'ADRB1' 'ADRB2' 'ADRB3'\n",
      " 'AGTR1' 'AGTR2' 'APLNR' 'AVPR1A' 'AVPR1B' 'AVPR2' 'BDKRB1' 'BDKRB2'\n",
      " 'BRS3' 'BRS3' 'C3AR1' 'C5AR1' 'C5AR2' 'CALCR' 'CALCRL' 'CASR' 'CCKAR'\n",
      " 'CCKBR' 'CCR1' 'CCR10' 'CCR2' 'CCR3' 'CCR4' 'CCR5' 'CCR6' 'CCR7' 'CCR8'\n",
      " 'CCR9' 'CCRL2' 'CELSR1' 'CELSR2' 'CELSR3' 'CHRM1' 'CHRM2' 'CHRM3' 'CHRM4'\n",
      " 'CHRM5' 'CMKLR1' 'CNR1' 'CNR2' 'CRHR1' 'CRHR2' 'CX3CR1' 'CXCR1' 'CXCR2'\n",
      " 'CXCR3' 'CXCR4' 'CXCR5' 'CXCR6' 'CYSLTR1' 'CYSLTR2' 'DRD1' 'DRD2' 'DRD3'\n",
      " 'DRD4' 'DRD5' 'EDNRA' 'EDNRB' 'F2R' 'F2RL1' 'F2RL2' 'F2RL3' 'FFAR1'\n",
      " 'FFAR2' 'FFAR3' 'FFAR4' 'FPR1' 'FPR2' 'FPR2' 'FPR3' 'FSHR' 'FZD1' 'FZD10'\n",
      " 'FZD2' 'FZD3' 'FZD4' 'FZD5' 'FZD6' 'FZD7' 'FZD8' 'FZD9' 'GABBR1' 'GABBR2'\n",
      " 'GALR1' 'GALR2' 'GALR3' 'GCGR' 'GHRHR' 'GHSR' 'GIPR' 'GLP1R' 'GLP2R'\n",
      " 'GNRHR' 'GNRHR2' 'GPBAR1' 'GPER1' 'GPR1' 'GPR101' 'GPR107' 'GPR119'\n",
      " 'GPR119' 'GPR12' 'GPR132' 'GPR135' 'GPR137' 'GPR139' 'GPR141' 'GPR142'\n",
      " 'GPR143' 'GPR146' 'GPR148' 'GPR149' 'GPR15' 'GPR150' 'GPR151' 'GPR152'\n",
      " 'GPR153' 'GPR156' 'GPR157' 'GPR158' 'GPR160' 'GPR161' 'GPR162' 'GPR17'\n",
      " 'GPR171' 'GPR173' 'GPR174' 'GPR176' 'GPR179' 'GPR18' 'GPR18' 'GPR182'\n",
      " 'GPR183' 'GPR19' 'GPR20' 'GPR21' 'GPR22' 'GPR25' 'GPR26' 'GPR27' 'GPR3'\n",
      " 'GPR31' 'GPR32' 'GPR33' 'GPR34' 'GPR35' 'GPR37' 'GPR37L1' 'GPR39' 'GPR4'\n",
      " 'GPR42' 'GPR42' 'GPR45' 'GPR50' 'GPR52' 'GPR55' 'GPR55' 'GPR6' 'GPR61'\n",
      " 'GPR62' 'GPR63' 'GPR65' 'GPR68' 'GPR75' 'GPR78' 'GPR79' 'GPR82' 'GPR83'\n",
      " 'GPR84' 'GPR85' 'GPR87' 'GPR88' 'GPRC5A' 'GPRC5B' 'GPRC5C' 'GPRC5D'\n",
      " 'GPRC6A' 'GRM1' 'GRM2' 'GRM3' 'GRM4' 'GRM5' 'GRM6' 'GRM7' 'GRM8' 'GRPR'\n",
      " 'HCAR1' 'HCAR2' 'HCAR3' 'HCRTR1' 'HCRTR2' 'HRH1' 'HRH2' 'HRH3' 'HRH4'\n",
      " 'HTR1A' 'HTR1B' 'HTR1D' 'HTR1E' 'HTR1F' 'HTR2A' 'HTR2B' 'HTR2C' 'HTR4'\n",
      " 'HTR5A' 'HTR5BP' 'HTR6' 'HTR7' 'KISS1R' 'LGR4' 'LGR5' 'LGR6' 'LHCGR'\n",
      " 'LPAR1' 'LPAR2' 'LPAR3' 'LPAR4' 'LPAR5' 'LPAR6' 'LTB4R' 'LTB4R2' 'MAS1'\n",
      " 'MAS1L' 'MC1R' 'MC2R' 'MC3R' 'MC4R' 'MC5R' 'MCHR1' 'MCHR2' 'MLNR'\n",
      " 'MRGPRD' 'MRGPRE' 'MRGPRF' 'MRGPRG' 'MRGPRX1' 'MRGPRX2' 'MRGPRX3'\n",
      " 'MRGPRX4' 'MTNR1A' 'MTNR1B' 'NMBR' 'NMUR1' 'NMUR2' 'NPBWR1' 'NPBWR2'\n",
      " 'NPFFR1' 'NPFFR2' 'NPSR1' 'NPY1R' 'NPY2R' 'NPY4R' 'NPY5R' 'NPY6R' 'NTSR1'\n",
      " 'NTSR2' 'OPN1LW' 'OPN1MW' 'OPN1SW' 'OPN3' 'OPN4' 'OPN5' 'OPRD1' 'OPRK1'\n",
      " 'OPRL1' 'OPRM1' 'OXER1' 'OXGR1' 'OXTR' 'P2RY1' 'P2RY10' 'P2RY11' 'P2RY12'\n",
      " 'P2RY13' 'P2RY14' 'P2RY2' 'P2RY4' 'P2RY6' 'P2RY8' 'PRLHR' 'PROKR1'\n",
      " 'PROKR2' 'PTAFR' 'PTGDR' 'PTGDR2' 'PTGER1' 'PTGER2' 'PTGER3' 'PTGER4'\n",
      " 'PTGFR' 'PTGIR' 'PTH1R' 'PTH2R' 'QRFPR' 'RHO' 'RXFP1' 'RXFP2' 'RXFP3'\n",
      " 'RXFP4' 'S1PR1' 'S1PR2' 'S1PR3' 'S1PR4' 'S1PR5' 'SCTR' 'SMO' 'SSTR1'\n",
      " 'SSTR2' 'SSTR3' 'SSTR4' 'SSTR5' 'SUCNR1' 'TAAR1' 'TAAR2' 'TAAR3P'\n",
      " 'TAAR4P' 'TAAR5' 'TAAR6' 'TAAR8' 'TAAR9' 'TACR1' 'TACR2' 'TACR3' 'TAS1R1'\n",
      " 'TAS1R2' 'TAS1R3' 'TAS2R1' 'TAS2R10' 'TAS2R13' 'TAS2R14' 'TAS2R16'\n",
      " 'TAS2R19' 'TAS2R20' 'TAS2R3' 'TAS2R30' 'TAS2R31' 'TAS2R38' 'TAS2R39'\n",
      " 'TAS2R4' 'TAS2R40' 'TAS2R41' 'TAS2R42' 'TAS2R43' 'TAS2R45' 'TAS2R46'\n",
      " 'TAS2R5' 'TAS2R50' 'TAS2R60' 'TAS2R7' 'TAS2R8' 'TAS2R9' 'TBXA2R' 'TPRA1'\n",
      " 'TRHR' 'TSHR' 'UTS2R' 'VIPR1' 'VIPR2' 'XCR1' nan nan nan nan nan nan nan\n",
      " nan]\n"
     ]
    }
   ],
   "source": [
    "gpcr_genes = gpcr_df['HGNC symbol'].sort_values().values\n",
    "print(gpcr_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>diseases</th>\n",
       "      <th>mpra_logfc_mean</th>\n",
       "      <th>type</th>\n",
       "      <th>in_omim</th>\n",
       "      <th>is_coding</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>idgFamily</th>\n",
       "      <th>idgTDL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [gene, diseases, mpra_logfc_mean, type, in_omim, is_coding, name, description, idgFamily, idgTDL]\n",
       "Index: []"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_drug_info_filt[all_drug_info_filt.gene.isin(gpcr_df['HGNC symbol'].values)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LPAR2\n",
      "MCHR1\n"
     ]
    }
   ],
   "source": [
    "all_genes = pd.read_csv('D_mpraanalyze_barcode_allelic/egene_gtex_psychencode/all.txt',header=None).iloc[:,0].values\n",
    "for g in gpcr_genes:\n",
    "    if g in all_genes:\n",
    "        print(g)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "rowname                                                                chr19_19447470\n",
       "Linked_SNP                                                                  rs1989867\n",
       "Chr                                                                             chr19\n",
       "Position                                                                     19447470\n",
       "Ref/Alt                                                                           A/G\n",
       "Index_SNP                                                         rs1064395,rs2011503\n",
       "ConsScore                                                                           2\n",
       "ConsDetail                                                                     intron\n",
       "diseases                                                                          BPD\n",
       "num_dz                                                                              1\n",
       "gwas_pval                                                                       2e-09\n",
       "source                                                                           gwas\n",
       "gene_annon                                                                       MAU2\n",
       "ngene_haploreg                                                                   MAU2\n",
       "egene_alltissue_gtex                           CILP2|HAPLN4|LPAR2|SUGP1|TM6SF2|YJEFN3\n",
       "egene_neurotissue_gtex              CILP2|GATAD2A|GMIP|HAPLN4|LPAR2|TM6SF2|TSSK6|Y...\n",
       "egene_neurotissue_mapgene_gtex      CILP2|CILP2|CILP2|CILP2|GATAD2A|GMIP|GMIP|HAPL...\n",
       "egene_neurotissue_maptissue_gtex    Brain_Nucleus_accumbens_basal_ganglia|Brain_Ca...\n",
       "atac_tissues                                                                         \n",
       "bool_in_atac_pk                                                                 False\n",
       "hichip_tissues                                                        H9D0|H9D2|H9D10\n",
       "bool_in_hichip_pk                                                                True\n",
       "broken_motifs                                                                   NHLH1\n",
       "gained_motifs                                                                    MBD2\n",
       "drug_target_gene                                                                     \n",
       "drug_target_pertname                                                                 \n",
       "drug_target_moa                                                                      \n",
       "drug_target_clinical_phase                                                           \n",
       "drug_target_indication                                                               \n",
       "drug_cmap_gene_up                        GMIP|GMIP|GMIP|LPAR2|LPAR2|LPAR2|LPAR2|LPAR2\n",
       "drug_cmap_pertname_up               biperiden|penfluridol|thiocolchicoside|chlorpr...\n",
       "drug_cmap_gene_dn                   GATAD2A|GATAD2A|LPAR2|LPAR2|LPAR2|LPAR2|LPAR2|...\n",
       "drug_cmap_pertname_dn               biperiden|trifluoperazine|clozapine|dihydroerg...\n",
       "tissue_asatac                                                                        \n",
       "bool_is_asatac                                                                  False\n",
       "tissue_ashichip                                                                      \n",
       "bool_is_ashichip                                                                False\n",
       "mpra_tissue                                                                   SLC|SLC\n",
       "mpra_pval_str                                                             0.004|0.004\n",
       "mpra_pval_mean                                                             0.00356842\n",
       "mpra_logfc_str                                                            0.658|0.658\n",
       "mpra_logfc_mean                                                              0.658366\n",
       "bool_mpra_sig                                                                    True\n",
       "cgene_egene_agree                                                                    \n",
       "cgene_egene_agree_num                                                               0\n",
       "bool_cgene_egene_agree                                                          False\n",
       "ukbb_drug                                                                            \n",
       "ukbb_phenotype                      inadequacy|material deprivation|past abuse|sel...\n",
       "ukbb_phenotype_beta                                    0.011|0.009|0.017|0.004|-0.004\n",
       "SNP_id                                                                    19:19447470\n",
       "psychencode_egenes                                  GATAD2A|HAPLN4|TM6SF2|TSSK6|ZNF14\n",
       "egenes_all_sources                  CILP2|GATAD2A|GMIP|HAPLN4|LPAR2|TM6SF2|TSSK6|Y...\n",
       "abc_genes                                                                            \n",
       "abc_gene_tissues                                                                     \n",
       "Name: 20, dtype: object"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_df[data_df.egenes_all_sources.str.contains('LPAR2')].iloc[0,:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MCHR1 is a Melanin-concentrating hormone receptor 1, but the new bipolar paper says it's targetted by haloperidol\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "rowname                                                          chr22_41024854\n",
       "Linked_SNP                                                           rs73169089\n",
       "Chr                                                                       chr22\n",
       "Position                                                               41024854\n",
       "Ref/Alt                                                                     T/C\n",
       "Index_SNP                                                  rs17002034,rs6001982\n",
       "ConsScore                                                                     4\n",
       "ConsDetail                                                           regulatory\n",
       "diseases                                                                BPD|SCZ\n",
       "num_dz                                                                        2\n",
       "gwas_pval                                                                 1e-34\n",
       "source                                                                     gwas\n",
       "gene_annon                                                                     \n",
       "ngene_haploreg                                                             MKL1\n",
       "egene_alltissue_gtex                                                           \n",
       "egene_neurotissue_gtex                                                     MKL1\n",
       "egene_neurotissue_mapgene_gtex                                             MKL1\n",
       "egene_neurotissue_maptissue_gtex                    Brain_Putamen_basal_ganglia\n",
       "atac_tissues                                                                   \n",
       "bool_in_atac_pk                                                           False\n",
       "hichip_tissues                                                 H9D0|SLC|SL|H9D2\n",
       "bool_in_hichip_pk                                                          True\n",
       "broken_motifs                       BATF|FOSB|FOSL1|FOSL2|JUN|JUNB|JUND|SMARCC1\n",
       "gained_motifs                                               NFIA+NFIB+NFIC+NFIX\n",
       "drug_target_gene                                                               \n",
       "drug_target_pertname                                                           \n",
       "drug_target_moa                                                                \n",
       "drug_target_clinical_phase                                                     \n",
       "drug_target_indication                                                         \n",
       "drug_cmap_gene_up                                                          MKL1\n",
       "drug_cmap_pertname_up                                               penfluridol\n",
       "drug_cmap_gene_dn                                                              \n",
       "drug_cmap_pertname_dn                                                          \n",
       "tissue_asatac                                                                  \n",
       "bool_is_asatac                                                            False\n",
       "tissue_ashichip                                                                \n",
       "bool_is_ashichip                                                          False\n",
       "mpra_tissue                                                         H9D10|H9D10\n",
       "mpra_pval_str                                                       0.000|0.000\n",
       "mpra_pval_mean                                                      1.52086e-27\n",
       "mpra_logfc_str                                                    -1.423|-1.423\n",
       "mpra_logfc_mean                                                        -1.42288\n",
       "bool_mpra_sig                                                              True\n",
       "cgene_egene_agree                                                              \n",
       "cgene_egene_agree_num                                                         0\n",
       "bool_cgene_egene_agree                                                    False\n",
       "ukbb_drug                                                         betamethasone\n",
       "ukbb_phenotype                                                                 \n",
       "ukbb_phenotype_beta                                                            \n",
       "SNP_id                                                              22:41024854\n",
       "psychencode_egenes                                                  MCHR1|SGSM3\n",
       "egenes_all_sources                                             MCHR1|MKL1|SGSM3\n",
       "abc_genes                                                                      \n",
       "abc_gene_tissues                                                               \n",
       "Name: 205, dtype: object"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_df[data_df.egenes_all_sources.str.contains('MCHR1')].iloc[0,:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# intersection all gene list with pharos drug targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['gene', 'name', 'description', 'idgFamily', 'idgTDL'], dtype='object')\n",
      "----------\n",
      "CACNA2D2 Ion Channel Tclin\n",
      "The alpha-2/delta subunit of voltage-dependent calcium channels regulates calcium current density and activation/inactivation kinetics of the calcium channel. Acts as a regulatory subunit for P/Q-type calcium channel (CACNA1A), N-type (CACNA1B), L-type (CACNA1C OR CACNA1D) and possibly T-type (CACNA1G). Overexpression induces apoptosis.\n",
      "----------\n",
      "TNFRSF13C Non-IDG Tbio\n",
      "B-cell receptor specific for TNFSF13B/TALL1/BAFF/BLyS. Promotes the survival of mature B-cells and the B-cell response.\n",
      "----------\n",
      "PSMC3 Enzyme Tbio\n",
      "Component of the 26S proteasome, a multiprotein complex involved in the ATP-dependent degradation of ubiquitinated proteins. This complex plays a key role in the maintenance of protein homeostasis by removing misfolded or damaged proteins, which could impair cellular functions, and by removing proteins whose functions are no longer required. Therefore, the proteasome participates in numerous cellular processes, including cell cycle progression, apoptosis, or DNA damage repair. PSMC3 belongs to the heterohexameric ring of AAA (ATPases associated with diverse cellular activities) proteins that unfolds ubiquitinated target proteins that are concurrently translocated into a proteolytic chamber and degraded into peptides.\n",
      "----------\n",
      "SMOX Enzyme Tbio\n",
      "Flavoenzyme which catalyzes the oxidation of spermine to spermidine. Can also use N(1)-acetylspermine and spermidine as substrates, with different affinity depending on the isoform (isozyme) and on the experimental conditions. Plays an important role in the regulation of polyamine intracellular concentration and has the potential to act as a determinant of cellular sensitivity to the antitumor polyamine analogs. May contribute to beta-alanine production via aldehyde dehydrogenase conversion of 3-amino-propanal.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "PML Non-IDG Tbio\n",
      "Exhibits antiviral activity against both DNA and RNA viruses. The antiviral activity can involve one or several isoform(s) and can be enhanced by the permanent PML-NB-associated protein DAXX or by the recruitment of p53/TP53 within these structures. Isoform PML-4 restricts varicella zoster virus (VZV) via sequestration of virion capsids in PML-NBs thereby preventing their nuclear egress and inhibiting formation of infectious virus particles. The sumoylated isoform PML-4 restricts rabies virus by inhibiting viral mRNA and protein synthesis. The cytoplasmic isoform PML-14 can restrict herpes simplex virus-1 (HHV-1) replication by sequestering the viral E3 ubiquitin-protein ligase ICP0 in the cytoplasm. Isoform PML-6 shows restriction activity towards human cytomegalovirus (HCMV) and influenza A virus strains PR8(H1N1) and ST364(H3N2). Sumoylated isoform PML-4 and isoform PML-12 show antiviral activity against encephalomyocarditis virus (EMCV) by promoting nuclear sequestration of viral polymerase (P3D-POL) within PML NBs. Isoform PML-3 exhibits antiviral activity against poliovirus by inducing apoptosis in infected cells through the recruitment and the activation of p53/TP53 in the PML-NBs. Isoform PML-3 represses human foamy virus (HFV) transcription by complexing the HFV transactivator, bel1/tas, preventing its binding to viral DNA. PML may positively regulate infectious hepatitis C viral (HCV) production and isoform PML-2 may enhance adenovirus transcription.\n",
      "----------\n",
      "LPAR2 GPCR Tchem\n",
      "Receptor for lysophosphatidic acid (LPA), a mediator of diverse cellular activities. Seems to be coupled to the G(i)/G(o), G(12)/G(13), and G(q) families of heteromeric G proteins. Plays a key role in phospholipase C-beta (PLC-beta) signaling pathway. Stimulates phospholipase C (PLC) activity in a manner that is independent of RALA activation.\n",
      "----------\n",
      "TEK Kinase Tclin\n",
      "Tyrosine-protein kinase that acts as cell-surface receptor for ANGPT1, ANGPT2 and ANGPT4 and regulates angiogenesis, endothelial cell survival, proliferation, migration, adhesion and cell spreading, reorganization of the actin cytoskeleton, but also maintenance of vascular quiescence. Has anti-inflammatory effects by preventing the leakage of proinflammatory plasma proteins and leukocytes from blood vessels. Required for normal angiogenesis and heart development during embryogenesis. Required for post-natal hematopoiesis. After birth, activates or inhibits angiogenesis, depending on the context. Inhibits angiogenesis and promotes vascular stability in quiescent vessels, where endothelial cells have tight contacts. In quiescent vessels, ANGPT1 oligomers recruit TEK to cell-cell contacts, forming complexes with TEK molecules from adjoining cells, and this leads to preferential activation of phosphatidylinositol 3-kinase and the AKT1 signaling cascades. In migrating endothelial cells that lack cell-cell adhesions, ANGT1 recruits TEK to contacts with the extracellular matrix, leading to the formation of focal adhesion complexes, activation of PTK2/FAK and of the downstream kinases MAPK1/ERK2 and MAPK3/ERK1, and ultimately to the stimulation of sprouting angiogenesis. ANGPT1 signaling triggers receptor dimerization and autophosphorylation at specific tyrosine residues that then serve as binding sites for scaffold proteins and effectors. Signaling is modulated by ANGPT2 that has lower affinity for TEK, can promote TEK autophosphorylation in the absence of ANGPT1, but inhibits ANGPT1-mediated signaling by competing for the same binding site. Signaling is also modulated by formation of heterodimers with TIE1, and by proteolytic processing that gives rise to a soluble TEK extracellular domain. The soluble extracellular domain modulates signaling by functioning as decoy receptor for angiopoietins. TEK phosphorylates DOK2, GRB7, GRB14, PIK3R1; SHC1 and TIE1.\n",
      "----------\n",
      "MCHR1 GPCR Tchem\n",
      "Receptor for melanin-concentrating hormone, coupled to both G proteins that inhibit adenylyl cyclase and G proteins that activate phosphoinositide hydrolysis.\n",
      "----------\n",
      "AGT Non-IDG Tbio\n",
      "Angiotensin 1-7: is a ligand for the G-protein coupled receptor MAS1. Has vasodilator and antidiuretic effects. Has an antithrombotic effect that involves MAS1-mediated release of nitric oxide from platelets.\n",
      "----------\n",
      "ERAP2 Enzyme Tchem\n",
      "Aminopeptidase that plays a central role in peptide trimming, a step required for the generation of most HLA class I-binding peptides. Peptide trimming is essential to customize longer precursor peptides to fit them to the correct length required for presentation on MHC class I molecules. Preferentially hydrolyzes the basic residues Arg and Lys.\n",
      "----------\n",
      "SLC12A4 Transporter Tbio\n",
      "Mediates electroneutral potassium-chloride cotransport when activated by cell swelling. May contribute to cell volume homeostasis in single cells. May be involved in the regulation of basolateral Cl(-) exit in NaCl absorbing epithelia (By similarity). Isoform 4 has no transport activity.\n",
      "----------\n",
      "NPPA Non-IDG Tchem\n",
      "Hormone playing a key role in cardiovascular homeostasis through regulation of natriuresis, diuresis, and vasodilation. Also plays a role in female pregnancy by promoting trophoblast invasion and spiral artery remodeling in uterus. Specifically binds and stimulates the cGMP production of the NPR1 receptor. Binds the clearance receptor NPR3.\n",
      "----------\n",
      "NDUFA2 Enzyme Tclin\n",
      "Accessory subunit of the mitochondrial membrane respiratory chain NADH dehydrogenase (Complex I), that is believed not to be involved in catalysis. Complex I functions in the transfer of electrons from NADH to the respiratory chain. The immediate electron acceptor for the enzyme is believed to be ubiquinone.\n",
      "----------\n",
      "PSCA Non-IDG Tbio\n",
      "May act as a modulator of nicotinic acetylcholine receptors (nAChRs) activity. In vitro inhibits nicotine-induced signaling probably implicating alpha-3:beta-2- or alpha-7-containing nAChRs.\n",
      "----------\n",
      "MAPK12 Kinase Tchem\n",
      "Serine/threonine kinase which acts as an essential component of the MAP kinase signal transduction pathway. MAPK12 is one of the four p38 MAPKs which play an important role in the cascades of cellular responses evoked by extracellular stimuli such as proinflammatory cytokines or physical stress leading to direct activation of transcription factors such as ELK1 and ATF2. Accordingly, p38 MAPKs phosphorylate a broad range of proteins and it has been estimated that they may have approximately 200 to 300 substrates each. Some of the targets are downstream kinases such as MAPKAPK2, which are activated through phosphorylation and further phosphorylate additional targets. Plays a role in myoblast differentiation and also in the down-regulation of cyclin D1 in response to hypoxia in adrenal cells suggesting MAPK12 may inhibit cell proliferation while promoting differentiation. Phosphorylates DLG1. Following osmotic shock, MAPK12 in the cell nucleus increases its association with nuclear DLG1, thereby causing dissociation of DLG1-SFPQ complexes. This function is independent of its catalytic activity and could affect mRNA processing and/or gene transcription to aid cell adaptation to osmolarity changes in the environment. Regulates UV-induced checkpoint signaling and repair of UV-induced DNA damage and G2 arrest after gamma-radiation exposure. MAPK12 is involved in the regulation of SLC2A1 expression and basal glucose uptake in L6 myotubes; and negatively regulates SLC2A4 expression and contraction-mediated glucose uptake in adult skeletal muscle. C-Jun (JUN) phosphorylation is stimulated by MAPK14 and inhibited by MAPK12, leading to a distinct AP-1 regulation. MAPK12 is required for the normal kinetochore localization of PLK1, prevents chromosomal instability and supports mitotic cell viability. MAPK12-signaling is also positively regulating the expansion of transient amplifying myogenic precursor cells during muscle growth and regeneration.\n",
      "----------\n",
      "SUFU Non-IDG Tbio\n",
      "Negative regulator in the hedgehog signaling pathway. Down-regulates GLI1-mediated transactivation of target genes (PubMed:15367681, PubMed:24311597, PubMed:24217340). Down-regulates GLI2-mediated transactivation of target genes (PubMed:24311597, PubMed:24217340). Part of a corepressor complex that acts on DNA-bound GLI1. May also act by linking GLI1 to BTRC and thereby targeting GLI1 to degradation by the proteasome. Sequesters GLI1, GLI2 and GLI3 in the cytoplasm, this effect is overcome by binding of STK36 to both SUFU and a GLI protein (PubMed:10806483, PubMed:24217340). Negative regulator of beta-catenin signaling. Regulates the formation of either the repressor form (GLI3R) or the activator form (GLI3A) of the full-length form of GLI3 (GLI3FL). GLI3FL is complexed with SUFU in the cytoplasm and is maintained in a neutral state. Without the Hh signal, the SUFU-GLI3 complex is recruited to cilia, leading to the efficient processing of GLI3FL into GLI3R. When Hh signaling is initiated, SUFU dissociates from GLI3FL and the latter translocates to the nucleus, where it is phosphorylated, destabilized, and converted to a transcriptional activator (GLI3A). Required for normal embryonic development. Required for the proper formation of hair follicles and the control of epidermal differentiation (By similarity).\n",
      "----------\n",
      "POU5F1 Transcription Factor Tbio\n",
      "Transcription factor that binds to the octamer motif (5'-ATTTGCAT-3'). Forms a trimeric complex with SOX2 on DNA and controls the expression of a number of genes involved in embryonic development such as YES1, FGF4, UTF1 and ZFP206. Critical for early embryogenesis and for embryonic stem cell pluripotency.\n",
      "----------\n",
      "RIPK2 Kinase Tchem\n",
      "Serine/threonine/tyrosine kinase that plays an essential role in modulation of innate and adaptive immune responses. Upon stimulation by bacterial peptidoglycans, NOD1 and NOD2 are activated, oligomerize and recruit RIPK2 through CARD-CARD domains. Contributes to the tyrosine phosphorylation of the guanine exchange factor ARHGEF2 through Src tyrosine kinase leading to NF-kappaB activation by NOD2. Once recruited, RIPK2 autophosphorylates and undergoes 'Lys-63'-linked polyubiquitination by E3 ubiquitin ligases XIAP, BIRC2 and BIRC3. The polyubiquitinated protein mediates the recruitment of MAP3K7/TAK1 to IKBKG/NEMO and induces 'Lys-63'-linked polyubiquitination of IKBKG/NEMO and subsequent activation of IKBKB/IKKB. In turn, NF-kappa-B is released from NF-kappa-B inhibitors and translocates into the nucleus where it activates the transcription of hundreds of genes involved in immune response, growth control, or protection against apoptosis. Plays also a role during engagement of the T-cell receptor (TCR) in promoting BCL10 phosphorylation and subsequent NF-kappa-B activation.\n",
      "----------\n",
      "ARNT Transcription Factor Tbio\n",
      "Required for activity of the Ah (dioxin) receptor. This protein is required for the ligand-binding subunit to translocate from the cytosol to the nucleus after ligand binding. The complex then initiates transcription of genes involved in the activation of PAH procarcinogens. The heterodimer with HIF1A or EPAS1/HIF2A functions as a transcriptional regulator of the adaptive response to hypoxia.\n",
      "----------\n",
      "ERAP1 Enzyme Tchem\n",
      "Aminopeptidase that plays a central role in peptide trimming, a step required for the generation of most HLA class I-binding peptides. Peptide trimming is essential to customize longer precursor peptides to fit them to the correct length required for presentation on MHC class I molecules. Strongly prefers substrates 9-16 residues long. Rapidly degrades 13-mer to a 9-mer and then stops. Preferentially hydrolyzes the residue Leu and peptides with a hydrophobic C-terminus, while it has weak activity toward peptides with charged C-terminus. May play a role in the inactivation of peptide hormones. May be involved in the regulation of blood pressure through the inactivation of angiotensin II and/or the generation of bradykinin in the kidney.\n",
      "----------\n",
      "CLCN3 Ion Channel Tbio\n",
      "Mediates the exchange of chloride ions against protons. Functions as antiporter and contributes to the acidification of the endosome and synaptic vesicle lumen, and may thereby affect vesicle trafficking and exocytosis. May play an important role in neuronal cell function through regulation of membrane excitability by protein kinase C. It could help neuronal cells to establish short-term memory.\n",
      "----------\n",
      "CLCN6 Ion Channel Tbio\n",
      "Chloride transport protein, initially identified as voltage-gated chloride channel. The presence of the conserved gating glutamate residues suggests that is functions as antiporter.\n",
      "----------\n",
      "BRD2 Epigenetic Tchem\n",
      "May play a role in spermatogenesis or folliculogenesis (By similarity). Binds hyperacetylated chromatin and plays a role in the regulation of transcription, probably by chromatin remodeling. Regulates transcription of the CCND1 gene. Plays a role in nucleosome assembly.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "ABCC8 Transporter Tclin\n",
      "Subunit of the beta-cell ATP-sensitive potassium channel (KATP). Regulator of ATP-sensitive K(+) channels and insulin release.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "PPID Enzyme Tchem\n",
      "PPIases accelerate the folding of proteins. It catalyzes the cis-trans isomerization of proline imidic peptide bonds in oligopeptides. Proposed to act as a co-chaperone in HSP90 complexes such as in unligated steroid receptors heterocomplexes. Different co-chaperones seem to compete for association with HSP90 thus establishing distinct HSP90-co-chaperone-receptor complexes with the potential to exert tissue-specific receptor activity control. May have a preference for estrogen receptor complexes and is not found in glucocorticoid receptor complexes. May be involved in cytoplasmic dynein-dependent movement of the receptor from the cytoplasm to the nucleus. May regulate MYB by inhibiting its DNA-binding activity. Involved in regulation of AHR signaling by promoting the formation of the AHR:ARNT dimer; the function is independent of HSP90 but requires the chaperone activity. Involved in regulation of UV radiation-induced apoptosis. Promotes cell viability in anaplastic lymphoma kinase-positive anaplastic large-cell lymphoma (ALK+ ALCL) cell lines. May be involved in hepatitis C virus (HCV) replication and release.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "SOCS1 Non-IDG Tbio\n",
      "SOCS family proteins form part of a classical negative feedback system that regulates cytokine signal transduction. SOCS1 is involved in negative regulation of cytokines that signal through the JAK/STAT3 pathway. Through binding to JAKs, inhibits their kinase activity. In vitro, also suppresses Tec protein-tyrosine activity. Appears to be a major regulator of signaling by interleukin 6 (IL6) and leukemia inhibitory factor (LIF). Regulates interferon-gamma mediated sensory neuron survival (By similarity). Probable substrate recognition component of an ECS (Elongin BC-CUL2/5-SOCS-box protein) E3 ubiquitin ligase complex which mediates the ubiquitination and subsequent proteasomal degradation of target proteins. Seems to recognize JAK2. SOCS1 appears to be a negative regulator in IGF1R signaling pathway.\n",
      "----------\n",
      "ACADM Enzyme Tbio\n",
      "Acyl-CoA dehydrogenase specific for acyl chain lengths of 4 to 16 that catalyzes the initial step of fatty acid beta-oxidation. Utilizes the electron transfer flavoprotein (ETF) as an electron acceptor to transfer electrons to the main mitochondrial respiratory chain via ETF-ubiquinone oxidoreductase (ETF dehydrogenase).\n",
      "----------\n",
      "PNMT Enzyme Tchem\n",
      "Converts noradrenaline to adrenaline.\n",
      "----------\n",
      "RARRES1 Enzyme Tbio\n",
      "Inhibitor of the cytoplasmic carboxypeptidase AGBL2, may regulate the alpha-tubulin tyrosination cycle.\n",
      "----------\n",
      "FANCA Non-IDG Tbio\n",
      "DNA repair protein that may operate in a postreplication repair or a cell cycle checkpoint function. May be involved in interstrand DNA cross-link repair and in the maintenance of normal chromosome stability.\n",
      "----------\n",
      "CPT1B Enzyme Tchem\n",
      "nan\n",
      "----------\n",
      "SREBF1 Transcription Factor Tbio\n",
      "Transcriptional activator required for lipid homeostasis. Regulates transcription of the LDL receptor gene as well as the fatty acid and to a lesser degree the cholesterol synthesis pathway (By similarity). Binds to the sterol regulatory element 1 (SRE-1) (5'-ATCACCCCAC-3'). Has dual sequence specificity binding to both an E-box motif (5'-ATCACGTGA-3') and to SRE-1 (5'-ATCACCCCAC-3').\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-DRB5 Non-IDG Tbio\n",
      "Binds peptides derived from antigens that access the endocytic route of antigen presenting cells (APC) and presents them on the cell surface for recognition by the CD4 T-cells. The peptide binding cleft accommodates peptides of 10-30 residues. The peptides presented by MHC class II molecules are generated mostly by degradation of proteins that access the endocytic route, where they are processed by lysosomal proteases and other hydrolases. Exogenous antigens that have been endocytosed by the APC are thus readily available for presentation via MHC II molecules, and for this reason this antigen presentation pathway is usually referred to as exogenous. As membrane proteins on their way to degradation in lysosomes as part of their normal turn-over are also contained in the endosomal/lysosomal compartments, exogenous antigens must compete with those derived from endogenous components. Autophagy is also a source of endogenous peptides, autophagosomes constitutively fuse with MHC class II loading compartments. In addition to APCs, other cells of the gastrointestinal tract, such as epithelial cells, express MHC class II molecules and CD74 and act as APCs, which is an unusual trait of the GI tract. To produce a MHC class II molecule that presents an antigen, three MHC class II molecules (heterodimers of an alpha and a beta chain) associate with a CD74 trimer in the ER to form a heterononamer. Soon after the entry of this complex into the endosomal/lysosomal system where antigen processing occurs, CD74 undergoes a sequential degradation by various proteases, including CTSS and CTSL, leaving a small fragment termed CLIP (class-II-associated invariant chain peptide). The removal of CLIP is facilitated by HLA-DM via direct binding to the alpha-beta-CLIP complex so that CLIP is released. HLA-DM stabilizes MHC class II molecules until primary high affinity antigenic peptides are bound. The MHC II molecule bound to a peptide is then transported to the cell membrane surface. In B-cells, the interaction between HLA-DM and MHC class II molecules is regulated by HLA-DO. Primary dendritic cells (DCs) also to express HLA-DO. Lysosomal microenvironment has been implicated in the regulation of antigen loading into MHC II molecules, increased acidification produces increased proteolysis and efficient peptide loading.\n",
      "----------\n",
      "TIE1 Kinase Tchem\n",
      "Transmembrane tyrosine-protein kinase that may modulate TEK/TIE2 activity and contribute to the regulation of angiogenesis.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "AGER Non-IDG Tchem\n",
      "Mediates interactions of advanced glycosylation end products (AGE). These are nonenzymatically glycosylated proteins which accumulate in vascular tissue in aging and at an accelerated rate in diabetes. Acts as a mediator of both acute and chronic vascular inflammation in conditions such as atherosclerosis and in particular as a complication of diabetes. AGE/RAGE signaling plays an important role in regulating the production/expression of TNF-alpha, oxidative stress, and endothelial dysfunction in type 2 diabetes. Interaction with S100A12 on endothelium, mononuclear phagocytes, and lymphocytes triggers cellular activation, with generation of key proinflammatory mediators. Interaction with S100B after myocardial infarction may play a role in myocyte apoptosis by activating ERK1/2 and p53/TP53 signaling (By similarity). Receptor for amyloid beta peptide. Contributes to the translocation of amyloid-beta peptide (ABPP) across the cell membrane from the extracellular to the intracellular space in cortical neurons. ABPP-initiated RAGE signaling, especially stimulation of p38 mitogen-activated protein kinase (MAPK), has the capacity to drive a transport system delivering ABPP as a complex with RAGE to the intraneuronal space. Can also bind oligonucleotides.\n",
      "----------\n",
      "EPHX2 Enzyme Tchem\n",
      "Bifunctional enzyme (PubMed:12574510). The C-terminal domain has epoxide hydrolase activity and acts on epoxides (alkene oxides, oxiranes) and arene oxides (PubMed:12869654, PubMed:12574510, PubMed:22798687). Plays a role in xenobiotic metabolism by degrading potentially toxic epoxides (By similarity). Also determines steady-state levels of physiological mediators (PubMed:12869654, PubMed:12574510, PubMed:22798687). The N-terminal domain has lipid phosphatase activity, with the highest activity towards threo-9,10-phosphonooxy-hydroxy-octadecanoic acid, followed by erythro-9,10-phosphonooxy-hydroxy-octadecanoic acid, 12-phosphonooxy-octadec-9Z-enoic acid and 12-phosphonooxy-octadec-9E-enoic acid (PubMed:12574510).\n",
      "----------\n",
      "PIK3C2A Kinase Tchem\n",
      "Generates phosphatidylinositol 3-phosphate (PtdIns3P) and phosphatidylinositol 3,4-bisphosphate (PtdIns(3,4)P2) that act as second messengers. Has a role in several intracellular trafficking events. Functions in insulin signaling and secretion. Required for translocation of the glucose transporter SLC2A4/GLUT4 to the plasma membrane and glucose uptake in response to insulin-mediated RHOQ activation. Regulates insulin secretion through two different mechanisms: involved in glucose-induced insulin secretion downstream of insulin receptor in a pathway that involves AKT1 activation and TBC1D4/AS160 phosphorylation, and participates in the late step of insulin granule exocytosis probably in insulin granule fusion. Synthesizes PtdIns3P in response to insulin signaling. Functions in clathrin-coated endocytic vesicle formation and distribution. Regulates dynamin-independent endocytosis, probably by recruiting EEA1 to internalizing vesicles. In neurosecretory cells synthesizes PtdIns3P on large dense core vesicles. Participates in calcium induced contraction of vascular smooth muscle by regulating myosin light chain (MLC) phosphorylation through a mechanism involving Rho kinase-dependent phosphorylation of the MLCP-regulatory subunit MYPT1. May play a role in the EGF signaling cascade. May be involved in mitosis and UV-induced damage response. Required for maintenance of normal renal structure and function by supporting normal podocyte function.\n",
      "----------\n",
      "CD14 Non-IDG Tbio\n",
      "Coreceptor for bacterial lipopolysaccharide (PubMed:1698311, PubMed:23264655). In concert with LBP, binds to monomeric lipopolysaccharide and delivers it to the LY96/TLR4 complex, thereby mediating the innate immune response to bacterial lipopolysaccharide (LPS) (PubMed:20133493, PubMed:23264655). Acts via MyD88, TIRAP and TRAF6, leading to NF-kappa-B activation, cytokine secretion and the inflammatory response (PubMed:8612135). Acts as a coreceptor for TLR2:TLR6 heterodimer in response to diacylated lipopeptides and for TLR2:TLR1 heterodimer in response to triacylated lipopeptides, these clusters trigger signaling from the cell surface and subsequently are targeted to the Golgi in a lipid-raft dependent pathway (PubMed:16880211). Binds electronegative LDL (LDL(-)) and mediates the cytokine release induced by LDL(-) (PubMed:23880187).\n",
      "----------\n",
      "CHEK1 Kinase Tchem\n",
      "Isoform 2: Endogenous repressor of isoform 1, interacts with, and antagonizes CHK1 to promote the S to G2/M phase transition.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "PPP1R1B Enzyme Tbio\n",
      "Inhibitor of protein-phosphatase 1.\n",
      "----------\n",
      "NDUFA6 Enzyme Tclin\n",
      "Accessory subunit of the mitochondrial membrane respiratory chain NADH dehydrogenase (Complex I), that is believed to be not involved in catalysis. Complex I functions in the transfer of electrons from NADH to the respiratory chain. The immediate electron acceptor for the enzyme is believed to be ubiquinone.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "PTK2B Kinase Tchem\n",
      "Non-receptor protein-tyrosine kinase that regulates reorganization of the actin cytoskeleton, cell polarization, cell migration, adhesion, spreading and bone remodeling. Plays a role in the regulation of the humoral immune response, and is required for normal levels of marginal B-cells in the spleen and normal migration of splenic B-cells. Required for normal macrophage polarization and migration towards sites of inflammation. Regulates cytoskeleton rearrangement and cell spreading in T-cells, and contributes to the regulation of T-cell responses. Promotes osteoclastic bone resorption; this requires both PTK2B/PYK2 and SRC. May inhibit differentiation and activity of osteoprogenitor cells. Functions in signaling downstream of integrin and collagen receptors, immune receptors, G-protein coupled receptors (GPCR), cytokine, chemokine and growth factor receptors, and mediates responses to cellular stress. Forms multisubunit signaling complexes with SRC and SRC family members upon activation; this leads to the phosphorylation of additional tyrosine residues, creating binding sites for scaffold proteins, effectors and substrates. Regulates numerous signaling pathways. Promotes activation of phosphatidylinositol 3-kinase and of the AKT1 signaling cascade. Promotes activation of NOS3. Regulates production of the cellular messenger cGMP. Promotes activation of the MAP kinase signaling cascade, including activation of MAPK1/ERK2, MAPK3/ERK1 and MAPK8/JNK1. Promotes activation of Rho family GTPases, such as RHOA and RAC1. Recruits the ubiquitin ligase MDM2 to P53/TP53 in the nucleus, and thereby regulates P53/TP53 activity, P53/TP53 ubiquitination and proteasomal degradation. Acts as a scaffold, binding to both PDPK1 and SRC, thereby allowing SRC to phosphorylate PDPK1 at 'Tyr-9, 'Tyr-373', and 'Tyr-376'. Promotes phosphorylation of NMDA receptors by SRC family members, and thereby contributes to the regulation of NMDA receptor ion channel activity and intracellular Ca(2+) levels. May also regulate potassium ion transport by phosphorylation of potassium channel subunits. Phosphorylates SRC; this increases SRC kinase activity. Phosphorylates ASAP1, NPHP1, KCNA2 and SHC1. Promotes phosphorylation of ASAP2, RHOU and PXN; this requires both SRC and PTK2/PYK2.\n",
      "----------\n",
      "FYN Kinase Tclin\n",
      "Non-receptor tyrosine-protein kinase that plays a role in many biological processes including regulation of cell growth and survival, cell adhesion, integrin-mediated signaling, cytoskeletal remodeling, cell motility, immune response and axon guidance. Inactive FYN is phosphorylated on its C-terminal tail within the catalytic domain. Following activation by PKA, the protein subsequently associates with PTK2/FAK1, allowing PTK2/FAK1 phosphorylation, activation and targeting to focal adhesions. Involved in the regulation of cell adhesion and motility through phosphorylation of CTNNB1 (beta-catenin) and CTNND1 (delta-catenin). Regulates cytoskeletal remodeling by phosphorylating several proteins including the actin regulator WAS and the microtubule-associated proteins MAP2 and MAPT. Promotes cell survival by phosphorylating AGAP2/PIKE-A and preventing its apoptotic cleavage. Participates in signal transduction pathways that regulate the integrity of the glomerular slit diaphragm (an essential part of the glomerular filter of the kidney) by phosphorylating several slit diaphragm components including NPHS1, KIRREL1 and TRPC6. Plays a role in neural processes by phosphorylating DPYSL2, a multifunctional adapter protein within the central nervous system, ARHGAP32, a regulator for Rho family GTPases implicated in various neural functions, and SNCA, a small pre-synaptic protein. Participates in the downstream signaling pathways that lead to T-cell differentiation and proliferation following T-cell receptor (TCR) stimulation. Also participates in negative feedback regulation of TCR signaling through phosphorylation of PAG1, thereby promoting interaction between PAG1 and CSK and recruitment of CSK to lipid rafts. CSK maintains LCK and FYN in an inactive form. Promotes CD28-induced phosphorylation of VAV1.\n",
      "----------\n",
      "TNFRSF17 Non-IDG Tbio\n",
      "Receptor for TNFSF13B/BLyS/BAFF and TNFSF13/APRIL. Promotes B-cell survival and plays a role in the regulation of humoral immunity. Activates NF-kappa-B and JNK.\n",
      "----------\n",
      "MVK Kinase Tbio\n",
      "May be a regulatory site in cholesterol biosynthetic pathway.\n",
      "----------\n",
      "PEMT Enzyme Tbio\n",
      "Catalyzes the three sequential steps of the methylation pathway of phosphatidylcholine biosynthesis, the SAM-dependent methylation of phosphatidylethanolamine (PE) to phosphatidylmonomethylethanolamine (PMME), PMME to phosphatidyldimethylethanolamine (PDME), and PDME to phosphatidylcholine (PC).\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "MICA Non-IDG Tbio\n",
      "Seems to have no role in antigen presentation. Acts as a stress-induced self-antigen that is recognized by gamma delta T-cells. Ligand for the KLRK1/NKG2D receptor. Binding to KLRK1 leads to cell lysis.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "ALMS1 Non-IDG Tbio\n",
      "Involved in PCM1-dependent intracellular transport. Required, directly or indirectly, for the localization of NCAPD2 to the proximal ends of centrioles. Required for proper formation and/or maintenance of primary cilia (PC), microtubule-based structures that protrude from the surface of epithelial cells.\n",
      "----------\n",
      "MAP3K11 Kinase Tchem\n",
      "Activates the JUN N-terminal pathway. Required for serum-stimulated cell proliferation and for mitogen and cytokine activation of MAPK14 (p38), MAPK3 (ERK) and MAPK8 (JNK1) through phosphorylation and activation of MAP2K4/MKK4 and MAP2K7/MKK7. Plays a role in mitogen-stimulated phosphorylation and activation of BRAF, but does not phosphorylate BRAF directly. Influences microtubule organization during the cell cycle.\n",
      "----------\n",
      "MDK Non-IDG Tchem\n",
      "Developmentally regulated, secreted growth factor homologous to pleiotrophin (PTN), which has heparin binding activity. Binds anaplastic lymphoma kinase (ALK) which induces ALK activation and subsequent phosphorylation of the insulin receptor substrate (IRS1), followed by the activation of mitogen-activated protein kinase (MAPK) and PI3-kinase, and the induction of cell proliferation. Involved in neointima formation after arterial injury, possibly by mediating leukocyte recruitment. Also involved in early fetal adrenal gland development (By similarity).\n",
      "----------\n",
      "OSGIN2 Non-IDG Tbio\n",
      "May be involved in meiosis or the maturation of germ cells.\n",
      "----------\n",
      "SRPK2 Kinase Tchem\n",
      "Serine/arginine-rich protein-specific kinase which specifically phosphorylates its substrates at serine residues located in regions rich in arginine/serine dipeptides, known as RS domains and is involved in the phosphorylation of SR splicing factors and the regulation of splicing. Promotes neuronal apoptosis by up-regulating cyclin-D1 (CCND1) expression. This is done by the phosphorylation of SRSF2, leading to the suppression of p53/TP53 phosphorylation thereby relieving the repressive effect of p53/TP53 on cyclin-D1 (CCND1) expression. Phosphorylates ACIN1, and redistributes it from the nuclear speckles to the nucleoplasm, resulting in cyclin A1 but not cyclin A2 up-regulation. Plays an essential role in spliceosomal B complex formation via the phosphorylation of DDX23/PRP28. Can mediate hepatitis B virus (HBV) core protein phosphorylation. Plays a negative role in the regulation of HBV replication through a mechanism not involving the phosphorylation of the core protein but by reducing the packaging efficiency of the pregenomic RNA (pgRNA) without affecting the formation of the viral core particles.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HMOX2 Enzyme Tbio\n",
      "Heme oxygenase cleaves the heme ring at the alpha methene bridge to form biliverdin. Biliverdin is subsequently converted to bilirubin by biliverdin reductase. Under physiological conditions, the activity of heme oxygenase is highest in the spleen, where senescent erythrocytes are sequestrated and destroyed. Heme oxygenase 2 could be implicated in the production of carbon monoxide in brain where it could act as a neurotransmitter.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "NOTCH4 Non-IDG Tchem\n",
      "Functions as a receptor for membrane-bound ligands Jagged1, Jagged2 and Delta1 to regulate cell-fate determination. Upon ligand activation through the released notch intracellular domain (NICD) it forms a transcriptional activator complex with RBPJ/RBPSUH and activates genes of the enhancer of split locus. Affects the implementation of differentiation, proliferation and apoptotic programs. May regulate branching morphogenesis in the developing vascular system (By similarity).\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "LCAT Enzyme Tbio\n",
      "Central enzyme in the extracellular metabolism of plasma lipoproteins. Synthesized mainly in the liver and secreted into plasma where it converts cholesterol and phosphatidylcholines (lecithins) to cholesteryl esters and lysophosphatidylcholines on the surface of high and low density lipoproteins (HDLs and LDLs) (PubMed:10329423, PubMed:19065001, PubMed:26195816). The cholesterol ester is then transported back to the liver. Has a preference for plasma 16:0-18:2 or 18:O-18:2 phosphatidylcholines (PubMed:8820107). Also produced in the brain by primary astrocytes, and esterifies free cholesterol on nascent APOE-containing lipoproteins secreted from glia and influences cerebral spinal fluid (CSF) APOE- and APOA1 levels. Together with APOE and the cholesterol transporter ABCA1, plays a key role in the maturation of glial-derived, nascent lipoproteins. Required for remodeling high-density lipoprotein particles into their spherical forms (PubMed:10722751).\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-C Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "MAP3K1 Kinase Tbio\n",
      "Component of a protein kinase signal transduction cascade (PubMed:9808624). Activates the ERK and JNK kinase pathways by phosphorylation of MAP2K1 and MAP2K4 (PubMed:9808624). May phosphorylate the MAPK8/JNK1 kinase (PubMed:17761173). Activates CHUK and IKBKB, the central protein kinases of the NF-kappa-B pathway (PubMed:9808624).\n",
      "----------\n",
      "CAST Non-IDG Tbio\n",
      "Specific inhibition of calpain (calcium-dependent cysteine protease). Plays a key role in postmortem tenderization of meat and have been proposed to be involved in muscle protein degradation in living tissue.\n",
      "----------\n",
      "NR1H3 Nuclear Receptor Tchem\n",
      "Nuclear receptor. Interaction with RXR shifts RXR from its role as a silent DNA-binding partner to an active ligand-binding subunit in mediating retinoid responses through target genes defined by LXRES. LXRES are DR4-type response elements characterized by direct repeats of two similar hexanuclotide half-sites spaced by four nucleotides. Plays an important role in the regulation of cholesterol homeostasis, regulating cholesterol uptake through MYLIP-dependent ubiquitination of LDLR, VLDLR and LRP8. Interplays functionally with RORA for the regulation of genes involved in liver metabolism (By similarity). Exhibits a ligand-dependent transcriptional activation activity (PubMed:25661920).\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "HLA-B Non-IDG Tbio\n",
      "Involved in the presentation of foreign antigens to the immune system.\n",
      "----------\n",
      "SERPING1 Enzyme Tbio\n",
      "Activation of the C1 complex is under control of the C1-inhibitor. It forms a proteolytically inactive stoichiometric complex with the C1r or C1s proteases. May play a potentially crucial role in regulating important physiological pathways including complement activation, blood coagulation, fibrinolysis and the generation of kinins. Very efficient inhibitor of FXIIa. Inhibits chymotrypsin and kallikrein.\n",
      "----------\n",
      "TYMP Enzyme Tclin\n",
      "Catalyzes the reversible phosphorolysis of thymidine. The produced molecules are then utilized as carbon and energy sources or in the rescue of pyrimidine bases for nucleotide synthesis.\n",
      "----------\n",
      "XPNPEP3 Enzyme Tbio\n",
      "nan\n"
     ]
    }
   ],
   "source": [
    "druggable_genes = drug_df[drug_df.gene.isin(all_genes)].iloc[:,1:6].drop_duplicates()\n",
    "druggable_genes_filt = druggable_genes[~druggable_genes.gene.isin(all_drug_info_filt.gene)]#.gene.sort_values().values\n",
    "print(druggable_genes_filt.columns)\n",
    "for idx,row in druggable_genes_filt.iterrows():\n",
    "    print('----------')\n",
    "    print(row['gene'], row['idgFamily'],row['idgTDL'])\n",
    "    print(row['description'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "from reading above, added: \n",
    "- CACNA2D2\n",
    "- CLCN3\n",
    "- CLCN6\n",
    "- PTK2B\n",
    "\n",
    "b/c they have generally neural functions (ion channels are especially emphasied) and/or protein atlas states some brain-specific tissue specificity\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rowname</th>\n",
       "      <th>Linked_SNP</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref/Alt</th>\n",
       "      <th>Index_SNP</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>diseases</th>\n",
       "      <th>num_dz</th>\n",
       "      <th>...</th>\n",
       "      <th>cgene_egene_agree_num</th>\n",
       "      <th>bool_cgene_egene_agree</th>\n",
       "      <th>ukbb_drug</th>\n",
       "      <th>ukbb_phenotype</th>\n",
       "      <th>ukbb_phenotype_beta</th>\n",
       "      <th>SNP_id</th>\n",
       "      <th>psychencode_egenes</th>\n",
       "      <th>egenes_all_sources</th>\n",
       "      <th>abc_genes</th>\n",
       "      <th>abc_gene_tissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>632</td>\n",
       "      <td>chr8_27317337</td>\n",
       "      <td>rs2280375</td>\n",
       "      <td>chr8</td>\n",
       "      <td>27317337</td>\n",
       "      <td>A/G</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>8:27317337</td>\n",
       "      <td></td>\n",
       "      <td>PTK2B|TRIM35</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>634</td>\n",
       "      <td>chr8_27318391</td>\n",
       "      <td>rs2292974</td>\n",
       "      <td>chr8</td>\n",
       "      <td>27318391</td>\n",
       "      <td>C/T</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>depression history|worry</td>\n",
       "      <td>0.020|-0.009</td>\n",
       "      <td>8:27318391</td>\n",
       "      <td></td>\n",
       "      <td>CHRNA2|PTK2B</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>635</td>\n",
       "      <td>chr8_27324844</td>\n",
       "      <td>rs2565061</td>\n",
       "      <td>chr8</td>\n",
       "      <td>27324844</td>\n",
       "      <td>G/A</td>\n",
       "      <td></td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>SCZ</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>8:27324844</td>\n",
       "      <td></td>\n",
       "      <td>PTK2B|TRIM35</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           rowname Linked_SNP   Chr  Position Ref/Alt Index_SNP  ConsScore  \\\n",
       "632  chr8_27317337  rs2280375  chr8  27317337     A/G                    1   \n",
       "634  chr8_27318391  rs2292974  chr8  27318391     C/T                    1   \n",
       "635  chr8_27324844  rs2565061  chr8  27324844     G/A                    5   \n",
       "\n",
       "     ConsDetail diseases  num_dz  ... cgene_egene_agree_num  \\\n",
       "632  downstream      SCZ       1  ...                   0.0   \n",
       "634  downstream      SCZ       1  ...                   0.0   \n",
       "635  synonymous      SCZ       1  ...                   0.0   \n",
       "\n",
       "    bool_cgene_egene_agree ukbb_drug            ukbb_phenotype  \\\n",
       "632                  False                                       \n",
       "634                  False            depression history|worry   \n",
       "635                  False                                       \n",
       "\n",
       "    ukbb_phenotype_beta      SNP_id psychencode_egenes egenes_all_sources  \\\n",
       "632                      8:27317337                          PTK2B|TRIM35   \n",
       "634        0.020|-0.009  8:27318391                          CHRNA2|PTK2B   \n",
       "635                      8:27324844                          PTK2B|TRIM35   \n",
       "\n",
       "    abc_genes  abc_gene_tissues  \n",
       "632                              \n",
       "634                              \n",
       "635                              \n",
       "\n",
       "[3 rows x 54 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# data_df[data_df.egenes_all_sources.str.contains('CACNA2D2')].iloc[0,:]\n",
    "# data_df[data_df.egenes_all_sources.str.contains('CLCN3')].iloc[0,:]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.2546555795690967"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_df[data_df.egenes_all_sources.str.contains('CLCN6')].mpra_logfc_mean.mean()#.iloc[0,:]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.3227474124252467"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_df[data_df.egenes_all_sources.str.contains('PTK2B')].mpra_logfc_mean.mean()#.iloc[0,:]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "619"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# which genes have previously been druged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pert_iname</th>\n",
       "      <th>clinical_phase</th>\n",
       "      <th>moa</th>\n",
       "      <th>target</th>\n",
       "      <th>disease_area</th>\n",
       "      <th>indication</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>(R)-(-)-apomorphine</td>\n",
       "      <td>Launched</td>\n",
       "      <td>dopamine receptor agonist</td>\n",
       "      <td>ADRA2A|ADRA2B|ADRA2C|CALY|DRD1|DRD2|DRD3|DRD4|...</td>\n",
       "      <td>neurology/psychiatry</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>(R)-(-)-rolipram</td>\n",
       "      <td>Phase 1</td>\n",
       "      <td>phosphodiesterase inhibitor</td>\n",
       "      <td>PDE4A|PDE4B|PDE4C|PDE4D|PDE5A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>(R)-baclofen</td>\n",
       "      <td>Phase 3</td>\n",
       "      <td>benzodiazepine receptor agonist</td>\n",
       "      <td>GABBR1|GABBR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>(S)-(+)-rolipram</td>\n",
       "      <td>Phase 1</td>\n",
       "      <td>phosphodiesterase inhibitor</td>\n",
       "      <td>PDE4B|PDE4D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>[sar9,met(o2)11]-substance-p</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>tachykinin antagonist</td>\n",
       "      <td>TACR1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6793</td>\n",
       "      <td>8-M-PDOT</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>melatonin receptor agonist</td>\n",
       "      <td>MTNR1A|MTNR1B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6794</td>\n",
       "      <td>80841-78-7</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6795</td>\n",
       "      <td>9-aminoacridine</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6796</td>\n",
       "      <td>9-aminocamptothecin</td>\n",
       "      <td>Phase 2</td>\n",
       "      <td>topoisomerase inhibitor</td>\n",
       "      <td>TOP1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6797</td>\n",
       "      <td>9-anthracenecarboxylic-acid</td>\n",
       "      <td>Preclinical</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ANO1|CLCN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6798 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        pert_iname clinical_phase  \\\n",
       "0              (R)-(-)-apomorphine       Launched   \n",
       "1                 (R)-(-)-rolipram        Phase 1   \n",
       "2                     (R)-baclofen        Phase 3   \n",
       "3                 (S)-(+)-rolipram        Phase 1   \n",
       "4     [sar9,met(o2)11]-substance-p    Preclinical   \n",
       "...                            ...            ...   \n",
       "6793                      8-M-PDOT    Preclinical   \n",
       "6794                    80841-78-7    Preclinical   \n",
       "6795               9-aminoacridine    Preclinical   \n",
       "6796           9-aminocamptothecin        Phase 2   \n",
       "6797   9-anthracenecarboxylic-acid    Preclinical   \n",
       "\n",
       "                                  moa  \\\n",
       "0           dopamine receptor agonist   \n",
       "1         phosphodiesterase inhibitor   \n",
       "2     benzodiazepine receptor agonist   \n",
       "3         phosphodiesterase inhibitor   \n",
       "4               tachykinin antagonist   \n",
       "...                               ...   \n",
       "6793       melatonin receptor agonist   \n",
       "6794                              NaN   \n",
       "6795                              NaN   \n",
       "6796          topoisomerase inhibitor   \n",
       "6797                              NaN   \n",
       "\n",
       "                                                 target          disease_area  \\\n",
       "0     ADRA2A|ADRA2B|ADRA2C|CALY|DRD1|DRD2|DRD3|DRD4|...  neurology/psychiatry   \n",
       "1                         PDE4A|PDE4B|PDE4C|PDE4D|PDE5A                   NaN   \n",
       "2                                         GABBR1|GABBR2                   NaN   \n",
       "3                                           PDE4B|PDE4D                   NaN   \n",
       "4                                                 TACR1                   NaN   \n",
       "...                                                 ...                   ...   \n",
       "6793                                      MTNR1A|MTNR1B                   NaN   \n",
       "6794                                                NaN                   NaN   \n",
       "6795                                                NaN                   NaN   \n",
       "6796                                               TOP1                   NaN   \n",
       "6797                                         ANO1|CLCN1                   NaN   \n",
       "\n",
       "               indication  \n",
       "0     Parkinson's Disease  \n",
       "1                     NaN  \n",
       "2                     NaN  \n",
       "3                     NaN  \n",
       "4                     NaN  \n",
       "...                   ...  \n",
       "6793                  NaN  \n",
       "6794                  NaN  \n",
       "6795                  NaN  \n",
       "6796                  NaN  \n",
       "6797                  NaN  \n",
       "\n",
       "[6798 rows x 6 columns]"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('/Users/mguo123/Google Drive/0_altman/db/cmap/repurposing_drugs_20200324.txt',header=9, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
